<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>List of R packages for Ecology: A list of Ecology R packages</title>
  
  <meta property="description" itemprop="description" content="Feel free to suggest other packages to add to the list (link to issues)"/>
  
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="List of R packages for Ecology: A list of Ecology R packages"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Feel free to suggest other packages to add to the list (link to issues)"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="List of R packages for Ecology"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="List of R packages for Ecology: A list of Ecology R packages"/>
  <meta property="twitter:description" content="Feel free to suggest other packages to add to the list (link to issues)"/>
  
  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description"]}},"value":[{"type":"character","attributes":{},"value":["A list of Ecology R packages"]},{"type":"character","attributes":{},"value":["Feel free to suggest other packages to add to the list (link to issues)"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"A list of Ecology R packages","description":"Feel free to suggest other packages to add to the list (link to issues)","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">List of R packages for Ecology</a>
</div>
<div class="nav-right">
<a href="index.html">Home</a>
<a href="about.html">About</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>A list of Ecology R packages</h1>
<p><p>Feel free to suggest other packages to add to the list (link to issues)</p></p>
</div>


<div class="d-article">
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="r-packages">R packages</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Short Description
</th>
<th style="text-align:left;">
Long Description
</th>
<th style="text-align:left;">
Source
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
soundecology
</td>
<td style="text-align:left;">
Soundscape Ecology
</td>
<td style="text-align:left;">
Functions to calculate indices for soundscape ecology and other ecology research that uses audio recordings.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nimbleEcology
</td>
<td style="text-align:left;">
Distributions for Ecological Models in ‘nimble’
</td>
<td style="text-align:left;">
Common ecological distributions for ‘nimble’ models in the form of nimbleFunction objects. Includes Cormack-Jolly-Seber, occupancy, dynamic occupancy, hidden Markov, dynamic hidden Markov, and N-mixture models. (Jolly (1965) &lt;DOI: 10.2307/2333826&gt;, Seber (1965) &lt;DOI: 10.2307/2333827&gt;, Turek et al. (2016) &lt;<a href="doi:10.1007/s10651-016-0353-z%3E" class="uri">doi:10.1007/s10651-016-0353-z&gt;</a>;).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
RevEcoR
</td>
<td style="text-align:left;">
Reverse Ecology Analysis on Microbiome
</td>
<td style="text-align:left;">
An implementation of the reverse ecology framework. Reverse ecology refers to the use of genomics to study ecology with no a priori assumptions about the organism(s) under consideration, linking organisms to their environment. It allows researchers to reconstruct the metabolic networks and study the ecology of poorly characterized microbial species from their genomic information, and has substantial potentials for microbial community ecological analysis.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
vegan
</td>
<td style="text-align:left;">
Community Ecology Package
</td>
<td style="text-align:left;">
Ordination methods, diversity analysis and other functions for community and vegetation ecologists.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
CommEcol
</td>
<td style="text-align:left;">
Community Ecology Analyses
</td>
<td style="text-align:left;">
Autosimilarity curves, standardization of spatial extent, dissimilarity indexes that overweight rare species, phylogenetic and functional (pairwise and multisample) dissimilarity indexes and nestedness for phylogenetic, functional and other diversity metrics. This should be a complement to available packages, particularly ‘vegan’.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
ecospat
</td>
<td style="text-align:left;">
Spatial Ecology Miscellaneous Methods
</td>
<td style="text-align:left;">
Collection of R functions and data sets for the support of spatial ecology analyses with a focus on pre, core and post modelling analyses of species distribution, niche quantification and community assembly. Written by current and former members and collaborators of the ecospat group of Antoine Guisan, Department of Ecology and Evolution (DEE) and Institute of Earth Surface Dynamics (IDYST), University of Lausanne, Switzerland. Read Di Cola et al. (2016) &lt;<a href="doi:10.1111/ecog.02671%3E" class="uri">doi:10.1111/ecog.02671&gt;</a>; for details.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
BayesComm
</td>
<td style="text-align:left;">
Bayesian Community Ecology Analysis
</td>
<td style="text-align:left;">
Bayesian multivariate binary (probit) regression models for analysis of ecological communities.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
picante
</td>
<td style="text-align:left;">
Integrating Phylogenies and Ecology
</td>
<td style="text-align:left;">
Functions for phylocom integration, community analyses, null-models, traits and evolution. Implements numerous ecophylogenetic approaches including measures of community phylogenetic and trait diversity, phylogenetic signal, estimation of trait values for unobserved taxa, null models for community and phylogeny randomizations, and utility functions for data input/output and phylogeny plotting. A full description of package functionality and methods are provided by Kembel et al. (2010) &lt;<a href="doi:10.1093/bioinformatics/btq166%3E" class="uri">doi:10.1093/bioinformatics/btq166&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
dave
</td>
<td style="text-align:left;">
Functions for “Data Analysis in Vegetation Ecology”
</td>
<td style="text-align:left;">
A collection of functions accompanying the book “Data Analysis in Vegetation Ecology”. 3rd ed. CABI, Oxfordshire, Boston.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
letsR
</td>
<td style="text-align:left;">
Data Handling and Analysis in Macroecology
</td>
<td style="text-align:left;">
Handling, processing, and analyzing geographic data on species’ distributions and environmental variables. Read Vilela &amp; Villalobos (2015) &lt;doi: 10.1111/2041-210X.12401&gt; for details.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
forams
</td>
<td style="text-align:left;">
Foraminifera and Community Ecology Analyses
</td>
<td style="text-align:left;">
SHE, FORAM Index and ABC Method analyses and custom plot functions for community data.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
labdsv
</td>
<td style="text-align:left;">
Ordination and Multivariate Analysis for Ecology
</td>
<td style="text-align:left;">
A variety of ordination and community analyses useful in analysis of data sets in community ecology.<br />
Includes many of the common ordination methods, with graphical routines to facilitate their interpretation, as well as several novel analyses.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
phyloregion
</td>
<td style="text-align:left;">
Biogeographic Regionalization and Macroecology
</td>
<td style="text-align:left;">
Computational infrastructure for biogeography, community ecology, and biodiversity conservation (Daru et al. 2020) &lt;<a href="doi:10.1101/2020.02.12.945691%3E" class="uri">doi:10.1101/2020.02.12.945691&gt;</a>;. It is based on the methods described in Daru et al. (2020) &lt;<a href="doi:10.1038/s41467-020-15921-6%3E" class="uri">doi:10.1038/s41467-020-15921-6&gt;</a>;. The original conceptual work is described in Daru et al. (2017) &lt;<a href="doi:10.1016/j.tree.2017.08.013%3E" class="uri">doi:10.1016/j.tree.2017.08.013&gt;</a>; on patterns and processes of biogeographical regionalization. Additionally, the package contains fast and efficient functions to compute more standard conservation measures such as phylogenetic diversity, phylogenetic endemism, evolutionary distinctiveness and global endangerment, as well as compositional turnover (e.g., beta diversity).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
RAM
</td>
<td style="text-align:left;">
R for Amplicon-Sequencing-Based Microbial-Ecology
</td>
<td style="text-align:left;">
Characterizing environmental microbiota diversity using amplicon-based next generation sequencing (NGS) data. Functions are developed to manipulate operational taxonomic unit (OTU) table, perform descriptive and inferential statistics, and generate publication-quality plots.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
analogue
</td>
<td style="text-align:left;">
Analogue and Weighted Averaging Methods for Palaeoecology
</td>
<td style="text-align:left;">
Fits Modern Analogue Technique and Weighted Averaging transfer function models for prediction of environmental data from species data, and related methods used in palaeoecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SSP
</td>
<td style="text-align:left;">
Simulated Sampling Procedure for Community Ecology
</td>
<td style="text-align:left;">
Simulation-based sampling protocol (SSP) is an R package design to estimate sampling effort in studies of ecological communities based on the definition of pseudo-multivariate standard error (MultSE) (Anderson &amp; Santana-Garcon, 2015) &lt;<a href="doi:10.1111/ele.12385%3E" class="uri">doi:10.1111/ele.12385&gt;</a>; and simulation of ecological data. The theoretical background is described in Guerra-Castro et al. (2020) &lt;<a href="doi:10.1101/2020.03.19.996991%3E" class="uri">doi:10.1101/2020.03.19.996991&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
BiodiversityR
</td>
<td style="text-align:left;">
Package for Community Ecology and Suitability Analysis
</td>
<td style="text-align:left;">
Graphical User Interface (via the R-Commander) and utility functions (often based on the vegan package) for statistical analysis of biodiversity and ecological communities, including species accumulation curves, diversity indices, Renyi profiles, GLMs for analysis of species abundance and presence-absence, distance matrices, Mantel tests, and cluster, constrained and unconstrained ordination analysis. A book on biodiversity and community ecology analysis is available for free download from the website. In 2012, methods for (ensemble) suitability modelling and mapping were expanded in the package.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
mefa
</td>
<td style="text-align:left;">
Multivariate Data Handling in Ecology and Biogeography
</td>
<td style="text-align:left;">
A framework package aimed to provide standardized computational environment for specialist work via object classes to represent the data coded by samples, taxa and segments (i.e. subpopulations, repeated measures). It supports easy processing of the data along with cross tabulation and relational data tables for samples and taxa. An object of class <code>mefa' is a project specific compendium of the data and can be easily used in further analyses. Methods are provided for extraction, aggregation, conversion, plotting, summary and reporting of</code>mefa’ objects. Reports can be generated in plain text or LaTeX format. Vignette contains worked examples.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
tsvr
</td>
<td style="text-align:left;">
Timescale-Specific Variance Ratio for Use in Community Ecology
</td>
<td style="text-align:left;">
Tools for timescale decomposition of the classic variance ratio of community ecology. Tools are as described in Zhao et al (in prep), extending commonly used methods introduced by Peterson et al (1975) &lt;doi: 10.2307/1936306&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
meteR
</td>
<td style="text-align:left;">
Fitting and Plotting Tools for the Maximum Entropy Theory of Ecology (METE)
</td>
<td style="text-align:left;">
Fit and plot macroecological patterns predicted by the Maximum Entropy Theory of Ecology (METE).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
swapClass
</td>
<td style="text-align:left;">
A Null Model Adapted to Abundance Class Data in Ecology
</td>
<td style="text-align:left;">
A null model randomizing semi-quantitative multi-classes (or ordinal) data by swapping sub-matrices while both the row and the column marginal sums are held constant.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
primer
</td>
<td style="text-align:left;">
Functions and Data for the Book, A Primer of Ecology with R
</td>
<td style="text-align:left;">
Functions are primarily functions for systems of ordinary differential equations, difference equations, and eigenanalysis and projection of demographic matrices; data are for examples. Documentation of methods is provided in Stevens, MHH (2009, &lt;<a href="https://www.springer.com/gp/book/9780387898810%3E" class="uri">https://www.springer.com/gp/book/9780387898810&gt;</a>;).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
CNull
</td>
<td style="text-align:left;">
Fast Algorithms for Frequency-Preserving Null Models in Ecology
</td>
<td style="text-align:left;">
Efficient computations for null models that require shuffling columns on big matrix data. This package provides functions for faster computation of diversity measure statistics when independent random shuffling is applied to the columns of a given matrix. Given a diversity measure f and a matrix M, the provided functions can generate random samples (shuffled matrix rows of M), the mean and variance of f, and the p-values of this measure for two different null models that involve independent random shuffling of the columns of M. The package supports computations of alpha and beta diversity measures.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
rangeMapper
</td>
<td style="text-align:left;">
A Platform for the Study of Macro-Ecology of Life History Traits
</td>
<td style="text-align:left;">
Tools for easy generation of (life-history) traits maps based on species range (extent-of-occurrence) maps.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
BIEN
</td>
<td style="text-align:left;">
Tools for Accessing the Botanical Information and Ecology Network Database
</td>
<td style="text-align:left;">
Provides Tools for Accessing the Botanical Information and Ecology Network Database. The BIEN database contains cleaned and standardized botanical data including occurrence, trait, plot and taxonomic data (See &lt;<a href="http://Bien.nceas.ucsb.edu/bien/%3E" class="uri">http://Bien.nceas.ucsb.edu/bien/&gt;</a>; for more Information). This package provides functions that query the BIEN database by constructing and executing optimized SQL queries.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
wsyn
</td>
<td style="text-align:left;">
Wavelet Approaches to Studies of Synchrony in Ecology and Other Fields
</td>
<td style="text-align:left;">
Tools for a wavelet-based approach to analyzing spatial synchrony, principally in ecological data. Some tools will be useful for studying community synchrony. See, for instance, Sheppard et al (2016) &lt;doi: 10.1038/NCLIMATE2991&gt;, Sheppard et al (2017) &lt;doi: 10.1051/epjnbp/2017000&gt;, Sheppard et al (2019) &lt;doi: 10.1371/journal.pcbi.1006744&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
DSAIDE
</td>
<td style="text-align:left;">
Dynamical Systems Approach to Infectious Disease Epidemiology (Ecology/Evolution)
</td>
<td style="text-align:left;">
Exploration of simulation models (apps) of various infectious disease transmission dynamics scenarios. The purpose of the package is to help individuals learn about infectious disease epidemiology (ecology/evolution) from a dynamical systems perspective. All apps include explanations of the underlying models and instructions on what to do with the models.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
AHMbook
</td>
<td style="text-align:left;">
Functions and Data for the Book “Applied Hierarchical Modeling in Ecology” Vols 1 and 2
</td>
<td style="text-align:left;">
Provides functions and data sets to accompany the two volume publication “Applied Hierarchical Modeling in Ecology: Analysis of distribution, abundance and species richness in R and BUGS” by Marc Kéry and Andy Royle: volume 1 (2016, ISBN: 978-0-12-801378-6) and volume 2 (2020, ISBN: 978-0-12-809585-0), &lt;<a href="https://www.mbr-pwrc.usgs.gov/pubanalysis/keryroylebook%3E" class="uri">https://www.mbr-pwrc.usgs.gov/pubanalysis/keryroylebook&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
MCMC.OTU
</td>
<td style="text-align:left;">
Bayesian Analysis of Multivariate Counts Data in DNA Metabarcoding and Ecology
</td>
<td style="text-align:left;">
Poisson-lognormal generalized linear mixed model analysis of multivariate counts data using MCMC, aiming to infer the changes in relative proportions of individual variables. The package was originally designed for sequence-based analysis of microbial communities (“metabarcoding”, variables = operational taxonomic units, OTUs), but can be used for other types of multivariate counts, such as in ecological applications (variables = species). The results are summarized and plotted using ‘ggplot2’ functions. Includes functions to remove sample and variable outliers and reformat counts into normalized log-transformed values for correlation and principal component/coordinate analysis. Walkthrough and examples: <a href="http://www.bio.utexas.edu/research/matz_lab/matzlab/Methods_files/walkthroughExample_mcmcOTU_R.txt" class="uri">http://www.bio.utexas.edu/research/matz_lab/matzlab/Methods_files/walkthroughExample_mcmcOTU_R.txt</a>.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
FD
</td>
<td style="text-align:left;">
Measuring functional diversity (FD) from multiple traits, and other tools for functional ecology
</td>
<td style="text-align:left;">
FD is a package to compute different multidimensional FD indices. It implements a distance-based framework to measure FD that allows any number and type of functional traits, and can also consider species relative abundances. It also contains other useful tools for functional ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
blmeco
</td>
<td style="text-align:left;">
Data Files and Functions Accompanying the Book “Bayesian Data Analysis in Ecology using R, BUGS and Stan”
</td>
<td style="text-align:left;">
Data files and functions accompanying the book Korner-Nievergelt, Roth, von Felten, Guelat, Almasi, Korner-Nievergelt (2015) “Bayesian Data Analysis in Ecology using R, BUGS and Stan”, Elsevier, New York.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
loop
</td>
<td style="text-align:left;">
loop decomposition of weighted directed graphs for life cycle analysis, providing flexbile network plotting methods, and analyzing food chain properties in ecology
</td>
<td style="text-align:left;">
The program can perform loop analysis and plot network structure (especially for food webs),including minimum spanning tree, loop decomposition of weighted directed graphs, and other network properties which may be related to food chain properties in ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
paleoMAS
</td>
<td style="text-align:left;">
Paleoecological Analysis
</td>
<td style="text-align:left;">
Transfer functions and statistical operations for paleoecology
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
cepreader
</td>
<td style="text-align:left;">
Read ‘CEP’ and Legacy ‘CANOCO’ Files
</td>
<td style="text-align:left;">
Read Condensed Cornell Ecology Program (‘CEP’) and legacy ‘CANOCO’ files into R data frames.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
hydrostats
</td>
<td style="text-align:left;">
Hydrologic Indices for Daily Time Series Data
</td>
<td style="text-align:left;">
Calculates a suite of hydrologic indices for daily time series data that are widely used in hydrology and stream ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
inpdfr
</td>
<td style="text-align:left;">
Analyse Text Documents Using Ecological Tools
</td>
<td style="text-align:left;">
A set of functions to analyse and compare texts, using classical text mining functions, as well as those from theoretical ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
FLightR
</td>
<td style="text-align:left;">
Reconstruct Animal Paths from Solar Geolocation Loggers Data
</td>
<td style="text-align:left;">
Spatio-temporal locations of an animal are computed from annotated data with a hidden Markov model via particle filter algorithm. The package is relatively robust to varying degrees of shading. The hidden Markov model is described in Movement Ecology (Rakhimberdiev et al., 2015) &lt;<a href="doi:10.1186/s40462-015-0062-5%3E" class="uri">doi:10.1186/s40462-015-0062-5&gt;</a>;, general package description is in the Methods in Ecology and Evolution (Rakhimberdiev et al., 2017) &lt;<a href="doi:10.1111/2041-210X.12765%3E" class="uri">doi:10.1111/2041-210X.12765&gt;</a>; and package accuracy assessed in the Journal of Avian Biology (Rakhimberdiev et al. 2016) &lt;<a href="doi:10.1111/jav.00891%3E" class="uri">doi:10.1111/jav.00891&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
treeclim
</td>
<td style="text-align:left;">
Numerical Calibration of Proxy-Climate Relationships
</td>
<td style="text-align:left;">
Bootstrapped response and correlation functions, seasonal correlations and evaluation of reconstruction skills for use in dendroclimatology and dendroecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
DAISIE
</td>
<td style="text-align:left;">
Dynamical Assembly of Islands by Speciation, Immigration and Extinction
</td>
<td style="text-align:left;">
Simulates and computes the (maximum) likelihood of a dynamical model of island biota assembly through speciation, immigration and extinction. See e.g. Valente et al. 2015. Ecology Letters 18: 844-852, &lt;<a href="DOI:10.1111/ele.12461%3E" class="uri">DOI:10.1111/ele.12461&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
expp
</td>
<td style="text-align:left;">
Spatial Analysis of Extra-Pair Paternity
</td>
<td style="text-align:left;">
Tools and data to accompany Schlicht, L., Valcu, M., &amp; Kempenaers, B. (2015) &lt;<a href="doi:10.1111/1365-2656.12293%3E" class="uri">doi:10.1111/1365-2656.12293&gt;</a>;. Spatial patterns of extra-pair paternity: beyond paternity gains and losses. Journal of Animal Ecology, 84(2), 518-531.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
betaper
</td>
<td style="text-align:left;">
Taxonomic Uncertainty on Multivariate Analyses of Ecological Data
</td>
<td style="text-align:left;">
Permutational method to incorporate taxonomic uncertainty and some functions to assess its effects on parameters of some widely used multivariate methods in ecology, as explained in Cayuela et al. (2011) &lt;<a href="doi:10.1111/j.1600-0587.2009.05899.x%3E" class="uri">doi:10.1111/j.1600-0587.2009.05899.x&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
enveomics.R
</td>
<td style="text-align:left;">
Various Utilities for Microbial Genomics and Metagenomics
</td>
<td style="text-align:left;">
A collection of functions for microbial ecology and other applications of genomics and metagenomics. Companion package for the Enveomics Collection (Rodriguez-R, L.M. and Konstantinidis, K.T., 2016 &lt;<a href="DOI:10.7287/peerj.preprints.1900v1%3E" class="uri">DOI:10.7287/peerj.preprints.1900v1&gt;</a>;).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
sharx
</td>
<td style="text-align:left;">
Models and Data Sets for the Study of Species-Area Relationships
</td>
<td style="text-align:left;">
Hierarchical models for the analysis of species-area relationships (SARs) by combining several data sets and covariates; with a global data set combining individual SAR studies; as described in Solymos and Lele (2012, Global Ecology and Biogeography 21, 109-120).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
move
</td>
<td style="text-align:left;">
Visualizing and Analyzing Animal Track Data
</td>
<td style="text-align:left;">
Contains functions to access movement data stored in ‘movebank.org’ as well as tools to visualize and statistically analyze animal movement data, among others functions to calculate dynamic Brownian Bridge Movement Models. Move helps addressing movement ecology questions.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
petitr
</td>
<td style="text-align:left;">
Relative Growth Rate
</td>
<td style="text-align:left;">
Calculates the relative growth rate (RGR) of a series of individuals by building a life table and solving the Lotka-Birch equation. (See Birch, L. C. 1948. The intrinsic rate of natural increase of an insect population. - Journal of Animal Ecology 17: 15-26) &lt;<a href="doi:10.2307/1605%3E" class="uri">doi:10.2307/1605&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
portalr
</td>
<td style="text-align:left;">
Create Useful Summaries of the Portal Data
</td>
<td style="text-align:left;">
Download and generate summaries for the rodent, plant, ant, and weather data from the Portal Project. Portal is a long-term (and ongoing) experimental monitoring site in the Chihuahua desert. The raw data files can be found at &lt;<a href="https://github.com/weecology/portaldata%3E" class="uri">https://github.com/weecology/portaldata&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
fso
</td>
<td style="text-align:left;">
Fuzzy Set Ordination
</td>
<td style="text-align:left;">
Fuzzy set ordination is a multivariate analysis used in ecology to relate the composition of samples to possible explanatory variables. While differing in theory and method, in practice, the use is similar to ‘constrained ordination.’ The package contains plotting and summary functions as well as the analyses.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
polyfreqs
</td>
<td style="text-align:left;">
Bayesian Population Genomics in Autopolyploids
</td>
<td style="text-align:left;">
Implements a Gibbs sampling algorithm to perform Bayesian inference on biallelic SNP frequencies, genotypes and heterozygosity (observed and expected) in a population of autopolyploids. See the published paper in Molecular Ecology Resources: Blischak et al. (2016) &lt;<a href="doi:10.1111/1755-0998.12493%3E" class="uri">doi:10.1111/1755-0998.12493&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
virtualspecies
</td>
<td style="text-align:left;">
Generation of Virtual Species Distributions
</td>
<td style="text-align:left;">
Provides a framework for generating virtual species distributions, a procedure increasingly used in ecology to improve species distribution models. This package integrates the existing methodological approaches with the objective of generating virtual species distributions with increased ecological realism.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
RADanalysis
</td>
<td style="text-align:left;">
Normalization and Study of Rank Abundance Distributions
</td>
<td style="text-align:left;">
It has tools for normalization of rank abundance distributions (RAD) to a desired number of ranks using MaxRank Normalization method. RADs are commonly used in biology/ecology and mathematically equivalent to complementary cumulative distributions (CCDFs) which are used in physics, linguistics and sociology and more generally in data science.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SitesInterest
</td>
<td style="text-align:left;">
Inferring an Animal’s Sites of Interest from High Resolution Data
</td>
<td style="text-align:left;">
Within the area of movement ecology, it can be useful to identify an animal’s sites of interest. These will be places that an animal uses a lot and so will spend a proportionately longer time there. The aim of this package is to identify both the position and size of these sites of interest from high resolution (sub-second) positional data.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
bcpa
</td>
<td style="text-align:left;">
Behavioral change point analysis of animal movement
</td>
<td style="text-align:left;">
The Behavioral Change Point Analysis (BCPA) is a method of identifying hidden shifts in the underlying parameters of a time series, developed specifically to be applied to animal movement data which is irregularly sampled. The method is based on: E. Gurarie, R. Andrews and K. Laidre A novel method for identifying behavioural changes in animal movement data (2009) Ecology Letters 12:5 395-408.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
ecospace
</td>
<td style="text-align:left;">
Simulating Community Assembly and Ecological Diversification Using Ecospace Frameworks
</td>
<td style="text-align:left;">
Implements stochastic simulations of community assembly (ecological diversification) using customizable ecospace frameworks (functional trait spaces). Provides a wrapper to calculate common ecological disparity and functional ecology statistical dynamics as a function of species richness. Functions are written so they will work in a parallel-computing environment.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
mvabund
</td>
<td style="text-align:left;">
Statistical Methods for Analysing Multivariate Abundance Data
</td>
<td style="text-align:left;">
A set of tools for displaying, modeling and analysing multivariate abundance data in community ecology. See ‘mvabund-package.Rd’ for details of overall package organization. The package is implemented with the Gnu Scientific Library (&lt;<a href="http://www.gnu.org/software/gsl/%3E" class="uri">http://www.gnu.org/software/gsl/&gt;</a>;) and ‘Rcpp’ (&lt;<a href="http://dirk.eddelbuettel.com/code/rcpp.html%3E" class="uri">http://dirk.eddelbuettel.com/code/rcpp.html&gt;</a>;) ‘R’ / ‘C++’ classes.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
raceland
</td>
<td style="text-align:left;">
Pattern-Based Zoneless Method for Analysis and Visualization of Racial Topography
</td>
<td style="text-align:left;">
Implements a computational framework for a pattern-based, zoneless analysis, and visualization of (ethno)racial topography (Dmowska, Stepinski, and Nowosad (2020) &lt;<a href="doi:10.1016/j.apgeog.2020.102239%3E" class="uri">doi:10.1016/j.apgeog.2020.102239&gt;</a>;). It is a reimagined approach for analyzing residential segregation and racial diversity based on the concept of ‘landscape’ used in the domain of landscape ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
rfishbase
</td>
<td style="text-align:left;">
R Interface to ‘FishBase’
</td>
<td style="text-align:left;">
A programmatic interface to &lt;<a href="http://www.fishbase.org%3E" class="uri">http://www.fishbase.org&gt;</a>;, re-written based on an accompanying ‘RESTful’ API. Access tables describing over 30,000 species of fish, their biology, ecology, morphology, and more. This package also supports experimental access to &lt;<a href="http://www.sealifebase.org%3E" class="uri">http://www.sealifebase.org&gt;</a>; data, which contains nearly 200,000 species records for all types of aquatic life not covered by ‘FishBase.’
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
sampSurf
</td>
<td style="text-align:left;">
Sampling Surface Simulation for Areal Sampling Methods
</td>
<td style="text-align:left;">
Sampling surface simulation is useful in the comparison of different areal sampling methods in forestry, ecology and natural resources. The sampSurf package allows the simulation of numerous sampling methods for standing trees and downed woody debris in a spatial context. It also provides an S4 class and method structure that facilitates the addition of new sampling methods.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SiMRiv
</td>
<td style="text-align:left;">
Simulating Multistate Movements in River/Heterogeneous Landscapes
</td>
<td style="text-align:left;">
Provides functions to generate and analyze spatially-explicit individual-based multistate movements in rivers, heterogeneous and homogeneous spaces. This is done by incorporating landscape bias on local behaviour, based on resistance rasters. Although originally conceived and designed to simulate trajectories of species constrained to linear habitats/dendritic ecological networks (e.g. river networks), the simulation algorithm is built to be highly flexible and can be applied to any (aquatic, semi-aquatic or terrestrial) organism, independently on the landscape in which it moves. Thus, the user will be able to use the package to simulate movements either in homogeneous landscapes, heterogeneous landscapes (e.g. semi-aquatic animal moving mainly along rivers but also using the matrix), or even in highly contrasted landscapes (e.g. fish in a river network). The algorithm and its input parameters are the same for all cases, so that results are comparable. Simulated trajectories can then be used as mechanistic null models (Potts &amp; Lewis 2014, &lt;<a href="DOI:10.1098/rspb.2014.0231%3E" class="uri">DOI:10.1098/rspb.2014.0231&gt;</a>;) to test a variety of ‘Movement Ecology’ hypotheses (Nathan et al. 2008, &lt;<a href="DOI:10.1073/pnas.0800375105%3E" class="uri">DOI:10.1073/pnas.0800375105&gt;</a>;), including landscape effects (e.g. resources, infrastructures) on animal movement and species site fidelity, or for predictive purposes (e.g. road mortality risk, dispersal/connectivity). The package should be relevant to explore a broad spectrum of ecological phenomena, such as those at the interface of animal behaviour, management, landscape and movement ecology, disease and invasive species spread, and population dynamics.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
bmotif
</td>
<td style="text-align:left;">
Motif Analyses of Bipartite Networks
</td>
<td style="text-align:left;">
Counts occurrences of motifs in bipartite networks, as well as the number of times each node or link appears in each unique position within motifs. Has support for both binary and weighted motifs: can calculate the mean weight of motifs and the standard deviation of their mean weights. Intended for use in ecology, but its methods are general and can be applied to any bipartite network. Full details are given in Simmons et al. (2019) &lt;<a href="doi:10.1111/2041-210X.13149%3E" class="uri">doi:10.1111/2041-210X.13149&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
EMbC
</td>
<td style="text-align:left;">
Expectation-Maximization Binary Clustering
</td>
<td style="text-align:left;">
Unsupervised, multivariate, binary clustering for meaningful annotation of data, taking into account the uncertainty in the data. A specific constructor for trajectory analysis in movement ecology yields behavioural annotation of trajectories based on estimated local measures of velocity and turning angle, eventually with solar position covariate as a daytime indicator, (“Expectation-Maximization Binary Clustering for Behavioural Annotation”).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
NitrogenUptake2016
</td>
<td style="text-align:left;">
Data and Source Code From: Nitrogen Uptake and Allocation Estimates for Spartina Alterniflora and Distichlis Spicata
</td>
<td style="text-align:left;">
Contains data, code, and figures from Hill et al. 2018a (Journal of Experimental Marine Biology and Ecology; &lt;DOI: 10.1016/j.jembe.2018.07.006&gt;) and Hill et al. 2018b (Data In Brief &lt;DOI: 10.1016/j.dib.2018.09.133&gt;). Datasets document plant allometry, stem heights, nutrient and stable isotope content, and sediment denitrification enzyme assays. The data and analysis offer an examination of nitrogen uptake and allocation in two salt marsh plant species.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nna
</td>
<td style="text-align:left;">
Nearest-Neighbor Analysis
</td>
<td style="text-align:left;">
Calculates spatial pattern analysis using a T-square sample procedure. This method is based on two measures “x” and “y”. “x” - Distance from the random point to the nearest individual. “y” - Distance from individual to its nearest neighbor. This is a methodology commonly used in phytosociology or marine benthos ecology to analyze the species’ distribution (random, uniform or clumped patterns). Ludwig &amp; Reynolds (1988, <a href="ISBN:0471832359" class="uri">ISBN:0471832359</a>).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
VGAMdata
</td>
<td style="text-align:left;">
Data Supporting the ‘VGAM’ Package
</td>
<td style="text-align:left;">
Data sets to accompany the VGAM package and the book “Vector Generalized Linear and Additive Models: With an Implementation in R” (Yee, 2015) &lt;<a href="DOI:10.1007/978-1-4939-2818-7%3E" class="uri">DOI:10.1007/978-1-4939-2818-7&gt;</a>;. These are used to illustrate vector generalized linear and additive models (VGLMs/VGAMs), and associated models (Reduced-Rank VGLMs, Quadratic RR-VGLMs, Row-Column Interaction Models, and constrained and unconstrained ordination models in ecology).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
EcoSimR
</td>
<td style="text-align:left;">
Null Model Analysis for Ecological Data
</td>
<td style="text-align:left;">
Given a site by species interaction matrix, users can make inferences about species interactions by performance hypothesis comparing test statistics against a null distribution. The current package provides algorithms and metrics for niche-overlap, body size ratios and species co-occurrence. Users can also integrate their own algorithms and metrics within these frameworks or completely novel null models. Detailed explanations about the underlying assumptions of null model analysis in ecology can be found at <a href="http://ecosimr.org" class="uri">http://ecosimr.org</a>.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
gfcanalysis
</td>
<td style="text-align:left;">
Tools for Working with Hansen et al. Global Forest Change Dataset
</td>
<td style="text-align:left;">
Supports analyses using the Global Forest Change dataset released by Hansen et al. gfcanalysis was originally written for the Tropical Ecology Assessment and Monitoring (TEAM) Network. For additional details on the Global Forest Change dataset, see: Hansen, M. et al. 2013. “High-Resolution Global Maps of 21st-Century Forest Cover Change.” Science 342 (15 November): 850-53. The forest change data and more information on the product is available at &lt;<a href="http://earthenginepartners.appspot.com%3E" class="uri">http://earthenginepartners.appspot.com&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
OCNet
</td>
<td style="text-align:left;">
Optimal Channel Networks
</td>
<td style="text-align:left;">
Generate and analyze Optimal Channel Networks (OCNs): oriented spanning trees reproducing all scaling features characteristic of real, natural river networks. As such, they can be used in a variety of numerical experiments in the fields of hydrology, ecology and epidemiology. See Carraro et al. (2020) &lt;<a href="doi:10.1002/ece3.6479%3E" class="uri">doi:10.1002/ece3.6479&gt;</a>; for a presentation of the package; Rinaldo et al. (2014) &lt;<a href="doi:10.1073/pnas.1322700111%3E" class="uri">doi:10.1073/pnas.1322700111&gt;</a>; for a theoretical overview on the OCN concept; Furrer and Sain (2010) &lt;<a href="doi:10.18637/jss.v036.i10%3E" class="uri">doi:10.18637/jss.v036.i10&gt;</a>; for the construct used.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
philentropy
</td>
<td style="text-align:left;">
Similarity and Distance Quantification Between Probability Functions
</td>
<td style="text-align:left;">
Computes 46 optimized distance and similarity measures for comparing probability functions (Drost (2018) &lt;<a href="doi:10.21105/joss.00765%3E" class="uri">doi:10.21105/joss.00765&gt;</a>;). These comparisons between probability functions have their foundations in a broad range of scientific disciplines from mathematics to ecology. The aim of this package is to provide a core framework for clustering, classification, statistical inference, goodness-of-fit, non-parametric statistics, information theory, and machine learning tasks that are based on comparing univariate or multivariate probability functions.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
pop
</td>
<td style="text-align:left;">
A Flexible Syntax for Population Dynamic Modelling
</td>
<td style="text-align:left;">
Population dynamic models underpin a range of analyses and applications in ecology and epidemiology. The various approaches for analysing population dynamics models (MPMs, IPMs, ODEs, POMPs, PVA) each require the model to be defined in a different way. This makes it difficult to combine different modelling approaches and data types to solve a given problem. ‘pop’ aims to provide a flexible and easy to use common interface for constructing population dynamic models and enabling to them to be fitted and analysed in lots of different ways.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
QFASA
</td>
<td style="text-align:left;">
Quantitative Fatty Acid Signature Analysis
</td>
<td style="text-align:left;">
Accurate estimates of the diets of predators are required in many areas of ecology, but for many species current methods are imprecise, limited to the last meal, and often biased. The diversity of fatty acids and their patterns in organisms, coupled with the narrow limitations on their biosynthesis, properties of digestion in monogastric animals, and the prevalence of large storage reservoirs of lipid in many predators, led us to propose the use of quantitative fatty acid signature analysis (QFASA) to study predator diets.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SADISA
</td>
<td style="text-align:left;">
Species Abundance Distributions with Independent-Species Assumption
</td>
<td style="text-align:left;">
Computes the probability of a set of species abundances of a single or multiple samples of individuals with one or more guilds under a mainland-island model. One must specify the mainland (metacommunity) model and the island (local) community model. It assumes that species fluctuate independently. The package also contains functions to simulate under this model. See Haegeman, B. &amp; R.S. Etienne (2017). A general sampling formula for community structure data. Methods in Ecology &amp; Evolution 8: 1506-1519 &lt;<a href="doi:10.1111/2041-210X.12807%3E" class="uri">doi:10.1111/2041-210X.12807&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
virtualNicheR
</td>
<td style="text-align:left;">
Generating Virtual Fundamental and Realised Niches
</td>
<td style="text-align:left;">
The ability to generate virtual niches enables virtual ecology experiments that have been advocated for evaluating species-environment relationship models. Using multivariate normal distributions and matrix algebra this package provides a unique computational framework to generate integrated fundamental and realised niches, which is important given differences in possible complexity of realised niche shapes that result from biotic interactions Etherington and Omondiagbe (2019) &lt;<a href="doi:10.21105/joss.01661%3E" class="uri">doi:10.21105/joss.01661&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
aPCoA
</td>
<td style="text-align:left;">
Covariate Adjusted PCoA Plot
</td>
<td style="text-align:left;">
In fields such as ecology, microbiology, and genomics, non-Euclidean distances are widely applied to describe pairwise dissimilarity between samples. Given these pairwise distances, principal coordinates analysis (PCoA) is commonly used to construct a visualization of the data. However, confounding covariates can make patterns related to the scientific question of interest difficult to observe. We provide ‘aPCoA’ as an easy-to-use tool to improve data visualization in this context, enabling enhanced presentation of the effects of interest. Details are described in Yushu Shi, Liangliang Zhang, Kim-Anh Do, Christine Peterson and Robert Jenq (2020) &lt;arXiv:2003.09544&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
boral
</td>
<td style="text-align:left;">
Bayesian Ordination and Regression AnaLysis
</td>
<td style="text-align:left;">
Bayesian approaches for analyzing multivariate data in ecology. Estimation is performed using Markov Chain Monte Carlo (MCMC) methods via Three. JAGS types of models may be fitted: 1) With explanatory variables only, boral fits independent column Generalized Linear Models (GLMs) to each column of the response matrix; 2) With latent variables only, boral fits a purely latent variable model for model-based unconstrained ordination; 3) With explanatory and latent variables, boral fits correlated column GLMs with latent variables to account for any residual correlation between the columns of the response matrix.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
ENMTools
</td>
<td style="text-align:left;">
Analysis of Niche Evolution using Niche and Distribution Models
</td>
<td style="text-align:left;">
Description: Tools for constructing niche models and analyzing patterns of niche evolution. Acts as an interface for many popular modeling algorithms, and allows users to conduct Monte Carlo tests to address basic questions in evolutionary ecology and biogeography. Warren, D.L., R.E. Glor, and M. Turelli (2008) &lt;<a href="doi:10.1111/j.1558-5646.2008.00482.x%3E" class="uri">doi:10.1111/j.1558-5646.2008.00482.x&gt;</a>; Glor, R.E., and D.L. Warren (2011) &lt;<a href="doi:10.1111/j.1558-5646.2010.01177.x%3E" class="uri">doi:10.1111/j.1558-5646.2010.01177.x&gt;</a>; Warren, D.L., R.E. Glor, and M. Turelli (2010) &lt;<a href="doi:10.1111/j.1600-0587.2009.06142.x%3E" class="uri">doi:10.1111/j.1600-0587.2009.06142.x&gt;</a>; Cardillo, M., and D.L. Warren (2016) &lt;<a href="doi:10.1111/geb.12455%3E" class="uri">doi:10.1111/geb.12455&gt;</a>; D.L. Warren, L.J. Beaumont, R. Dinnage, and J.B. Baumgartner (2019) &lt;<a href="doi:10.1111/ecog.03900%3E" class="uri">doi:10.1111/ecog.03900&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nodiv
</td>
<td style="text-align:left;">
Compares the Distribution of Sister Clades Through a Phylogeny
</td>
<td style="text-align:left;">
An implementation of the nodiv algorithm, see Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. &lt;<a href="DOI:10.1111/2041-210X.12283%3E" class="uri">DOI:10.1111/2041-210X.12283&gt;</a>;. Package for phylogenetic analysis of species distributions. The main function goes through each node in the phylogeny, compares the distributions of the two descendant nodes, and compares the result to a null model. This highlights nodes where major distributional divergence have occurred. The distributional divergence for these nodes is mapped using the SOS statistic.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
pgirmess
</td>
<td style="text-align:left;">
Spatial Analysis and Data Mining for Field Ecologists
</td>
<td style="text-align:left;">
Set of tools for reading, writing and transforming spatial and seasonal data in ecology, model selection and specific statistical tests. It includes functions to discretize polylines into regular point intervals, link observations to those points, compute geographical coordinates at regular intervals between waypoints, read subsets of big rasters, compute zonal statistics or table of categories within polygons or circular buffers from raster. The package also provides miscellaneous functions for model selection, spatial statistics, geometries, writing data.frame with Chinese characters, and some other functions for field ecologists.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
seas
</td>
<td style="text-align:left;">
Seasonal Analysis and Graphics, Especially for Climatology
</td>
<td style="text-align:left;">
Capable of deriving seasonal statistics, such as “normals”, and analysis of seasonal data, such as departures. This package also has graphics capabilities for representing seasonal data, including boxplots for seasonal parameters, and bars for summed normals. There are many specific functions related to climatology, including precipitation normals, temperature normals, cumulative precipitation departures and precipitation interarrivals. However, this package is designed to represent any time-varying parameter with a discernible seasonal signal, such as found in hydrology and ecology.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SP2000
</td>
<td style="text-align:left;">
Catalogue of Life Toolkit
</td>
<td style="text-align:left;">
A programmatic interface to &lt;<a href="http://sp2000.org.cn%3E" class="uri">http://sp2000.org.cn&gt;</a>;, re-written based on an accompanying ‘Species 2000’ API. Access tables describing catalogue of the Chinese known species of animals, plants, fungi, micro-organisms, and more. This package also supports access to catalogue of life global &lt;<a href="http://catalogueoflife.org%3E" class="uri">http://catalogueoflife.org&gt;</a>; and catalogue of life Taiwan &lt;<a href="https://taibnet.sinica.edu.tw/home_eng.php%3E" class="uri">https://taibnet.sinica.edu.tw/home_eng.php&gt;</a>;. The development of ‘SP2000’ package were supported by Biodiversity Survey and Assessment Project of the Ministry of Ecology and Environment, China &lt;2019HJ2096001006&gt;,Yunnan University’s “Double First Class” Project &lt;C176240405&gt; and Yunnan University’s Research Innovation Fund for Graduate Students &lt;2019227&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
stockR
</td>
<td style="text-align:left;">
Identifying Stocks in Genetic Data
</td>
<td style="text-align:left;">
Provides a mixture model for clustering individuals (or sampling groups) into stocks based on their genetic profile. Here, sampling groups are individuals that are sure to come from the same stock (e.g. breeding adults or larvae). The mixture (log-)likelihood is maximised using the EM-algorithm after find good starting values via a K-means clustering of the genetic data. Details can be found in Foster, Feutry, Grewe, Berry, Hui, Davies (2019) Reliably Discriminating Stock Structure with Genetic Markers: Mixture Models with Robust and Fast Computation. Molecular Ecology Resources.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
zoon
</td>
<td style="text-align:left;">
Reproducible, Accessible &amp; Shareable Species Distribution Modelling
</td>
<td style="text-align:left;">
Reproducible and remixable species distribution modelling. The package reads user submitted modules from an online repository, runs full species distribution modelling workflows and returns output that is fully reproducible. For examples and detailed discussion refer to: N.Golding et al. (2017) ‘The zoon r package for reproducible and shareable species distribution modelling’. Methods in Ecology and Evolution. &lt;<a href="doi:10.1111/2041-210X.12858%3E" class="uri">doi:10.1111/2041-210X.12858&gt;</a>;. The package ‘SDMTools’ is used for testing and, though this package is archived, you can access it here if needed, &lt;<a href="https://cran.r-project.org/src/contrib/Archive/SDMTools/%3E" class="uri">https://cran.r-project.org/src/contrib/Archive/SDMTools/&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
allelematch
</td>
<td style="text-align:left;">
Identifying Unique Multilocus Genotypes where Genotyping Error and Missing Data may be Present
</td>
<td style="text-align:left;">
Tools for the identification of unique of multilocus genotypes when both genotyping error and missing data may be present. The package is targeted at those working with large datasets and databases containing multiple samples of each individual, a situation that is common in conservation genetics, and particularly in non-invasive wildlife sampling applications. Functions explicitly incorporate missing data, and can tolerate allele mismatches created by genotyping error. If you use this tool, please cite the package using the journal article in Molecular Ecology Resources (Galpern et al., 2012). Please use citation(‘allelematch’) to call the full citation. For users with access to the associated journal article, tutorial material is also available as supplementary material to the article describing this software, the citation for which can be called using citation(‘allelematch’).
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
CoordinateCleaner
</td>
<td style="text-align:left;">
Automated Cleaning of Occurrence Records from Biological Collections
</td>
<td style="text-align:left;">
Automated flagging of common spatial and temporal errors in biological and paleontological collection data, for the use in conservation, ecology and paleontology. Includes automated tests to easily flag (and exclude) records assigned to country or province centroid, the open ocean, the headquarters of the Global Biodiversity Information Facility, urban areas or the location of biodiversity institutions (museums, zoos, botanical gardens, universities). Furthermore identifies per species outlier coordinates, zero coordinates, identical latitude/longitude and invalid coordinates. Also implements an algorithm to identify data sets with a significant proportion of rounded coordinates. Especially suited for large data sets. The reference for the methodology is: Zizka et al. (2019) <a href="doi:10.1111/2041-210X.13152" class="uri">doi:10.1111/2041-210X.13152</a>.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeR
</td>
<td style="text-align:left;">
SNP Genotype Marker Design and Analysis
</td>
<td style="text-align:left;">
We implement a common genotyping workflow with a standardized software interface. ‘genotypeR’ designs genotyping markers from vcf files, outputs markers for multiplexing suitability on various platforms (Sequenom and Illumina GoldenGate), and provides various QA/QC and analysis functions. We developed this package to analyze data in Stevison LS, SA Sefick, CA Rushton, and RM Graze. 2017. Invited Review: Recombination rate plasticity: revealing mechanisms by design. Philosophical Transactions Royal Society London B Biol Sci 372:1-14. &lt;<a href="DOI:10.1098/rstb.2016.0459%3E" class="uri">DOI:10.1098/rstb.2016.0459&gt;</a>;, and have published it here Sefick, S.A., M.A. Castronova, and L.S. Stevison. 2017. GENOTYPER: An integrated R packages for single nuleotide polymorphism genotype marker design and data analysis. Methods in Ecology and Evolution 9: 1318-1323. &lt;DOI: 10.1111/2041-210X.12965&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
m2b
</td>
<td style="text-align:left;">
Movement to Behaviour Inference using Random Forest
</td>
<td style="text-align:left;">
Prediction of behaviour from movement characteristics using observation and random forest for the analyses of movement data in ecology. From movement information (speed, bearing…) the model predicts the observed behaviour (movement, foraging…) using random forest. The model can then extrapolate behavioural information to movement data without direct observation of behaviours. The specificity of this method relies on the derivation of multiple predictor variables from the movement data over a range of temporal windows. This procedure allows to capture as much information as possible on the changes and variations of movement and ensures the use of the random forest algorithm to its best capacity. The method is very generic, applicable to any set of data providing movement data together with observation of behaviour.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
NLMR
</td>
<td style="text-align:left;">
Simulating Neutral Landscape Models
</td>
<td style="text-align:left;">
Provides neutral landscape models (&lt;<a href="doi:10.1007/BF02275262%3E" class="uri">doi:10.1007/BF02275262&gt;</a>;, &lt;<a href="http://sci-hub.tw/10.1007/bf02275262%3E" class="uri">http://sci-hub.tw/10.1007/bf02275262&gt;</a>;).<br />
Neutral landscape models range from “hard” neutral models (completely random distributed), to “soft” neutral models (definable spatial characteristics) and generate landscape patterns that are independent of ecological processes. Thus, these patterns can be used as null models in landscape ecology. ‘nlmr’ combines a large number of algorithms from other published software for simulating neutral landscapes. The simulation results are obtained in a geospatial data format (raster* objects from the ‘raster’ package) and can, therefore, be used in any sort of raster data operation that is performed with standard observation data.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nsRFA
</td>
<td style="text-align:left;">
Non-Supervised Regional Frequency Analysis
</td>
<td style="text-align:left;">
A collection of statistical tools for objective (non-supervised) applications of the Regional Frequency Analysis methods in hydrology. The package refers to the index-value method and, more precisely, helps the hydrologist to: (1) regionalize the index-value; (2) form homogeneous regions with similar growth curves; (3) fit distribution functions to the empirical regional growth curves. Most of the methods are those described in the Flood Estimation Handbook (Centre for Ecology &amp; Hydrology, 1999, <a href="ISBN:9781906698003" class="uri">ISBN:9781906698003</a>). Homogeneity tests from Hosking and Wallis (1993) &lt;<a href="doi:10.1029/92WR01980%3E" class="uri">doi:10.1029/92WR01980&gt;</a>; and Viglione et al. (2007) &lt;<a href="doi:10.1029/2006WR005095%3E" class="uri">doi:10.1029/2006WR005095&gt;</a>; are available.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
RRPP
</td>
<td style="text-align:left;">
Linear Model Evaluation with Randomized Residuals in a Permutation Procedure
</td>
<td style="text-align:left;">
Linear model calculations are made for many random versions of data.<br />
Using residual randomization in a permutation procedure, sums of squares are calculated over many permutations to generate empirical probability distributions for evaluating model effects. This packaged is described by Collyer &amp; Adams (2018) &lt;<a href="doi:10.1111/2041-210X.13029%3E" class="uri">doi:10.1111/2041-210X.13029&gt;</a>;. Additionally, coefficients, statistics, fitted values, and residuals generated over many permutations can be used for various procedures including pairwise tests, prediction, classification, and model comparison. This package should provide most tools one could need for the analysis of high-dimensional data, especially in ecology and evolutionary biology, but certainly other fields, as well.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
StAMPP
</td>
<td style="text-align:left;">
Statistical Analysis of Mixed Ploidy Populations
</td>
<td style="text-align:left;">
Allows users to calculate pairwise Nei’s Genetic Distances (Nei 1972), pairwise Fixation Indexes (Fst) (Weir &amp; Cockerham 1984) and also Genomic Relationship matrixes following Yang et al. (2010) in mixed and single ploidy populations. Bootstrapping across loci is implemented during Fst calculation to generate confidence intervals and p-values around pairwise Fst values. StAMPP utilises SNP genotype data of any ploidy level (with the ability to handle missing data) and is coded to<br />
utilise multithreading where available to allow efficient analysis of large datasets. StAMPP is able to handle genotype data from genlight objects allowing integration with other packages such adegenet. Please refer to LW Pembleton, NOI Cogan &amp; JW Forster, 2013, Molecular Ecology Resources, 13(5), 946-952. &lt;<a href="doi:10.1111/1755-0998.12129%3E" class="uri">doi:10.1111/1755-0998.12129&gt;</a>; for the appropriate citation and user manual. Thank you in advance.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
TITAN2
</td>
<td style="text-align:left;">
Threshold Indicator Taxa Analysis
</td>
<td style="text-align:left;">
Uses indicator species scores across binary partitions of a sample set to detect congruence in taxon-specific changes of abundance and occurrence frequency along an environmental gradient as evidence of an ecological community threshold. Relevant references include: Baker, ME and RS King. 2010. A new method for detecting and interpreting biodiversity and ecological community thresholds. Methods in Ecology and Evolution 1(1): 25:37. King, RS and ME Baker. 2010. Considerations for identifying and interpreting ecological community thresholds. Journal of the North American Benthological Association 29(3):998-1008. Baker ME and RS King. 2013. Of TITAN and straw men: an appeal for greater understanding of community data. Freshwater Science 32(2):489-506.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
eRTG3D
</td>
<td style="text-align:left;">
Empirically Informed Random Trajectory Generation in 3-D
</td>
<td style="text-align:left;">
Creates realistic random trajectories in a 3-D space between two given fix points, so-called conditional empirical random walks (CERWs). The trajectory generation is based on empirical distribution functions extracted from observed trajectories (training data) and thus reflects the geometrical movement characteristics of the mover. A digital elevation model (DEM), representing the Earth’s surface, and a background layer of probabilities (e.g. food sources, uplift potential, waterbodies, etc.) can be used to influence the trajectories. Unterfinger M (2018). “3-D Trajectory Simulation in Movement Ecology: Conditional Empirical Random Walk”. Master’s thesis, University of Zurich. &lt;<a href="https://www.geo.uzh.ch/dam/jcr:6194e41e-055c-4635-9807-53c5a54a3be7/MasterThesis_Unterfinger_2018.pdf%3E" class="uri">https://www.geo.uzh.ch/dam/jcr:6194e41e-055c-4635-9807-53c5a54a3be7/MasterThesis_Unterfinger_2018.pdf&gt;</a>;. Technitis G, Weibel R, Kranstauber B, Safi K (2016). “An algorithm for empirically informed random trajectory generation between two endpoints”. GIScience 2016: Ninth International Conference on Geographic Information Science, 9, online. &lt;<a href="doi:10.5167/uzh-130652%3E" class="uri">doi:10.5167/uzh-130652&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
foieGras
</td>
<td style="text-align:left;">
Fit Continuous-Time State-Space and Latent Variable Models for Quality Control of Argos Satellite (and Other) Telemetry Data and for Estimating Movement Behaviour
</td>
<td style="text-align:left;">
Fits continuous-time random walk and correlated random walk state-space models for quality control animal tracking data (‘Argos’, processed light-level ‘geolocation’, ‘GPS’). Template Model Builder (‘TMB’) is used for fast estimation. The ‘Argos’ data can be: (older) least squares-based locations; (newer) Kalman filter-based locations with error ellipse information; or a mixture of both. The models estimate two sets of location states corresponding to: 1) each observation, which are (usually) irregularly timed; and 2) user-specified time intervals (regular or irregular). Latent variable models are provided to estimate move persistence along tracks as an index of behaviour. ‘Jonsen I’, ‘McMahon CR’, ‘Patterson TA’, ‘Auger-Méthé M’, ‘Harcourt R’, ‘Hindell MA’, ‘Bestley S’ (2019) Movement responses to environment: fast inference of variation among southern elephant seals with a mixed effects model. Ecology 100:e02566 &lt;<a href="doi:10.1002/ecy.2566%3E" class="uri">doi:10.1002/ecy.2566&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
lconnect
</td>
<td style="text-align:left;">
Simple Tools to Compute Landscape Connectivity Metrics
</td>
<td style="text-align:left;">
Provides functions to upload vectorial data and derive landscape connectivity metrics in habitat or matrix systems. Additionally, includes an approach to assess individual patch contribution to the overall landscape connectivity, enabling the prioritization of habitat patches. The computation of landscape connectivity and patch importance are very useful in Landscape Ecology research. The metrics available are: number of components, number of links, size of the largest component, mean size of components, class coincidence probability, landscape coincidence probability, characteristic path length, expected cluster size, area-weighted flux and integral index of connectivity. Pascual-Hortal, L., and Saura, S. (2006) &lt;<a href="doi:10.1007/s10980-006-0013-z%3E" class="uri">doi:10.1007/s10980-006-0013-z&gt;</a>; Urban, D., and Keitt, T. (2001) &lt;<a href="doi:10.2307/2679983%3E" class="uri">doi:10.2307/2679983&gt;</a>; Laita, A., Kotiaho, J., Monkkonen, M. (2011) &lt;<a href="doi:10.1007/s10980-011-9620-4%3E" class="uri">doi:10.1007/s10980-011-9620-4&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nicheROVER
</td>
<td style="text-align:left;">
(Niche) (R)egion and Niche (Over)lap Metrics for Multidimensional Ecological Niches
</td>
<td style="text-align:left;">
This package uses a probabilistic method to calculate niche regions and pairwise niche overlap using multidimensional niche indicator data (e.g., stable isotopes, environmental variables, etc.). The niche region is defined as the joint probability density function of the multidimensional niche indicators at a user-defined probability alpha (e.g., 95%). Uncertainty is accounted for in a Bayesian framework, and the method can be extended to three or more indicator dimensions. It provides directional estimates of niche overlap, accounts for species-specific distributions in multivariate niche space, and produces unique and consistent bivariate projections of the multivariate niche region. A forthcoming article by Swanson et al. (Ecology, 2014) provides a detailed description of the methodology. See the package vignette for a worked example using fish stable isotope data.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
nlraa
</td>
<td style="text-align:left;">
Nonlinear Regression for Agricultural Applications
</td>
<td style="text-align:left;">
Additional nonlinear regression functions using self-start (SS) algorithms. One of the functions is the Beta growth function proposed by Yin et al. (2003) &lt;<a href="doi:10.1093/aob/mcg029%3E" class="uri">doi:10.1093/aob/mcg029&gt;</a>;. There are several other functions with breakpoints (e.g. linear-plateau, plateau-linear, exponential-plateau, plateau-exponential, quadratic-plateau, plateau-quadratic and bilinear), a non-rectangular hyperbola and a bell-shaped curve. Twenty one (21) new self-start (SS) functions in total. This package also supports the publication ‘Nonlinear regression Models and applications in agricultural research’ by Archontoulis and Miguez (2015) &lt;<a href="doi:10.2134/agronj2012.0506%3E" class="uri">doi:10.2134/agronj2012.0506&gt;</a>;, a book chapter with similar material &lt;<a href="doi:10.2134/appliedstatistics.2016.0003.c15%3E" class="uri">doi:10.2134/appliedstatistics.2016.0003.c15&gt;</a>; and a publication by Oddi et. al. (2019) in Ecology and Evolution &lt;<a href="doi:10.1002/ece3.5543%3E" class="uri">doi:10.1002/ece3.5543&gt;</a>;. The function ‘nlsLMList’ uses nlsLM for fitting, but it is otherwise almost identical to ‘nlme::nlsList’.In addition, this release of the package provides functions for conducting simulation for ‘nlme’ and ‘gnls’ objects as well as bootstrapping. These functions are intended to work with the modeling framework of the ‘nlme’ package. It also provides four vignettes with extended examples.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
Rdistance
</td>
<td style="text-align:left;">
Distance-Sampling Analyses for Density and Abundance Estimation
</td>
<td style="text-align:left;">
Distance-sampling is a popular method for estimating density and abundance of organisms in ecology. Rdistance contains routines that<br />
assist with analysis of distance-sampling data collected on point or line transects. Distance models are specified using regression-like formula (similar to lm, glm, etc.). Abundance routines perform automated bootstrapping and automated detection-function selection. Overall (study area) and site-level (transect or point) abundance estimates are available. A large suite of classical, parametric detection functions are included along with some uncommon parametric functions (e.g., Gamma, negative exponential) and non-parametric smoothed distance functions. Custom (user-defined) detection functions are easily implemented (see vignette). The help files and vignettes have been vetted by multiple authors and tested in workshop settings.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
Select
</td>
<td style="text-align:left;">
Determines Species Probabilities Based on Functional Traits
</td>
<td style="text-align:left;">
The objective of these functions is to derive a species assemblage that satisfies a functional trait profile. Restoring resilient ecosystems requires a flexible framework for selecting assemblages that are based on the functional traits of species. However, current trait-based models have been limited to algorithms that can only select species by optimising specific trait values, and could not elegantly accommodate the common desire among restoration ecologists to produce functionally diverse assemblages. We have solved this problem by applying a non-linear optimisation algorithm that optimises Rao Q, a closed-form functional trait diversity index that incorporates species abundances, subject to other linear constraints. This framework generalises previous models that only optimised the entropy of the community, and can optimise both functional diversity and entropy simultaneously. This package can also be used to generate experimental assemblages to test the effects of community-level traits on community dynamics and ecosystem function. The method is based on theory discussed in Laughlin (2014, Ecology Letters) &lt;doi.org/10.1111/ele.12288&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
TipDatingBeast
</td>
<td style="text-align:left;">
Using Tip Dates with Phylogenetic Trees in BEAST
</td>
<td style="text-align:left;">
Assists performing tip-dating of phylogenetic trees with BEAST BEAST is a popular software for phylogenetic analysis. The package assists the implementation of various phylogenetic tip- dating tests using BEAST. It contains two main functions. The first one allows preparing date randomization analyses, which assess the temporal signal of a data set. The second function allows performing leave-one-out analyses, which test for the consistency between independent calibration sequences and allow pinpointing those leading to potential bias. The included tutorial provides detailed step-by-step instructions. An expanded description of the package can be found in article: Rieux, A. and Khatchikian, C.E. (2017), TIPDATINGBEAST: an R package to assist the implementation of phylogenetic tip-dating tests using BEAST. Molecular Ecology Resources, 17: 608-613. &lt;onlinelibrary.wiley.com/doi/full/10.1111/1755-0998.12603&gt;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
VGAM
</td>
<td style="text-align:left;">
Vector Generalized Linear and Additive Models
</td>
<td style="text-align:left;">
An implementation of about 6 major classes of statistical regression models. The central algorithm is Fisher scoring and iterative reweighted least squares. At the heart of this package are the vector generalized linear and additive model (VGLM/VGAM) classes. VGLMs can be loosely thought of as multivariate GLMs. VGAMs are data-driven VGLMs that use smoothing. The book “Vector Generalized Linear and Additive Models: With an Implementation in R” (Yee, 2015) &lt;<a href="DOI:10.1007/978-1-4939-2818-7%3E" class="uri">DOI:10.1007/978-1-4939-2818-7&gt;</a>; gives details of the statistical framework and the package. Currently only fixed-effects models are implemented. Many (150+) models and distributions are estimated by maximum likelihood estimation (MLE) or penalized MLE. The other classes are RR-VGLMs (reduced-rank VGLMs), quadratic RR-VGLMs, reduced-rank VGAMs, RCIMs (row-column interaction models)—these classes perform constrained and unconstrained quadratic ordination (CQO/UQO) models in ecology, as well as constrained additive ordination (CAO). Note that these functions are subject to change; see the NEWS and ChangeLog files for latest changes.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
algaeClassify
</td>
<td style="text-align:left;">
Determine Phytoplankton Functional Groups Based on Functional Traits
</td>
<td style="text-align:left;">
Verify accepted taxonomic nomenclature of phytoplankton species, assign species to functional group classifications, and manipulate taxonomic and functional diversity data. Possible functional classifications include Morpho-functional group (MFG; Salmaso et al. 2015 &lt;<a href="doi:10.1111/fwb.12520%3E" class="uri">doi:10.1111/fwb.12520&gt;</a>;) and CSR (Reynolds 1988; Functional morphology and the adaptive strategies of phytoplankton. In C.D. Sandgren (ed). Growth and reproductive strategies of freshwater phytoplankton, 388-433. Cambridge University Press, New York). Version 1.2.0 also includes functions to query names using the algaebase online taxonomic database (&lt;<a href="https://www.algaebase.org%3E" class="uri">https://www.algaebase.org&gt;</a>;; &lt;<a href="doi:10.7872/crya.v35.iss2.2014.105%3E" class="uri">doi:10.7872/crya.v35.iss2.2014.105&gt;</a>;). The algaeClassify package is a product of the GEISHA (Global Evaluation of the Impacts of Storms on freshwater Habitat and Structure of phytoplankton Assemblages), funded by CESAB (Centre for Synthesis and Analysis of Biodiversity) and the USGS John Wesley Powell Center for Synthesis and Analysis, with data and other support provided by members of GLEON (Global Lake Ecology Observation Network). This software is preliminary or provisional and is subject to revision. It is being provided to meet the need for timely best science. The software has not received final approval by the U.S. Geological Survey (USGS). No warranty, expressed or implied, is made by the USGS or the U.S. Government as to the functionality of the software and related material nor shall the fact of release constitute any such warranty. The software is provided on the condition that neither the USGS nor the U.S. Government shall be held liable for any damages resulting from the authorized or unauthorized use of the software.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
hiphop
</td>
<td style="text-align:left;">
Parentage Assignment using Bi-Allelic Genetic Markers
</td>
<td style="text-align:left;">
Can be used for paternity and maternity assignment and outperforms conventional methods where closely related individuals occur in the pool of possible parents. The method compares the genotypes of offspring with any combination of potentials parents and scores the number of mismatches of these individuals at bi-allelic genetic markers (e.g. Single Nucleotide Polymorphisms). It elaborates on a prior exclusion method based on the Homozygous Opposite Test (HOT; Huisman 2017 &lt;<a href="doi:10.1111/1755-0998.12665%3E" class="uri">doi:10.1111/1755-0998.12665&gt;</a>;) by introducing the additional exclusion criterion HIPHOP (Homozygous Identical Parents, Heterozygous Offspring are Precluded; Cockburn et al., in revision). Potential parents are excluded if they have more mismatches than can be expected due to genotyping error and mutation, and thereby one can identify the true genetic parents and detect situations where one (or both) of the true parents is not sampled. Package ‘hiphop’ can deal with (a) the case where there is contextual information about parentage of the mother (i.e. a female has been seen to be involved in reproductive tasks such as nest building), but paternity is unknown (e.g. due to promiscuity), (b) where both parents need to be assigned, because there is no contextual information on which female laid eggs and which male fertilized them (e.g. polygynandrous mating system where multiple females and males deposit young in a common nest, or organisms with external fertilisation that breed in aggregations). For details: Cockburn, A., Penalba, J.V.,Jaccoud, D.,Kilian, A., Brouwer, L., Double, M.C., Margraf, N., Osmond, H.L., van de Pol, M. and Kruuk, L.E.B. (in revision). HIPHOP: improved paternity assignment among close relatives using a simple exclusion method for bi-allelic markers. Molecular Ecology Resources, DOI to be added upon acceptance.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
NST
</td>
<td style="text-align:left;">
Normalized Stochasticity Ratio
</td>
<td style="text-align:left;">
To estimate ecological stochasticity in community assembly. Understanding the community assembly mechanisms controlling biodiversity patterns is a central issue in ecology. Although it is generally accepted that both deterministic and stochastic processes play important roles in community assembly, quantifying their relative importance is challenging. The new index, normalized stochasticity ratio (NST), is to estimate ecological stochasticity, i.e. relative importance of stochastic processes, in community assembly. With functions in this package, NST can be calculated based on different similarity metrics and/or different null model algorithms, as well as some previous indexes, e.g. previous Stochasticity Ratio (ST), Standard Effect Size (SES), modified Raup-Crick metrics (RC). Functions for permutational test and bootstrapping analysis are also included. Previous ST is published by Zhou et al (2014) &lt;<a href="doi:10.1073/pnas.1324044111%3E" class="uri">doi:10.1073/pnas.1324044111&gt;</a>;. NST is modified from ST by considering two alternative situations and normalizing the index to range from 0 to 1 (Ning et al 2019) &lt;<a href="doi:10.1073/pnas.1904623116%3E" class="uri">doi:10.1073/pnas.1904623116&gt;</a>;. A modified version, MST, is a special case of NST, used in some recent or upcoming publications, e.g. Liang et al (2019) &lt;<a href="doi:10.1101/638908%3E" class="uri">doi:10.1101/638908&gt;</a>;. SES is calculated as described in Kraft et al (2011) &lt;<a href="doi:10.1126/science.1208584%3E" class="uri">doi:10.1126/science.1208584&gt;</a>;. RC is calculated as reported by Chase et al (2011) &lt;<a href="doi:10.1890/es10-00117.1%3E" class="uri">doi:10.1890/es10-00117.1&gt;</a>; and Stegen et al (2013) &lt;<a href="doi:10.1038/ismej.2013.93%3E" class="uri">doi:10.1038/ismej.2013.93&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
Rbeast
</td>
<td style="text-align:left;">
Bayesian Change-Point Detection and Time Series Decomposition
</td>
<td style="text-align:left;">
Interpretation of time series data is affected by model choices. Different models can give different or even contradicting estimates of patterns, trends, and mechanisms for the same data–a limitation alleviated by the Bayesian estimator of abrupt change,seasonality, and trend (BEAST) of this package. BEAST seeks to improve time series decomposition by forgoing the “single-best-model” concept and embracing all competing models into the inference via a Bayesian model averaging scheme. It is a flexible tool to uncover abrupt changes (i.e., change-points), cyclic variations (e.g., seasonality), and nonlinear trends in time-series observations. BEAST not just tells when changes occur but also quantifies how likely the detected changes are true. It detects not just piecewise linear trends but also arbitrary nonlinear trends. BEAST is applicable to real-valued time series data of all kinds, be it for remote sensing, economics, climate sciences, ecology, and hydrology. Example applications include its use to identify regime shifts in ecological data, map forest disturbance and land degradation from satellite imagery, detect market trends in economic data, pinpoint anomaly and extreme events in climate data, and unravel system dynamics in biological data. Details on BEAST are reported in Zhao et al. (2019) &lt;<a href="doi:10.1016/j.rse.2019.04.034%3E" class="uri">doi:10.1016/j.rse.2019.04.034&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
SRS
</td>
<td style="text-align:left;">
Scaling with Ranked Subsampling
</td>
<td style="text-align:left;">
Analysis of species count data in ecology often requires normalization to an identical sample size. Rarefying (random subsampling without replacement), which is a popular method for normalization, has been widely criticized for its poor reproducibility and potential distortion of the community structure. In the context of microbiome count data, researchers explicitly advised against the use of rarefying. An alternative to rarefying is scaling with ranked subsampling (SRS). SRS consists of two steps. In the first step, the total counts for all OTUs (operational taxonomic units) or species in each sample are divided by a scaling factor chosen in such a way that the sum of the scaled counts Cscaled equals Cmin. In the second step, the non-integer Cscaled values are converted into integers by an algorithm that we dub ranked subsampling. The Cscaled value for each OTU or species is split into the integer part Cint (Cint = floor(Cscaled)) and the fractional part Cfrac (Cfrac = Cscaled - Cints). Since the sum of Cint is smaller or equal to Cmin, additional delta C = Cmin - the sum of Cint counts have to be added to the library to reach the total count of Cmin. This is achieved as follows. OTUs are ranked in the descending order of their Cfrac values. Beginning with the OTU of the highest rank, single count per OTU is added to the normalized library until the total number of added counts reaches delta C and the sum of all counts in the normalized library equals Cmin. When the lowest Cfrag involved in picking delta C counts is shared by several OTUs, the OTUs used for adding a single count to the library are selected in the order of their Cint values. This selection minimizes the effect of normalization on the relative frequencies of OTUs. OTUs with identical Cfrag as well as Cint are sampled randomly without replacement. See Beule &amp; Karlovsky (2020) &lt;<a href="doi:10.7717/peerj.9593%3E" class="uri">doi:10.7717/peerj.9593&gt;</a>; for details.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
TreeSimGM
</td>
<td style="text-align:left;">
Simulating Phylogenetic Trees under General Bellman Harris and Lineage Shift Model
</td>
<td style="text-align:left;">
Provides a flexible simulation tool for phylogenetic trees under a general model for speciation and extinction. Trees with a user-specified number of extant tips, or a user-specified stem age are simulated. It is possible to assume any probability distribution for the waiting time until speciation and extinction. Furthermore, the waiting times to speciation / extinction may be scaled in different parts of the tree, meaning we can simulate trees with clade-dependent diversification processes. At a speciation event, one species splits into two. We allow for two different modes at these splits: (i) symmetric, where for every speciation event new waiting times until speciation and extinction are drawn for both daughter lineages; and (ii) asymmetric, where a speciation event results in one species with new waiting times, and another that carries the extinction time and age of its ancestor. The symmetric mode can be seen as an vicariant or allopatric process where divided populations suffer equal evolutionary forces while the asymmetric mode could be seen as a peripatric speciation where a mother lineage continues to exist. Reference: O. Hagen and T. Stadler (2017). TreeSimGM: Simulating phylogenetic trees under general Bellman Harris models with lineage-specific shifts of speciation and extinction in R. Methods in Ecology and Evolution. &lt;<a href="doi:10.1111/2041-210X.12917%3E" class="uri">doi:10.1111/2041-210X.12917&gt;</a>;.
</td>
<td style="text-align:left;">
packagefinder
</td>
</tr>
<tr>
<td style="text-align:left;">
MetaboAnalystR
</td>
<td style="text-align:left;">
An R Package for Comprehensive Analysis of Metabolomics Data
</td>
<td style="text-align:left;">
<p>MetaboAnalystR 3.0 contains the R functions and libraries underlying the popular MetaboAnalyst web server, including metabolomic data analysis, visualization, and functional interpretation. The package is synchronized with the MetaboAnalyst web server. After installing and loading the package, users will be able to reproduce the same results from their local computers using the corresponding R command history downloaded from MetaboAnalyst web site, thereby achieving maximum flexibility and reproducibility.</p>
The version 3.0 aims to improve the current global metabolomics workflow by implementing a fast parameter optimization algorithm for peak picking, and automated identification of the most suitable method for batch effect correction from 12 well-established approaches. In addition, more support for functional interpretation directly from m/z peaks via mummichog2 (PMID: 23861661), and a new pathway-based method to integrate multi’omics data has been added. To demonstrate this new functionality, we provide the “MetaboAnalystR 3.0 Workflow: Towards an Optimized Workflow for Global Metabolomics” vignette, available here as a PDF. In this vignette, we perform end-to-end metabolomics data analysis on the full batch of clinical IBD samples.
</td>
<td style="text-align:left;">
volePhD
</td>
</tr>
<tr>
<td style="text-align:left;">
camtrapR
</td>
<td style="text-align:left;">
Camera Trap Data Management and Preparation of Occupancy and Spatial Capture-Recapture Analyses
</td>
<td style="text-align:left;">
See the Article in Methods in Ecology and Evolution &lt;<a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12600%3E" class="uri">https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12600&gt;</a>; for an overview of the package. The four package vignettes provide examples for the entire workflow.
</td>
<td style="text-align:left;">
volePhD
</td>
</tr>
<tr>
<td style="text-align:left;">
animaltracker
</td>
<td style="text-align:left;">
Spatial-temporal analysis and visualisation of animal movement data
</td>
<td style="text-align:left;">
Utilities for spatial-temporal analysis and visualization of animal (e.g. cattle) tracking data. The core feature is a ‘shiny’ web application for customized processing of GPS logs, including features for data augmentation (e.g. elevation lookup), data selection, export, plotting, and statistical summaries. A data validation application allows for side-by-side comparison via time series plots and extreme value detection described by J.P. van Brakel &lt;<a href="https://stackoverflow.com/questions/22583391/peak-signal-detection-inrealtime-timeseries-data/%3E" class="uri">https://stackoverflow.com/questions/22583391/peak-signal-detection-inrealtime-timeseries-data/&gt;</a>;.
</td>
<td style="text-align:left;">
volePhD
</td>
</tr>
<tr>
<td style="text-align:left;">
animaltracker
</td>
<td style="text-align:left;">
Spatial-temporal analysis and visualisation of animal movement data
</td>
<td style="text-align:left;">
Utilities for spatial-temporal analysis and visualization of animal (e.g. cattle) tracking data. The core feature is a ‘shiny’ web application for customized processing of GPS logs, including features for data augmentation (e.g. elevation lookup), data selection, export, plotting, and statistical summaries. A data validation application allows for side-by-side comparison via time series plots and extreme value detection described by J.P. van Brakel &lt;<a href="https://stackoverflow.com/questions/22583391/peak-signal-detection-inrealtime-timeseries-data/%3E" class="uri">https://stackoverflow.com/questions/22583391/peak-signal-detection-inrealtime-timeseries-data/&gt;</a>;.
</td>
<td style="text-align:left;">
volePhD
</td>
</tr>
<tr>
<td style="text-align:left;">
dismo
</td>
<td style="text-align:left;">
Species distribution modelling
</td>
<td style="text-align:left;">
Functions for species distribution modeling, that is, predicting entire geographic distributions form occurrences at a number of sites and the environment at these sites
</td>
<td style="text-align:left;">
StatsForBios
</td>
</tr>
<tr>
<td style="text-align:left;">
wallace
</td>
<td style="text-align:left;">
Wallace is a modular platform for reproducible modeling of species niches and distributions, written in R
</td>
<td style="text-align:left;">
Wallace is a modular platform for reproducible modeling of species niches and distributions, written in R. The application guides users through a complete analysis, from the acquisition of data to visualizing model predictions on an interactive map, thus bundling complex workflows into a single, streamlined interface.
</td>
<td style="text-align:left;">
StatsForBios
</td>
</tr>
<tr>
<td style="text-align:left;">
spocc
</td>
<td style="text-align:left;">
A programmatic interface to many species occurrence data sources
</td>
<td style="text-align:left;">
<p>At rOpenSci, we have been writing R packages to interact with many sources of species occurrence data, including GBIF, Vertnet, BISON, iNaturalist, the Berkeley ecoengine, and eBird. Other databases are out there as well, which we can pull in. spocc is an R package to query and collect species occurrence data from many sources. The goal is to to create a seamless search experience across data sources, as well as creating unified outputs across data sources.</p>
<p>spocc currently interfaces with nine major biodiversity repositories</p>
<p>Global Biodiversity Information Facility (GBIF) (via rgbif) GBIF is a government funded open data repository with several partner organizations with the express goal of providing access to data on Earth’s biodiversity. The data are made available by a network of member nodes, coordinating information from various participant organizations and government agencies.</p>
<p>Berkeley Ecoengine (via ecoengine) The ecoengine is an open API built by the Berkeley Initiative for Global Change Biology. The repository provides access to over 3 million specimens from various Berkeley natural history museums. These data span more than a century and provide access to georeferenced specimens, species checklists, photographs, vegetation surveys and resurveys and a variety of measurements from environmental sensors located at reserves across University of California’s natural reserve system.</p>
<p>iNaturalist iNaturalist provides access to crowd sourced citizen science data on species observations.</p>
<p>VertNet (via rvertnet) Similar to rgbif, ecoengine, and rbison (see below), VertNet provides access to more than 80 million vertebrate records spanning a large number of institutions and museums primarly covering four major disciplines (mammology, herpetology, ornithology, and icthyology).</p>
<p>Biodiversity Information Serving Our Nation (via rbison) Built by the US Geological Survey’s core science analytic team, BISON is a portal that provides access to species occurrence data from several participating institutions.</p>
<p>eBird (via rebird) ebird is a database developed and maintained by the Cornell Lab of Ornithology and the National Audubon Society. It provides real-time access to checklist data, data on bird abundance and distribution, and communtiy reports from birders.</p>
<p>iDigBio (via ridigbio) iDigBio facilitates the digitization of biological and paleobiological specimens and their associated data, and houses specimen data, as well as providing their specimen data via RESTful web services.</p>
<p>OBIS OBIS (Ocean Biogeographic Information System) allows users to search marine species datasets from all of the world’s oceans.</p>
<p>Atlas of Living Australia ALA (Atlas of Living Australia) contains information on all the known species in Australia aggregated from a wide range of data providers: museums, herbaria, community groups, government departments, individuals and universities; it contains more than 50 million occurrence records.</p>
<p>The inspiration for this comes from users requesting a more seamless experience across data sources, and from our work on a similar package for taxonomy data (taxize).</p>
BEWARE: In cases where you request data from multiple providers, especially when including GBIF, there could be duplicate records since many providers’ data eventually ends up with GBIF. See ?spocc_duplicates, after installation, for more.
</td>
<td style="text-align:left;">
StatsForBios
</td>
</tr>
</tbody>
</table>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
