<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>List of R packages for Ecology</title>
  
  <meta property="description" itemprop="description" content="Here is a list (clearly incomplete) of R packages that are focused on Ecology!"/>
  
  
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="List of R packages for Ecology"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Here is a list (clearly incomplete) of R packages that are focused on Ecology!"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="List of R packages for Ecology"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="List of R packages for Ecology"/>
  <meta property="twitter:description" content="Here is a list (clearly incomplete) of R packages that are focused on Ecology!"/>
  
  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","site"]}},"value":[{"type":"character","attributes":{},"value":["List of R packages for Ecology"]},{"type":"character","attributes":{},"value":["Here is a list (clearly incomplete) of R packages that are focused on Ecology!\n"]},{"type":"character","attributes":{},"value":["distill::distill_website"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }
  
  </style>
  
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });
  
  });
  
  </script>
  
  <style type="text/css">
  
  .nav-search {
    font-size: x-small;
  }
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }
  
  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }
  
  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  /* Citation hover box */
  
  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }
  
  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  /* Include appendix styles here so they can be overridden */
  
  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }
  
  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }
  
  d-appendix h3 + * {
    margin-top: 1em;
  }
  
  d-appendix ol {
    padding: 0 0 0 15px;
  }
  
  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }
  
  d-appendix li {
    margin-bottom: 1em;
  }
  
  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  d-appendix > * {
    grid-column: text;
  }
  
  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }
  
  /* Include footnote styles here so they can be overridden */
  
  d-footnote-list {
    contain: layout style;
  }
  
  d-footnote-list > * {
    grid-column: text;
  }
  
  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }
  
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }
  
  /* Styles for posts lists (not auto-injected) */
  
  
  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }
  
  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }
  
  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }
  
  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }
  
  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }
  
  .post-preview-last {
    border-bottom: none !important;
  }
  
  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }
  
  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }
  
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }
  
  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }
  
  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }
  
  .posts-list .metadata > * {
    display: inline-block;
  }
  
  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }
  
  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }
  
  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }
  
  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  .posts-list img {
    opacity: 1;
  }
  
  .posts-list img[data-src] {
    opacity: 0;
  }
  
  .posts-more {
    clear: both;
  }
  
  
  .posts-sidebar {
    font-size: 16px;
  }
  
  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }
  
  .sidebar-section {
    margin-bottom: 30px;
  }
  
  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }
  
  .categories li>a {
    border-bottom: none;
  }
  
  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }
  
  .categories .active {
    font-weight: 600;
  }
  
  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }
  
  
  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  .downlevel .posts-list .post-preview {
    color: inherit;
  }
  
  
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
  <script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.17/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"List of R packages for Ecology","description":"Here is a list (clearly incomplete) of R packages that are focused on Ecology!","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">List of R packages for Ecology</a>
</div>
<div class="nav-right">
<a href="index.html">Home</a>
<a href="about.html">About</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>List of R packages for Ecology</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Here is a list (clearly incomplete) of R packages that are focused on Ecology!</p></p>
</div>


<div class="d-article">
<p>Feel free to add to the list by sending me a pull request or adding an issue (<a href="https://github.com/DrMattG/REcol-verse" class="uri">https://github.com/DrMattG/REcol-verse</a>)</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="r-packages">R packages</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><em>We currently have 452 packages listed.</em></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-c9e614bd8803d50cb498" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c9e614bd8803d50cb498">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452"],["soundecology","nimbleEcology","RevEcoR","vegan","CommEcol","ecospat","BayesComm","picante","dave","letsR","forams","labdsv","phyloregion","RAM","analogue","SSP","BiodiversityR","mefa","tsvr","meteR","swapClass","primer","CNull","rangeMapper","BIEN","wsyn","DSAIDE","AHMbook","MCMC.OTU","FD","blmeco","loop","paleoMAS","cepreader","hydrostats","inpdfr","FLightR","treeclim","DAISIE","expp","betaper","enveomics.R","sharx","move","petitr","portalr","fso","polyfreqs","virtualspecies","RADanalysis","SitesInterest","bcpa","ecospace","mvabund","raceland","rfishbase","sampSurf","SiMRiv","bmotif","EMbC","NitrogenUptake2016","nna","VGAMdata","EcoSimR","gfcanalysis","OCNet","philentropy","pop","QFASA","SADISA","virtualNicheR","aPCoA","boral","ENMTools","nodiv","pgirmess","seas","SP2000","stockR","zoon","allelematch","CoordinateCleaner","genotypeR","m2b","NLMR","nsRFA","RRPP","StAMPP","TITAN2","eRTG3D","foieGras","lconnect","nicheROVER","nlraa","Rdistance","Select","TipDatingBeast","VGAM","algaeClassify","hiphop","NST","Rbeast","SRS","TreeSimGM","MetaboAnalystR","camtrapR","animaltracker","dismo","wallace","spocc","marked","unmarked","Rmark","ctmm","moveHMM","adehabitat","landscapemetrics","amt","adehabitatHR","momentuHMM","recurse","MLWIC2","ade4","rdacca.hp","abdiv","adagio","adehabitatHS","adehabitatLT","adehabitatMA","adiv","AGHmatrix","AICcmodavg","animalTrack","anipaths","apTreeshape","aws.signature","AzureKeyVault","BacArena","BALCONY","BAMMtools","BAT","BBMM","BBMV","BENMMI","BinMat","biogeo","biomod2","bite","blockmatrix","BMTME","botor","BPEC","bsam","Cairo","canprot","carfima","castor","cde","CDROM","celltrackR","cg","CHNOSZ","circlize","Claddis","clinUtiDNA","coala","cocorresp","coexist","colorist","config","CornerstoneR","crawl","ctmcmove","DatabionicSwarm","datamap","DChaos","dendRoAnalyst","densitr","dequer","divDyn","DivE","diverge","doRNG","dprint","dragon","eadrm","eclust","educineq","eesim","emma","emoa","envoutliers","EpiILM","EpiILMCT","eSDM","FinePop2","fishmove","fluoSurv","fluxweb","fossil","FSK2R","fugeR","funbarRF","fuzzySim","gcbd","genpwr","geoSpectral","ggspectra","Ghat","globals","goeveg","gramEvol","gRbase","Haplin","heatex","helminthR","hSDM","hydroPSO","hydroTSM","hypervolume","HyPhy","ICAFF","IDE","IDPmisc","import","IndexNumber","iNEXT","iSDM","isocat","J4R","jointNmix","jrich","jSDM","keyring","kissmig","KnowBR","LaplacesDemon","LCAvarsel","leastcostpath","lira","littler","LSPFP","LSRS","lulcc","macleish","Mapinguari","marcher","MDSGUI","memgene","metaheuristicOpt","MetaLandSim","metaMix","MGMS2","MHCtools","mkde","MMLR","modEvA","MoLE","motmot","mousetrack","mousetrap","movecost","moveVis","moveWindSpeed","MultiBD","multiDimBio","mvMORPH","MVR","myTAI","NetLogoR","NetworkExtinction","nlrx","nmfgpu4R","nmixgof","NobBS","oceanis","ofGEM","opticut","osmose","otsad","ouxy","overlap","pairheatmap","paleomorph","pampe","pedigreeTools","pepr","performance","phenability","photobiologyInOut","photobiologySensors","PhylogeneticEM","phytools","PIGShift","plinkFile","POUMM","PSPManalysis","R.SamBada","r6extended","RAPTOR","raptr","Rarefy","rareGE","rase","rAvis","rCAT","RcppDE","rdwplus","rerddapXtracto","restorepoint","RFreak","rgdax","RGENERATE","rgrass7","ribiosUtils","rmetasim","Rmpi","RNCBIEUtilsLibs","roben","RODM","ropensecretsapi","rotl","rpmodel","rpostgisLT","RRphylo","rscontract","rSHAPE","rsMove","RSuite","rusda","RWildbook","samc","scrm","SDDE","SDEFSR","SDMPlay","sdmpredictors","SDMtune","sdmvspecies","sdStaf","secr","secrdesign","secret","secrettext","secrlinear","selectMeta","sensiPhy","sesem","ShapeRotator","shar","siplab","slouch","smam","SOFIA","SpaDES","spathial","SPECIES","spectralGP","spider","spThin","SSDM","ssdtools","stability","startup","subgxe","SubpathwayGMir","Sunder","svMisc","syntaxr","taxlist","tergm","tigerstats","timetree","TR8","trackdf","traipse","transformr","transmem","treebase","treedater","TreePar","TreeSim","trendchange","trioGxE","trip","usdm","vaultr","vegan3d","vegetarian","VLF","VTrack","wildlifeDI","windex","wrProteo","zetadiv","acmeR","adegenet","adephylo","auk","bbsBayes","bioRad","birdring","bSims","carcass","climwin","Distance","diveMove","dsm","ebirdst","emmeans","flying","FlyingR","GeoLight","geomorph","ggThemeAssist","lme4","MCMCglmm","mgcv","MixSIAR","MODIS","MODISTools","monitoR","moult","mrds","MuMIn","nimbleSCR","openCR","osmdata","patchwork","PresenceAbsence","R2ucare","raster","Rcapture","rebird","ResourceSelection","rgbif","rinat","rjags","rr2","rredlist","rstan","rstanarm","seewave","sf","SongEvo","sp","taxize","tidyverse","tmap","tuneR","visreg","weathercan"],["Soundscape Ecology","Distributions for Ecological Models in 'nimble'","Reverse Ecology Analysis on Microbiome","Community Ecology Package","Community Ecology Analyses","Spatial Ecology Miscellaneous Methods","Bayesian Community Ecology Analysis","Integrating Phylogenies and Ecology","Functions for \"Data Analysis in Vegetation Ecology\"","Data Handling and Analysis in Macroecology","Foraminifera and Community Ecology Analyses","Ordination and Multivariate Analysis for Ecology","Biogeographic Regionalization and Macroecology","R for Amplicon-Sequencing-Based Microbial-Ecology","Analogue and Weighted Averaging Methods for Palaeoecology","Simulated Sampling Procedure for Community Ecology","Package for Community Ecology and Suitability Analysis","Multivariate Data Handling in Ecology and Biogeography","Timescale-Specific Variance Ratio for Use in Community Ecology","Fitting and Plotting Tools for the Maximum Entropy Theory of\nEcology (METE)","A Null Model Adapted to Abundance Class Data in Ecology","Functions and Data for the Book, A Primer of Ecology with R","Fast Algorithms for Frequency-Preserving Null Models in Ecology","A Platform for the Study of Macro-Ecology of Life History Traits","Tools for Accessing the Botanical Information and Ecology\nNetwork Database","Wavelet Approaches to Studies of Synchrony in Ecology and Other\nFields","Dynamical Systems Approach to Infectious Disease Epidemiology\n(Ecology/Evolution)","Functions and Data for the Book \"Applied Hierarchical Modeling\nin Ecology\" Vols 1 and 2","Bayesian Analysis of Multivariate Counts Data in DNA\nMetabarcoding and Ecology","Measuring functional diversity (FD) from multiple traits, and\nother tools for functional ecology","Data Files and Functions Accompanying the Book \"Bayesian Data\nAnalysis in Ecology using R, BUGS and Stan\"","loop decomposition of weighted directed graphs for life cycle\nanalysis, providing flexbile network plotting methods, and\nanalyzing food chain properties in ecology","Paleoecological Analysis","Read 'CEP' and Legacy 'CANOCO' Files","Hydrologic Indices for Daily Time Series Data","Analyse Text Documents Using Ecological Tools","Reconstruct Animal Paths from Solar Geolocation Loggers Data","Numerical Calibration of Proxy-Climate Relationships","Dynamical Assembly of Islands by Speciation, Immigration and\nExtinction","Spatial Analysis of Extra-Pair Paternity","Taxonomic Uncertainty on Multivariate Analyses of Ecological\nData","Various Utilities for Microbial Genomics and Metagenomics","Models and Data Sets for the Study of Species-Area Relationships","Visualizing and Analyzing Animal Track Data","Relative Growth Rate","Create Useful Summaries of the Portal Data","Fuzzy Set Ordination","Bayesian Population Genomics in Autopolyploids","Generation of Virtual Species Distributions","Normalization and Study of Rank Abundance Distributions","Inferring an Animal's Sites of Interest from High Resolution\nData","Behavioral change point analysis of animal movement","Simulating Community Assembly and Ecological Diversification\nUsing Ecospace Frameworks","Statistical Methods for Analysing Multivariate Abundance Data","Pattern-Based Zoneless Method for Analysis and Visualization of\nRacial Topography","R Interface to 'FishBase'","Sampling Surface Simulation for Areal Sampling Methods","Simulating Multistate Movements in River/Heterogeneous\nLandscapes","Motif Analyses of Bipartite Networks","Expectation-Maximization Binary Clustering","Data and Source Code From: Nitrogen Uptake and Allocation\nEstimates for Spartina Alterniflora and Distichlis Spicata","Nearest-Neighbor Analysis","Data Supporting the 'VGAM' Package","Null Model Analysis for Ecological Data","Tools for Working with Hansen et al. Global Forest Change\nDataset","Optimal Channel Networks","Similarity and Distance Quantification Between Probability\nFunctions","A Flexible Syntax for Population Dynamic Modelling","Quantitative Fatty Acid Signature Analysis","Species Abundance Distributions with Independent-Species\nAssumption","Generating Virtual Fundamental and Realised Niches","Covariate Adjusted PCoA Plot","Bayesian Ordination and Regression AnaLysis","Analysis of Niche Evolution using Niche and Distribution Models","Compares the Distribution of Sister Clades Through a Phylogeny","Spatial Analysis and Data Mining for Field Ecologists","Seasonal Analysis and Graphics, Especially for Climatology","Catalogue of Life Toolkit","Identifying Stocks in Genetic Data","Reproducible, Accessible &amp; Shareable Species Distribution\nModelling","Identifying Unique Multilocus Genotypes where Genotyping Error\nand Missing Data may be Present","Automated Cleaning of Occurrence Records from Biological\nCollections","SNP Genotype Marker Design and Analysis","Movement to Behaviour Inference using Random Forest","Simulating Neutral Landscape Models","Non-Supervised Regional Frequency Analysis","Linear Model Evaluation with Randomized Residuals in a\nPermutation Procedure","Statistical Analysis of Mixed Ploidy Populations","Threshold Indicator Taxa Analysis","Empirically Informed Random Trajectory Generation in 3-D","Fit Continuous-Time State-Space and Latent Variable Models for\nQuality Control of Argos Satellite (and Other) Telemetry Data\nand for Estimating Movement Behaviour","Simple Tools to Compute Landscape Connectivity Metrics","(Niche) (R)egion and Niche (Over)lap Metrics for\nMultidimensional Ecological Niches","Nonlinear Regression for Agricultural Applications","Distance-Sampling Analyses for Density and Abundance Estimation","Determines Species Probabilities Based on Functional Traits","Using Tip Dates with Phylogenetic Trees in BEAST","Vector Generalized Linear and Additive Models","Determine Phytoplankton Functional Groups Based on Functional\nTraits","Parentage Assignment using Bi-Allelic Genetic Markers","Normalized Stochasticity Ratio","Bayesian Change-Point Detection and Time Series Decomposition","Scaling with Ranked Subsampling","Simulating Phylogenetic Trees under General Bellman Harris and\nLineage Shift Model"," An R Package for Comprehensive Analysis of Metabolomics Data"," Camera Trap Data Management and Preparation of Occupancy and Spatial Capture-Recapture Analyses"," Spatial-temporal analysis and visualisation of animal movement data"," Species distribution modelling"," Wallace is a modular platform for reproducible modeling of species niches and distributions, written in R","A programmatic interface to many species occurrence data sources","Mark-Recapture Analysis for Survival and Abundance Estimation","Models for Data from Unmarked Animals","R Code for Mark Analysis","Continuous-Time Movement Modeling","Animal Movement Modelling using Hidden Markov Models","Analysis of Habitat Selection by Animals","Calculates landscape metrics for categorical landscape patterns in a tidy workflow.","Animal Movement Tools","Home Range Estimation","Maximum Likelihood Analysis of Animal Movement Behavior Using\nMultivariate Hidden Markov Models","Computes Revisitation Metrics for Trajectory Data\n","Machine Learning for Wildlife Image Classification","Analysis of Ecological Data: Exploratory and Euclidean Methods\nin Environmental Sciences","Hierarchical and Variation Partitioning for Canonical Analysis","Alpha and Beta Diversity Measures","Discrete and Global Optimization Routines","Analysis of Habitat Selection by Animals","Analysis of Animal Movements","Tools to Deal with Raster Maps","Analysis of Diversity","Relationship Matrices for Diploid and Autopolyploid Species","Model Selection and Multimodel Inference Based on (Q)AIC(c)","Animal track reconstruction for high frequency 2-dimensional\r\n \r\n (2D) or 3-dimensional (3D) movement data","Animation of Observed Trajectories Using Spline-Based\r\n \r\n Interpolation","Analyses of Phylogenetic Treeshape","Analogue and Weighted Averaging Methods for Palaeoecology","Key and Secret Management in 'Azure'","Modeling Framework for Cellular Communities in their\r\n \r\n Environments","Amazon Web Services Request Signatures","Better ALignment CONsensus analYsis","Biodiversity Assessment Tools","Brownian bridge movement model","Analysis and Visualization of Macroevolutionary Dynamics on\r\n \r\n Phylogenetic Trees","Benthic Multi-Metric Index","Processes Binary Data Obtained from Fragment Analysis","Point Data Quality Assessment and Coordinate Conversion","Ensemble Platform for Species Distribution Modeling","Bayesian Integrative Models of Trait Evolution","Models for Continuous Traits Evolving in Macroevolutionary\r\n \r\n Landscapes of any Shape","blockmatrix: Tools to solve algebraic systems with partitioned\r\n \r\n matrices","AWS Python SDK' ('boto3') for R","Bayesian Multi-Trait Multi-Environment for Genomic Selection\r\n \r\n Analysis","Bayesian State-Space Models for Animal Movement","Bayesian Phylogeographic and Ecological Clustering","R Graphics Device using Cairo Graphics Library for Creating\r\n \r\n High-Quality Bitmap (PNG, JPEG, TIFF), Vector (PDF, SVG,\r\n \r\n PostScript) and Display (X11 and Win32) Output","Compositional Analysis of Differentially Expressed Proteins in\r\n \r\n Cancer","Continuous-Time Fractionally Integrated ARMA Process for\r\n \r\n Irregularly Spaced Long-Memory Time Series Data","Efficient Phylogenetics on Large Trees","Phylogenetically Classifies Retention Mechanisms of Duplicate\r\nGenes from Gene Expression Data","Motion Trajectory Analysis","Download Data from the Catchment Data Explorer Website","Thermodynamic Calculations and Diagrams for Geochemistry","Compare Groups, Analytically and Graphically","Circular Visualization","Measuring Morphological Diversity and Evolutionary Tempo","Clinical Utility of DNA Testing","Co-Correspondence Analysis Methods","Species coexistence modeling and analysis","Coloring Wildlife Distributions in Space-Time","A Framework for Coalescent Simulation","Manage Environment Specific Configuration Values","Fit Continuous-Time Correlated Random Walk Models to Animal\r\n \r\n Movement Data","Modeling Animal Movement with Continuous-Time Discrete-Space\r\n \r\n Markov Chains","Collection of Scripts for Interface Between 'Cornerstone' and\r\n \r\n 'R'","Swarm Intelligence for Self-Organized Clustering","A system for mapping foreign objects to R variables and\r\n \r\n environments","Chaotic Time Series Analysis","Analysing Density Profiles from Resistance Drilling of Trees","A Complete Tool for Processing and Analyzing Dendrometer Data","Stacks, Queues, and 'Deques' for R","Diversity Estimator","Diversity Dynamics using Fossil Sampling Data","Evolutionary Trait Divergence Between Sister Species and Other\r\n \r\n Paired Lineages","Generic Reproducible Parallel Backend for 'foreach' Loops","Print Tabular Data to Graphics Device","Deep Time Redox Analysis of the Geobiology Ontology Network","Fitting Dose-Response Models Using an Evolutionary Algorithm","Compute and Decompose Inequality in Education","Simulating Community Assembly and Ecological Diversification\r\n \r\n Using Ecospace Frameworks","Simulate and Evaluate Time Series for Environmental Epidemiology","Evolutionary Multiobjective Optimization Algorithms","Evolutionary model-based multiresponse approach","Spatial and Network Based Individual Level Models for Epidemics","Continuous Time Distance-Based and Network-Based Individual\r\n \r\n Level Models for Epidemics","Ensemble Tool for Predictions from Species Distribution Models","Methods for Identification of Outliers in Environmental Data","Prediction of Fish Movement Parameters","Estimate Insect Survival from Fluorescence Data","Estimate Energy Fluxes in Food Webs","Palaeoecological and Palaeogeographical Analysis Tools","Fit Continuous-Time State-Space and Latent Variable Models for\r\n \r\n Filtering Argos Satellite (and Other) Telemetry Data and\r\n \r\n Estimating Movement Behaviour","FUzzy GEnetic, a machine learning algorithm to construct\r\nprediction model based on fuzzy logic","Fungal Species Identification using DNA Barcode with Random\r\nForest","Fuzzy Similarity in Species Distributions","An Interface Between the 'FSK-ML' Standard and 'R'","GPU'/CPU Benchmarking in Debian-Based Systems","Power Calculations Under Genetic Model Misspecification","Classes and Methods for Working with Spectral Data with\r\n \r\n Space-Time Attributes","Extensions to 'ggplot2' for Radiation Spectra","Quantifying Evolution and Selection on Complex Traits","Functions for Community Data and Ordinations","Identify Global Objects in R Expressions","A Package for Graphical Modelling in R","Grammatical Evolution for R","Analyzing Case-Parent Triad and/or Case-Control Data with SNP\r\n \r\n Haplotypes","Access London Natural History Museum Host-Helminth Record\r\nDatabase","Hierarchical Bayesian Species Distribution Models","Heat exchange calculations during physical activity","Particle Swarm Optimisation, with Focus on Environmental Models","High Dimensional Geometry and Set Operations Using Kernel\r\n \r\n Density Estimation, Support Vector Machines, and Convex Hulls","Time Series Management, Analysis and Interpolation for\r\n \r\n Hydrological Modelling","Imperialist Competitive Algorithm","Integro-Difference Equation Spatio-Temporal Models","Macroevolutionary phylogentic analysis of species trees and gene\r\n \r\n trees","Utilities of Institute of Data Analyses and Process Design\r\n \r\n (www.zhaw.ch/idp)'","An Import Mechanism for R","Interpolation and Extrapolation for Species Diversity","Invasive Species Distribution Modelling","Index Numbers in Social Sciences","Isotope Origin Clustering and Assignment Tools","Joint N-Mixture Models for Site-Associated Species","Jack-Knife Support for Evolutionary Distinctiveness Indices I\r\nand W","Joint Species Distribution Models","Create 'Java' Objects and Execute 'Java' Methods","a Keep It Simple Species Migration Model","Discriminating Well Surveyed Spatial Units from Exhaustive\r\nBiodiversity Databases","Ordination and Multivariate Analysis for Ecology","Complete Environment for Bayesian Inference","Modelling Pathways and Movement Potential Within a Landscape","Variable Selection for Latent Class Analysis","LInear Regression in Astronomy","Lysate and Secretome Peptide Feature Plotter","R at the Command-Line via 'r'","Land Use Change Modelling in R","Land Surface Remote Sensing","Process-Based Biogeographical Analysis","Migration and Range Change Estimation in R","Retrieve Data from MacLeish Field Station","Spatial Pattern Detection in Genetic Distance Data Using Moran's\r\n \r\n Eigenvector Maps","A GUI for interactive MDS in R","Landscape and Range Expansion Simulation","Bayesian Mixture Analysis for Metagenomic Community Profiling","MGMS2' for Polymicrobial Samples","Analysis of MHC Data in Non-Model Species","2D and 3D movement-based kernel density estimates (MKDEs)","Multivariate Data Handling in Ecology and Biogeography","Model Evaluation and Analysis","Modeling Language Evolution","Metaheuristic for Optimization","Mouse-Tracking Measures from Trajectory Data","Process and Analyze Mouse-Tracking Data","Calculation of Accumulated Cost Surface and Least-Cost Paths\r\n \r\n Related to Human Movement Across the Landscape","Movement Data Visualization","Estimate Wind Speeds from Bird Trajectories","Multivariate Birth-Death Processes","Fitting Markov-Modulated Linear Regression Models","Models of Trait Macroevolution on Trees","Multivariate Analysis and Visualization for Biological Data","Multivariate Comparative Tools for Fitting Evolutionary Models\r\n \r\n to Morphometric Data","Build and Run Spatially Explicit Agent-Based Models","Extinction Simulation in Food Webs","Mean-Variance Regularization","Evolutionary Transcriptomics","Goodness of Fit Checks for Binomial N-Mixture Models","Setup, Run and Analyze 'NetLogo' Model Simulations from 'R' via\r\n \r\n 'XML'","Non-Negative Matrix Factorization (NMF) using CUDA","Nearest-Neighbor Analysis","Likelihood Based Optimal Partitioning and Indicator Species\r\nAnalysis","Object Oriented Simulator of Marine Ecosystems","Nowcasting by Bayesian Smoothing","Cartography for Statistical Analysis","Estimates of Coefficient of Overlapping for Animal Activity\r\nPatterns","A Meta-Analysis Approach with Filtering for Identifying\r\n \r\n Gene-Level Gene-Environment Interactions with Genetic\r\n \r\n Association Data","Online Time Series Anomaly Detectors","Implementation of the Panel Data Approach Method for Program\r\nEvaluation","Versatile Functions for Working with Pedigrees","Model of Adaptive Trait Evolution","Assessment of Regression Models Performance","Geometric Morphometric Tools for Paleobiology","Reading Portable Encapsulated Projects","Spatial Analysis and Data Mining for Field Ecologists","Nonparametric Stability Analysis","Read Spectral and Logged Data from Foreign Files","Polygenic Inverse Gamma Shifts","Spectral Response Data for Light Sensors","Phylogenetic Tools for Comparative Biology (and Other Things)","PLINK' (and 'GCTA') File Helpers","A Flexible Syntax for Population Dynamic Modelling","Create Useful Summaries of the Portal Data","Row and Position Tracheid Organizer","Representative and Adequate Prioritization Toolkit in R","Rarefaction Methods","Processing Pipeline for 'SamBada' from Pre- To Post-Processing","Range Ancestral State Estimation for Phylogeography and\r\nComparative Analyses","Interface to the Bird-Watching Dataset Proyecto AVIS","Conservation Assessment Tools","Extension for 'R6' Base Class","An Implementation of IDW-PLUS","Normalization and Study of Rank Abundance Distributions","Testing Gene-Environment Interaction for Rare Genetic Variants","R/FrEAK interface","Wrapper for 'Coinbase Pro (GDAX)' Cryptocurrency Exchange","Global Optimization by Differential Evolution in C++","Distance-Sampling Analyses for Density and Abundance Estimation","Extracts Environmental Data from 'ERDDAP' Web Services","Debugging with Restore Points","Tools to Generate Vector Time Series","Interface Between GRASS 7 Geographical Information System and R","Utilities from and Interface to the Bioinfo-C (BIOS) Library","An Individual-Based Population Genetic Simulation Environment","R Package for the OpenSecrets.org API","Interface to the 'Open Tree of Life' API","Interface (Wrapper) to MPI (Message-Passing Interface)","Managing Animal Movement Data with 'PostGIS' and R","EUtils libraries for use in the R environment","Robust Bayesian Variable Selection for Gene-Environment\r\n \r\n Interactions","R interface to Oracle Data Mining","Guidelines for the use of Remote Sensing in Movement Ecology","P-Model","Phylogenetic Ridge Regression Methods for Comparative Studies","Interface for the 'Wildbook' Wildlife Data Management Framework","Spatial Absorbing Markov Chains","Generic implementation of the 'RStudio' connections contract","Shortcuts, Detours and Dead Ends (SDDE) Path Types in Genome\r\nSimilarity Networks","Subgroup Discovery with Evolutionary Fuzzy Systems","Species Distribution Modelling Playground","Species Distribution Modelling Predictor Datasets","Species Distribution Model Selection","Create Virtual Species for Species Distribution Modelling","Species Distribution and Stability Future Models","Spatially Explicit Capture-Recapture","Sampling Design for Spatially Explicit Capture-Recapture","Share Sensitive Information in R Packages","Encrypt Text Using a Shifting Substitution Cipher","Spatially Explicit Capture-Recapture for Linear Habitats","Simulated Haploid Asexual Population Evolution","Sensitivity Analysis for Comparative Methods","Spatially Explicit Structural Equation Modeling","Standardised Rigid Rotations of Articulated Three-Dimensional\r\n \r\n Structures","Species-Habitat Associations","Spatial Individual-Plant Modelling","Supports Developing, Building and Deploying R Solution","Statistical Modeling of Animal Movements","Interface to USDA Databases","Develop and Run Spatially Explicit Discrete Event Simulation\r\n \r\n Models","Evolutionary Analysis","Statistical package for species richness estimation","Sampling Surface Simulation for Areal Sampling Methods","Species Identity and Evolution in R","Functions for Spatial Thinning of Species Occurrence Records for\r\nUse in Ecological Models","Stacked Species Distribution Modelling","Species Sensitivity Distributions","Simulating the Evolution of Biological Sequences","Species Distribution and Stability Future Models","Estimation of Weight Functions in Meta Analysis","Identify Metabolic Subpathways Mediated by MicroRNAs","Species-Habitat Associations","Stochastic Linear Ornstein-Uhlenbeck Comparative Hypotheses","Making Sophisticated and Aesthetical Figures in R","Handling Taxonomic Lists","Fit, Simulate and Diagnose Models for Network Evolution Based on\r\nExponential-Family Random Graph Models","Approximate Gaussian Processes Using the Fourier Basis","Stability Analysis of Genotype by Environment Interaction (GEI)","A Tool for Downloading Functional Traits Data for Plant Species","Data Frame Class for Tracking Data","Shared Tools for Tracking Data","Polygon and Path Transformations","Treatment of Membrane-Transport Data","Discovery, Access and Manipulation of 'TreeBASE' Phylogenies","Friendly R Startup Configuration","Estimating birth and death rates based on phylogenies","Simulating Phylogenetic Trees","Innovative Trend Analysis and Time-Series Change Point Analysis","Combine Multiple GWAS by Using Gene-Environment Interactions","Tools for the Analysis of Animal Track Data","Uncertainty Analysis for Species Distribution Models","Vault Client for Secrets and Sensitive Data","Static and Dynamic 3D Plots for the 'vegan' Package","Jost Diversity Measures for Community Data","Frequency Matrix Approach for Assessing Very Low Frequency\r\nVariants in Sequence Records","A Collection of Tools for the Analysis of Remote Acoustic\r\n \r\n Telemetry Data","Calculate Indices of Dynamic Interaction for Wildlife Tracking\r\nData","SciViews - Miscellaneous Functions","Proteomics Data Analysis Functions","Functions to Compute Compositional Turnover Using Zeta Diversity","Implements ACME Estimator of Bird and Bat Mortality by Wind\nTurbines","Exploratory Analysis of Genetic and Genomic Data","Exploratory Analyses for the Phylogenetic Comparative Method","eBird Data Extraction and Processing in R","Hierarchical Bayesian Analysis of North American BBS Data","Biological Analysis and Visualization of Weather Radar Data","Methods to Analyse Ring Re-Encounter Data","Bird Point Count Simulator","Estimation of the Number of Fatalities from Carcass Searches","Climate Window Analysis","Distance Sampling Detection Function and Abundance Estimation","Dive Analysis and Calibration","Density Surface Modelling of Distance Sampling Data","Access and Analyze eBird Status and Trends Data","Estimated Marginal Means, aka Least-Squares Means","Simulation of Bird Flight Range","Simulation of Bird Flight Range","Analysis of Light Based Geolocator Data","Geometric Morphometric Analyses of 2D/3D Landmark Data","Add-in to Customize 'ggplot2' Themes","Linear Mixed-Effects Models using 'Eigen' and S4","MCMC Generalised Linear Mixed Models","Mixed GAM Computation Vehicle with Automatic Smoothness\nEstimation","Bayesian Mixing Models in R","Acquisition and Processing of MODIS Products","Interface to the 'MODIS Land Products Subsets' Web Services","Acoustic Template Detection in R","Models for Analysing Moult in Birds","Mark-Recapture Distance Sampling","Multi-Model Inference","Spatial Capture-Recapture (SCR) Methods Using 'nimble'","Open Population Capture-Recapture","Import 'OpenStreetMap' Data as Simple Features or Spatial\nObjects","The Composer of Plots","Presence-Absence Model Evaluation.","Goodness-of-Fit Tests for Capture-Recapture Models","Geographic Data Analysis and Modeling","Loglinear Models for Capture-Recapture Experiments","R Client for the eBird Database of Bird Observations","Resource Selection (Probability) Functions for Use-Availability\nData","Interface to the Global 'Biodiversity' Information Facility API","Access 'iNaturalist' Data Through APIs","Bayesian Graphical Models using MCMC","R2s for Regression Models","'IUCN' Red List Client","R Interface to Stan","Bayesian Applied Regression Modeling via Stan","Sound Analysis and Synthesis","Simple Features for R","An Individual-Based Model of Bird Song Evolution","Classes and Methods for Spatial Data","Taxonomic Information from Around the Web","Easily Install and Load the 'Tidyverse'","Thematic Maps","Analysis of Music and Speech","Visualization of Regression Models","Download Weather Data from Environment and Climate Change Canada"],["Functions to calculate indices for soundscape ecology and other ecology research that uses audio recordings.","Common ecological distributions for 'nimble' models in the form of nimbleFunction objects. \n  Includes Cormack-Jolly-Seber, occupancy, dynamic occupancy, hidden Markov, dynamic hidden Markov, and N-mixture models.\n  (Jolly (1965) &lt;DOI: 10.2307/2333826&gt;, Seber (1965) &lt;DOI: 10.2307/2333827&gt;, Turek et al. (2016) &lt;doi:10.1007/s10651-016-0353-z&gt;).","An implementation of the reverse ecology framework. Reverse ecology\n    refers to the use of genomics to study ecology with no a priori assumptions\n    about the organism(s) under consideration, linking organisms to their\n    environment. It allows researchers to reconstruct the metabolic networks and\n    study the ecology of poorly characterized microbial species from their\n    genomic information, and has substantial potentials for microbial community\n    ecological analysis.","Ordination methods, diversity analysis and other\n  functions for community and vegetation ecologists.","Autosimilarity curves, standardization of spatial extent, dissimilarity indexes that overweight rare species, phylogenetic and functional (pairwise and multisample) dissimilarity indexes and nestedness for phylogenetic, functional and other diversity metrics. This should be a complement to available packages, particularly 'vegan'. ","Collection of R functions and data sets for the support of spatial ecology analyses with a focus on pre, core and post modelling analyses of species distribution, niche quantification and community assembly. Written by current and former members and collaborators of the ecospat group of Antoine Guisan, Department of Ecology and Evolution (DEE) and Institute of Earth Surface Dynamics (IDYST), University of Lausanne, Switzerland. Read Di Cola et al. (2016) &lt;doi:10.1111/ecog.02671&gt; for details.","Bayesian multivariate binary (probit) regression\n    models for analysis of ecological communities.","Functions for phylocom integration, community analyses, null-models, traits and evolution. Implements numerous ecophylogenetic approaches including measures of community phylogenetic and trait diversity, phylogenetic signal, estimation of trait values for unobserved taxa, null models for community and phylogeny randomizations, and utility functions for data input/output and phylogeny plotting. A full description of package functionality and methods are provided by Kembel et al. (2010) &lt;doi:10.1093/bioinformatics/btq166&gt;.","A collection of functions accompanying the book \"Data Analysis in Vegetation Ecology\". 3rd ed. CABI, Oxfordshire, Boston.","Handling, processing, and analyzing geographic\n    data on species' distributions and environmental variables. \n    Read Vilela &amp; Villalobos (2015) &lt;doi: 10.1111/2041-210X.12401&gt; for details.","SHE, FORAM Index and ABC Method analyses and custom plot\n        functions for community data.","A variety of ordination and community analyses\n   useful in analysis of data sets in community ecology.  \n   Includes many of the common ordination methods, with \n   graphical routines to facilitate their interpretation, \n   as well as several novel analyses.","Computational infrastructure for biogeography, community ecology, \n    and biodiversity conservation (Daru et al. 2020) &lt;doi:10.1101/2020.02.12.945691&gt;. \n    It is based on the methods described in Daru et al. (2020) &lt;doi:10.1038/s41467-020-15921-6&gt;. \n    The original conceptual work is described in Daru et al. (2017) &lt;doi:10.1016/j.tree.2017.08.013&gt; \n    on patterns and processes of biogeographical regionalization. Additionally, the package \n    contains fast and efficient functions to compute more standard conservation measures \n    such as phylogenetic diversity, phylogenetic endemism, evolutionary distinctiveness \n    and global endangerment, as well as compositional turnover (e.g., beta diversity). ","Characterizing environmental microbiota diversity using amplicon-based next generation sequencing (NGS) data. Functions are developed to manipulate operational taxonomic unit (OTU) table, perform descriptive and inferential statistics, and generate publication-quality plots.","Fits Modern Analogue Technique and Weighted Averaging transfer \n  \t     function models for prediction of environmental data from species \n\t     data, and related methods used in palaeoecology.","Simulation-based sampling protocol (SSP) is an R package design to estimate sampling effort in studies of\n    ecological communities based on the definition of pseudo-multivariate standard error (MultSE) (Anderson &amp; Santana-Garcon, 2015) &lt;doi:10.1111/ele.12385&gt; and simulation\n    of ecological data. The theoretical background is described in Guerra-Castro et al. (2020) &lt;doi:10.1101/2020.03.19.996991&gt;.","Graphical User Interface (via the R-Commander) and utility functions (often based on the vegan package) for statistical analysis of biodiversity and ecological communities, including species accumulation curves, diversity indices, Renyi profiles, GLMs for analysis of species abundance and presence-absence, distance matrices, Mantel tests, and cluster, constrained and unconstrained ordination analysis. A book on biodiversity and community ecology analysis is available for free download from the website. In 2012, methods for (ensemble) suitability modelling and mapping were expanded in the package.","A framework package aimed to provide standardized computational environment for specialist work via object classes to represent the data coded by samples, taxa and segments (i.e. subpopulations, repeated measures). It supports easy processing of the data along with cross tabulation and relational data tables for samples and taxa. An object of class `mefa' is a project specific compendium of the data and can be easily used in further analyses. Methods are provided for extraction, aggregation, conversion, plotting, summary and reporting of `mefa' objects. Reports can be generated in plain text or LaTeX format. Vignette contains worked examples.","Tools for timescale decomposition of the classic variance ratio of community ecology. Tools are as described in Zhao et al (in prep), extending commonly used methods introduced by Peterson et al (1975) &lt;doi: 10.2307/1936306&gt;.","Fit and plot macroecological patterns predicted by the Maximum\n    Entropy Theory of Ecology (METE).","A null model randomizing semi-quantitative multi-classes (or ordinal) data by swapping sub-matrices while both the row and the column marginal sums are held constant.","Functions are primarily functions for systems of ordinary differential equations, difference equations, and eigenanalysis and projection of demographic matrices; \n    data are for examples. Documentation of methods is provided in \n    Stevens, MHH (2009, &lt;https://www.springer.com/gp/book/9780387898810&gt;).","Efficient computations for null models that require shuffling columns on big matrix data.\n             This package provides functions for faster computation of diversity measure statistics\n             when independent random shuffling is applied to the columns of a given matrix. \n             Given a diversity measure f and a matrix M, the provided functions can generate random samples \n             (shuffled matrix rows of M), the mean and variance of f, and the p-values of this measure \n             for two different null models that involve independent random shuffling of the columns of M.\n             The package supports computations of alpha and beta diversity measures.  ","Tools for easy generation of (life-history) traits maps based on\n    species range (extent-of-occurrence) maps.","Provides Tools for Accessing the Botanical Information and Ecology Network Database.  The BIEN database contains cleaned and standardized botanical data including occurrence, trait, plot and taxonomic data (See &lt;http://Bien.nceas.ucsb.edu/bien/&gt; for more Information).  This package provides functions that query the BIEN database by constructing and executing optimized SQL queries.","Tools for a wavelet-based approach to analyzing spatial synchrony, principally in ecological data. Some tools will be useful for studying community synchrony. See, for instance, Sheppard et al (2016) &lt;doi: 10.1038/NCLIMATE2991&gt;, Sheppard et al (2017) &lt;doi: 10.1051/epjnbp/2017000&gt;, Sheppard et al (2019) &lt;doi: 10.1371/journal.pcbi.1006744&gt;.","Exploration of simulation models (apps) of various infectious disease transmission dynamics scenarios.\n    The purpose of the package is to help individuals learn \n    about infectious disease epidemiology (ecology/evolution) from a dynamical systems perspective.\n    All apps include explanations of the underlying models and instructions on what to do with the models. ","Provides functions and data sets to accompany the two volume publication \"Applied Hierarchical Modeling in Ecology: Analysis of distribution, abundance and species richness in R and BUGS\" by Marc Kry and Andy Royle: volume 1 (2016, ISBN: 978-0-12-801378-6) and volume 2 (2020, ISBN: 978-0-12-809585-0), &lt;https://www.mbr-pwrc.usgs.gov/pubanalysis/keryroylebook&gt;.","Poisson-lognormal generalized linear mixed model analysis of multivariate counts data using MCMC, aiming to infer the changes in relative proportions of individual variables. The package was originally designed for sequence-based analysis of microbial communities (\"metabarcoding\", variables = operational taxonomic units, OTUs), but can be used for other types of multivariate counts, such as in ecological applications (variables = species). The results are summarized and plotted using 'ggplot2' functions. Includes functions to remove sample and variable outliers and reformat counts into normalized log-transformed values for correlation and principal component/coordinate analysis. Walkthrough and examples: http://www.bio.utexas.edu/research/matz_lab/matzlab/Methods_files/walkthroughExample_mcmcOTU_R.txt. ","FD is a package to compute different multidimensional FD indices. It implements a distance-based framework to measure FD that allows any number and type of functional traits, and can also consider species relative abundances. It also contains other useful tools for functional ecology.","Data files and functions accompanying the book Korner-Nievergelt, Roth, von Felten, Guelat, Almasi, Korner-Nievergelt (2015) \"Bayesian Data Analysis in Ecology using R, BUGS and Stan\", Elsevier, New York.","The program can perform loop analysis and plot network\n        structure (especially for food webs),including minimum spanning\n        tree, loop decomposition of weighted directed graphs, and other\n        network properties which may be related to food chain\n        properties in ecology.","Transfer functions and statistical operations for\n        paleoecology","Read Condensed Cornell Ecology Program ('CEP') and legacy\n    'CANOCO' files into R data frames.","Calculates a suite of hydrologic indices for daily time series data that are widely used in hydrology and stream ecology.","A set of functions to analyse and compare texts, using classical \n  text mining\tfunctions, as well as those from theoretical ecology.","Spatio-temporal locations of an animal are computed \n    from annotated data with a hidden Markov  model via particle\n    filter algorithm. The package is relatively robust to varying\n    degrees of shading.\n    The hidden Markov model is described in Movement Ecology (Rakhimberdiev et al., 2015) &lt;doi:10.1186/s40462-015-0062-5&gt;,\n    general package description is in the Methods in Ecology and Evolution (Rakhimberdiev et al., 2017) &lt;doi:10.1111/2041-210X.12765&gt;\n    and package accuracy assessed in the Journal of Avian Biology (Rakhimberdiev et al. 2016) &lt;doi:10.1111/jav.00891&gt;.","Bootstrapped response and correlation functions,\n    seasonal correlations and evaluation of reconstruction\n    skills for use in dendroclimatology and dendroecology.","Simulates and computes the (maximum) likelihood of a dynamical model of island biota assembly through speciation, immigration and extinction. See e.g. Valente et al. 2015. Ecology Letters 18: 844-852, &lt;DOI:10.1111/ele.12461&gt;.","\n    Tools and data to accompany Schlicht, L., Valcu, M., &amp; Kempenaers, B. (2015) &lt;doi:10.1111/1365-2656.12293&gt;. Spatial patterns of extra-pair paternity: beyond paternity gains and losses. Journal of Animal Ecology, 84(2), 518-531.","Permutational method to incorporate taxonomic uncertainty  and some functions to assess its effects on parameters of some widely used multivariate methods in ecology, as explained in Cayuela et al. (2011) &lt;doi:10.1111/j.1600-0587.2009.05899.x&gt;.","A collection of functions for microbial ecology and other\n   applications of genomics and metagenomics. Companion package for the\n   Enveomics Collection (Rodriguez-R, L.M. and Konstantinidis, K.T., 2016\n   &lt;DOI:10.7287/peerj.preprints.1900v1&gt;).","Hierarchical models for the analysis of species-area \n  relationships (SARs) by combining several data sets and covariates; \n  with a global data set combining individual SAR studies; \n  as described in Solymos and Lele \n  (2012, Global Ecology and Biogeography 21, 109-120).","Contains functions to access movement data stored in 'movebank.org'\n    as well as tools to visualize and statistically analyze animal movement data,\n    among others functions to calculate dynamic Brownian Bridge Movement Models.\n    Move helps addressing movement ecology questions.","Calculates the relative growth rate (RGR) of a series of individuals by building a life table and solving the Lotka-Birch equation. (See Birch, L. C. 1948. The intrinsic rate of natural increase of an insect population. - Journal of Animal Ecology 17: 15-26) &lt;doi:10.2307/1605&gt;.","Download and generate summaries for the rodent,\n    plant, ant, and weather data from the Portal Project. Portal is a\n    long-term (and ongoing) experimental monitoring site in the Chihuahua\n    desert. The raw data files can be found at\n    &lt;https://github.com/weecology/portaldata&gt;.","Fuzzy set ordination is a multivariate analysis used in ecology to\n    relate the composition of samples to possible explanatory variables.  While\n    differing in theory and method, in practice, the use is similar to 'constrained\n    ordination.'  The package contains plotting and summary functions as well as\n    the analyses.  ","Implements a Gibbs sampling algorithm to perform Bayesian inference\n    on biallelic SNP frequencies, genotypes and heterozygosity (observed and\n    expected) in a population of autopolyploids. See the published paper in\n    Molecular Ecology Resources: Blischak et al. (2016) &lt;doi:10.1111/1755-0998.12493&gt;.","Provides a framework for generating virtual species distributions,\n    a procedure increasingly used in ecology to improve species distribution\n    models. This package integrates the existing methodological approaches with the\n    objective of generating virtual species distributions with increased ecological\n    realism.","It has tools for normalization of rank abundance\n     distributions (RAD) to a desired number of ranks using MaxRank\n     Normalization method.\n     RADs are commonly used in biology/ecology and mathematically equivalent\n     to complementary cumulative distributions (CCDFs) which are used in\n     physics, linguistics and sociology and more generally in data science.","Within the area of movement ecology, it can be useful to identify an animal's sites of interest. These will be places that an animal uses a lot and so will spend a proportionately longer time there. The aim of this package is to identify both the position and size of these sites of interest from high resolution (sub-second) positional data. ","The Behavioral Change Point Analysis (BCPA) is a method of\n    identifying hidden shifts in the underlying parameters of a time series,\n    developed specifically to be applied to animal movement data which is\n    irregularly sampled.  The method is based on: E.\n    Gurarie, R. Andrews and K. Laidre A novel method for identifying\n    behavioural changes in animal movement data (2009) Ecology Letters 12:5\n    395-408.","Implements stochastic simulations of community assembly (ecological\n    diversification) using customizable ecospace frameworks (functional trait\n    spaces). Provides a wrapper to calculate common ecological disparity and\n    functional ecology statistical dynamics as a function of species richness.\n    Functions are written so they will work in a parallel-computing environment.","A set of tools for displaying, modeling and analysing\n        multivariate abundance data in community ecology. See\n        'mvabund-package.Rd' for details of overall package organization.\n        The package is implemented with the Gnu Scientific Library\n        (&lt;http://www.gnu.org/software/gsl/&gt;) and 'Rcpp'\n        (&lt;http://dirk.eddelbuettel.com/code/rcpp.html&gt;) 'R' / 'C++' classes.","Implements a computational framework for a pattern-based, \n    zoneless analysis, and visualization of (ethno)racial topography \n    (Dmowska, Stepinski, and Nowosad (2020) &lt;doi:10.1016/j.apgeog.2020.102239&gt;).\n    It is a reimagined\n    approach for analyzing residential segregation and racial diversity based on \n    the concept of 'landscape used in the domain of landscape ecology.","A programmatic interface to &lt;http://www.fishbase.org&gt;, re-written\n    based on an accompanying 'RESTful' API. Access tables describing over 30,000\n    species of fish, their biology, ecology, morphology, and more. This package also\n    supports experimental access to &lt;http://www.sealifebase.org&gt; data, which contains\n    nearly 200,000 species records for all types of aquatic life not covered by\n    'FishBase.'","Sampling surface simulation is useful in the comparison of different areal sampling methods\n             in forestry, ecology and natural resources. The sampSurf package allows the simulation \n\t\t\t of numerous sampling methods for standing trees and downed woody debris in a spatial context.\n\t\t\t It also provides an S4 class and method structure that facilitates the addition of new sampling\n\t\t\t methods.","Provides functions to generate and analyze spatially-explicit individual-based multistate movements in rivers,\n  heterogeneous and homogeneous spaces. This is done by incorporating landscape bias on local behaviour, based on\n  resistance rasters. Although originally conceived and designed to simulate trajectories of species constrained to\n  linear habitats/dendritic ecological networks (e.g. river networks), the simulation algorithm is built to be\n  highly flexible and can be applied to any (aquatic, semi-aquatic or terrestrial) organism, independently on the\n  landscape in which it moves. Thus, the user will be able to use the package to simulate movements either in\n  homogeneous landscapes, heterogeneous landscapes (e.g. semi-aquatic animal moving mainly along rivers but also using\n  the matrix), or even in highly contrasted landscapes (e.g. fish in a river network). The algorithm and its input\n  parameters are the same for all cases, so that results are comparable. Simulated trajectories can then be used as\n  mechanistic null models (Potts &amp; Lewis 2014, &lt;DOI:10.1098/rspb.2014.0231&gt;) to test a variety of 'Movement Ecology'\n  hypotheses (Nathan et al. 2008, &lt;DOI:10.1073/pnas.0800375105&gt;), including landscape effects (e.g. resources, \n  infrastructures) on animal movement and species site fidelity, or for predictive purposes (e.g. road mortality risk,\n  dispersal/connectivity). The package should be relevant to explore a broad spectrum of ecological phenomena, such as\n  those at the interface of animal behaviour, management, landscape and movement ecology, disease and invasive species\n  spread, and population dynamics.","Counts occurrences of motifs in bipartite networks, as well as the number of\n    times each node or link appears in each unique position within motifs. Has support for both\n    binary and weighted motifs: can calculate the mean weight of motifs and the standard deviation \n    of their mean weights. Intended for use in ecology, but its methods are general and can be\n    applied to any bipartite network. Full details are given in Simmons et al. (2019)\n    &lt;doi:10.1111/2041-210X.13149&gt;.","Unsupervised, multivariate, binary clustering for meaningful annotation of data, taking into account the uncertainty in the data. A specific constructor for trajectory analysis in movement ecology yields behavioural annotation of trajectories based on estimated local measures of velocity and turning angle, eventually with solar position covariate as a daytime indicator, (\"Expectation-Maximization Binary Clustering for Behavioural Annotation\").","Contains data, code, and figures from Hill et al. 2018a (Journal of Experimental Marine Biology and Ecology; &lt;DOI: 10.1016/j.jembe.2018.07.006&gt;) and Hill et al. 2018b (Data In Brief &lt;DOI: 10.1016/j.dib.2018.09.133&gt;). Datasets document plant allometry, stem heights, nutrient and stable isotope content, and sediment denitrification enzyme assays. The data and analysis offer an examination of nitrogen uptake and allocation in two salt marsh plant species.","Calculates spatial pattern analysis using a T-square sample procedure.\n  This method is based on two measures \"x\" and \"y\".\n  \"x\" - Distance from the random point to the nearest individual.\n  \"y\" - Distance from individual to its nearest neighbor.\n  This is a methodology commonly used in phytosociology or marine benthos ecology to analyze the species' distribution (random, uniform or clumped patterns).\n  Ludwig &amp; Reynolds (1988, ISBN:0471832359).","Data sets to accompany the VGAM package and\n\tthe book \"Vector Generalized Linear and\n\tAdditive Models: With an Implementation in R\" (Yee, 2015)\n\t&lt;DOI:10.1007/978-1-4939-2818-7&gt;.\n\tThese are used to illustrate vector generalized\n\tlinear and additive models (VGLMs/VGAMs), and associated models\n\t(Reduced-Rank VGLMs, Quadratic RR-VGLMs, Row-Column\n\tInteraction Models, and constrained and unconstrained ordination\n\tmodels in ecology).","Given a site by species interaction matrix, users can make inferences about species interactions by performance hypothesis comparing test statistics against a null distribution. The current package provides algorithms and metrics for niche-overlap, body size ratios and species co-occurrence. Users can also integrate their own algorithms and metrics within these frameworks or completely novel null models. Detailed explanations about the underlying assumptions of null model analysis in ecology can be found at http://ecosimr.org. ","Supports analyses using the Global Forest Change dataset released\n    by Hansen et al. gfcanalysis was originally written for the Tropical Ecology \n    Assessment and Monitoring (TEAM) Network. For additional details on the \n    Global Forest Change dataset, see: Hansen, M. et al. 2013. \"High-Resolution \n    Global Maps of 21st-Century Forest Cover Change.\" Science 342 (15 \n    November): 850-53. The forest change data and more information on the \n    product is available at &lt;http://earthenginepartners.appspot.com&gt;.","Generate and analyze Optimal Channel Networks (OCNs): \n\toriented spanning trees reproducing all scaling features characteristic \n\tof real, natural river networks. As such, they can be used in a variety \n\tof numerical experiments in the fields of hydrology, ecology and \n\tepidemiology. See Carraro et al. (2020) &lt;doi:10.1002/ece3.6479&gt; \n\tfor a presentation of the package; Rinaldo et al. (2014) \n\t&lt;doi:10.1073/pnas.1322700111&gt; for a theoretical overview on the OCN \n\tconcept; Furrer and Sain (2010) &lt;doi:10.18637/jss.v036.i10&gt; for the \n\tconstruct used.","Computes 46 optimized distance and similarity measures for comparing probability functions (Drost (2018) &lt;doi:10.21105/joss.00765&gt;). These comparisons between probability functions have their foundations in a broad range of scientific disciplines from mathematics to ecology. The aim of this package is to provide a core framework for clustering, classification, statistical inference, goodness-of-fit, non-parametric statistics, information theory, and machine learning tasks that are based on comparing univariate or multivariate probability functions.","Population dynamic models underpin a range of analyses and applications in ecology and epidemiology. The various approaches for analysing population dynamics models (MPMs, IPMs, ODEs, POMPs, PVA) each require the model to be defined in a different way. This makes it difficult to combine different modelling approaches and data types to solve a given problem. 'pop' aims to provide a flexible and easy to use common interface for constructing population dynamic models and enabling to them to be fitted and analysed in lots of different ways.","Accurate estimates of the diets of predators are required\n    in many areas of ecology, but for many species current methods are\n    imprecise, limited to the last meal, and often biased. The diversity\n    of fatty acids and their patterns in organisms, coupled with the\n    narrow limitations on their biosynthesis, properties of digestion in\n    monogastric animals, and the prevalence of large storage reservoirs of\n    lipid in many predators, led us to propose the use of quantitative\n    fatty acid signature analysis (QFASA) to study predator diets.","Computes the probability of a set of species abundances of a single or multiple samples of individuals with one or more guilds under a mainland-island model. One must specify the mainland (metacommunity) model and the island (local) community model. It assumes that species fluctuate independently. The package also contains functions to simulate under this model. See Haegeman, B. &amp; R.S. Etienne (2017). A general sampling formula for community structure data. Methods in Ecology &amp; Evolution 8: 1506-1519 &lt;doi:10.1111/2041-210X.12807&gt;. ","The ability to generate virtual niches enables virtual ecology \n    experiments that have been advocated for evaluating species-environment \n    relationship models. Using multivariate normal distributions and matrix \n    algebra this package provides a unique computational framework to generate \n    integrated fundamental and realised niches, which is important given \n    differences in possible complexity of realised niche shapes that result \n    from biotic interactions Etherington and Omondiagbe (2019) &lt;doi:10.21105/joss.01661&gt;.","In fields such as ecology, microbiology, and genomics, non-Euclidean distances are widely applied to describe pairwise dissimilarity between samples. Given these pairwise distances, principal coordinates analysis (PCoA) is commonly used to construct a visualization of the data. However, confounding covariates can make patterns related to the scientific question of interest difficult to observe. We provide 'aPCoA' as an easy-to-use tool to improve data visualization in this context, enabling enhanced presentation of the effects of interest. Details are described in Yushu Shi, Liangliang Zhang, Kim-Anh Do, Christine Peterson and Robert Jenq (2020) &lt;arXiv:2003.09544&gt;.","Bayesian approaches for analyzing multivariate data in ecology. Estimation is performed using Markov Chain Monte Carlo (MCMC) methods via Three. JAGS types of models may be fitted: 1) With explanatory variables only, boral fits independent column Generalized Linear Models (GLMs) to each column of the response matrix; 2) With latent variables only, boral fits a purely latent variable model for model-based unconstrained ordination; 3) With explanatory and latent variables, boral fits correlated column GLMs with latent variables to account for any residual correlation between the columns of the response matrix. ","Description: Tools for constructing niche models and analyzing patterns of niche evolution.  Acts as an interface for many popular modeling algorithms, and allows users to conduct Monte Carlo tests to address basic questions in evolutionary ecology and biogeography.  Warren, D.L., R.E. Glor, and M. Turelli (2008) &lt;doi:10.1111/j.1558-5646.2008.00482.x&gt; Glor, R.E., and D.L. Warren (2011) &lt;doi:10.1111/j.1558-5646.2010.01177.x&gt;  Warren, D.L., R.E. Glor, and M. Turelli (2010) &lt;doi:10.1111/j.1600-0587.2009.06142.x&gt; Cardillo, M., and D.L. Warren (2016) &lt;doi:10.1111/geb.12455&gt; D.L. Warren, L.J. Beaumont, R. Dinnage, and J.B. Baumgartner (2019) &lt;doi:10.1111/ecog.03900&gt;.  ","An implementation of the nodiv algorithm, see Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. &lt;DOI:10.1111/2041-210X.12283&gt;. Package for phylogenetic analysis of species distributions. The main function goes through each node in the phylogeny, compares the distributions of the two descendant nodes, and compares the result to a null model. This highlights nodes where major distributional divergence have occurred. The distributional divergence for these nodes is mapped using the SOS statistic.","Set of tools for reading, writing and transforming spatial and seasonal data in ecology, model selection and specific statistical tests. It includes functions to discretize polylines into regular point intervals, link observations to those points, compute geographical coordinates at regular intervals between waypoints, read subsets of big rasters, compute zonal statistics or table of categories within polygons or circular buffers from raster. The package also provides miscellaneous functions for model selection, spatial statistics, geometries, writing data.frame with Chinese characters, and some other functions for field ecologists.","Capable of deriving seasonal statistics, such as \"normals\", and\n  analysis of seasonal data, such as departures. This package also has\n  graphics capabilities for representing seasonal data, including boxplots for\n  seasonal parameters, and bars for summed normals. There are many specific\n  functions related to climatology, including precipitation normals,\n  temperature normals, cumulative precipitation departures and precipitation\n  interarrivals. However, this package is designed to represent any\n  time-varying parameter with a discernible seasonal signal, such as found\n  in hydrology and ecology.","A programmatic interface to &lt;http://sp2000.org.cn&gt;, re-written based on an accompanying 'Species 2000' API. Access tables describing catalogue of the Chinese known species of animals, plants, fungi, micro-organisms, and more. This package also supports access to catalogue of life global &lt;http://catalogueoflife.org&gt; and catalogue of life Taiwan &lt;https://taibnet.sinica.edu.tw/home_eng.php&gt;. The development of 'SP2000' package were supported by Biodiversity Survey and Assessment Project of the Ministry of Ecology and Environment, China &lt;2019HJ2096001006&gt;,Yunnan University's \"Double First Class\" Project &lt;C176240405&gt; and Yunnan University's Research Innovation Fund for Graduate Students &lt;2019227&gt;.","Provides a mixture model for clustering individuals (or sampling groups) into stocks based on their genetic profile. Here, sampling groups are individuals that are sure to come from the same stock (e.g. breeding adults or larvae). The mixture (log-)likelihood is maximised using the EM-algorithm after find good starting values via a K-means clustering of the genetic data. Details can be found in Foster, Feutry, Grewe, Berry, Hui, Davies (2019) Reliably Discriminating Stock Structure with Genetic Markers: Mixture Models with Robust and Fast Computation. Molecular Ecology Resources.","Reproducible and remixable species distribution modelling.\n    The package reads user submitted modules from an online repository, runs\n    full species distribution modelling workflows and returns output that is\n    fully reproducible. For examples and detailed discussion refer to: N.Golding\n    et al. (2017) 'The zoon r package for reproducible and shareable species\n    distribution modelling'. Methods in Ecology and Evolution.\n    &lt;doi:10.1111/2041-210X.12858&gt;. The package 'SDMTools' is used for testing\n    and, though this package is archived, you can access it here if \n    needed, &lt;https://cran.r-project.org/src/contrib/Archive/SDMTools/&gt;. ","Tools for the identification of unique of multilocus genotypes when both genotyping error and missing data may be present. The package is targeted at those working with large datasets and databases containing multiple samples of each individual, a situation that is common in conservation genetics, and particularly in non-invasive wildlife sampling applications. Functions explicitly incorporate missing data, and can tolerate allele mismatches created by genotyping error. If you use this tool, please cite the package using the journal article in Molecular Ecology Resources (Galpern et al., 2012). Please use citation('allelematch') to call the full citation. For users with access to the associated journal article, tutorial material is also available as supplementary material to the article describing this software, the citation for which can be called using citation('allelematch').","Automated flagging of common spatial and temporal errors in biological and paleontological collection data, for the use in conservation, ecology and paleontology. Includes automated tests to easily flag (and exclude) records assigned to country or province centroid, the open ocean, the headquarters of the Global Biodiversity Information Facility, urban areas or the location of biodiversity institutions (museums, zoos, botanical gardens, universities). Furthermore identifies per species outlier coordinates, zero coordinates, identical latitude/longitude and invalid coordinates. Also implements an algorithm to identify data sets with a significant proportion of rounded coordinates. Especially suited for large data sets. The reference for the methodology is: Zizka et al. (2019) doi:10.1111/2041-210X.13152.","We implement a common genotyping workflow with a standardized software interface. 'genotypeR' designs genotyping markers from vcf files, outputs markers for multiplexing suitability on various platforms (Sequenom and Illumina GoldenGate), and provides various QA/QC and analysis functions. We developed this package to analyze data in Stevison LS, SA Sefick, CA Rushton, and RM Graze. 2017. Invited Review: Recombination rate plasticity: revealing mechanisms by design. Philosophical Transactions Royal Society London B Biol Sci 372:1-14. &lt;DOI:10.1098/rstb.2016.0459&gt;, and have published it here Sefick, S.A., M.A. Castronova, and L.S. Stevison. 2017. GENOTYPER: An integrated R packages for single nuleotide polymorphism genotype marker design and data analysis. Methods in Ecology and Evolution 9: 1318-1323. &lt;DOI: 10.1111/2041-210X.12965&gt;.  ","Prediction of behaviour from movement \n\tcharacteristics using observation and random forest for the analyses of movement\n\tdata in ecology.\n\tFrom movement information (speed, bearing...) the model predicts the\n\tobserved behaviour (movement, foraging...) using random forest. The\n\tmodel can then extrapolate behavioural information to movement data\n\twithout direct observation of behaviours.\n\tThe specificity of this method relies on the derivation of multiple predictor variables from the\n\tmovement data over a range of temporal windows. This procedure allows to capture\n\tas much information as possible on the changes and variations of movement and\n\tensures the use of the random forest algorithm to its best capacity. The method\n\tis very generic, applicable to any set of data providing movement data together with\n\tobservation of behaviour.","Provides neutral landscape models (&lt;doi:10.1007/BF02275262&gt;,\n    &lt;http://sci-hub.tw/10.1007/bf02275262&gt;).  \n    Neutral landscape models range from \"hard\" \n    neutral models (completely random distributed), to \"soft\" neutral models \n    (definable spatial characteristics) and generate landscape patterns that are \n    independent of ecological processes.\n    Thus, these patterns can be used as null models in landscape ecology. 'nlmr' \n    combines a large number of algorithms from other published software for \n    simulating neutral landscapes. The simulation results are obtained in a\n    geospatial data format (raster* objects from the 'raster' package) and can,\n    therefore, be used in any sort of raster data operation that is performed \n    with standard observation data.                                                                                                  ","A collection of statistical tools for objective (non-supervised) applications \n             of the Regional Frequency Analysis methods in hydrology. \n             The package refers to the index-value method and, more precisely, helps the\n             hydrologist to: (1) regionalize the index-value; (2) form homogeneous regions \n             with similar growth curves; (3) fit distribution functions to the \n             empirical regional growth curves.\n             Most of the methods are those described in the Flood Estimation Handbook \n            (Centre for Ecology &amp; Hydrology, 1999, ISBN:9781906698003).\n             Homogeneity tests from Hosking and Wallis (1993) &lt;doi:10.1029/92WR01980&gt; \n             and Viglione et al. (2007) &lt;doi:10.1029/2006WR005095&gt; are available.","Linear model calculations are made for many random versions of data.  \n    Using residual randomization in a permutation procedure, sums of squares are \n    calculated over many permutations to generate empirical probability distributions \n    for evaluating model effects.  This packaged is described by \n    Collyer &amp; Adams (2018) &lt;doi:10.1111/2041-210X.13029&gt;.  Additionally, coefficients, statistics, fitted values, and residuals generated over many \n    permutations can be used for various procedures including pairwise tests, prediction, classification, and\n    model comparison.  This package should provide most tools one could need for the analysis of\n    high-dimensional data, especially in ecology and evolutionary biology, but certainly other fields, as well.","Allows users to calculate pairwise Nei's Genetic Distances (Nei 1972), pairwise Fixation\n Indexes (Fst) (Weir &amp; Cockerham 1984) and also Genomic Relationship matrixes following Yang et al. (2010) in mixed and single\n ploidy populations. Bootstrapping across loci is implemented during Fst calculation to generate confidence intervals and p-values\n around pairwise Fst values. StAMPP utilises SNP genotype data of any ploidy level (with the ability to handle missing data) and is coded to  \n utilise multithreading where available to allow efficient analysis of large datasets. StAMPP is able to handle genotype data from genlight objects \n allowing integration with other packages such adegenet.\n Please refer to LW Pembleton, NOI Cogan &amp; JW Forster, 2013, Molecular Ecology Resources, 13(5), 946-952. &lt;doi:10.1111/1755-0998.12129&gt; for the appropriate citation and user manual. Thank you in advance.","Uses indicator species scores across binary partitions of\n  a sample set to detect congruence in taxon-specific changes of abundance\n  and occurrence frequency along an environmental gradient as evidence of\n  an ecological community threshold.  Relevant references include: Baker,\n  ME and RS King. 2010. A new method for detecting and interpreting\n  biodiversity and ecological community thresholds.  Methods in Ecology and\n  Evolution 1(1): 25:37. King, RS and ME Baker. 2010. Considerations for\n  identifying and interpreting ecological community thresholds.  Journal\n  of the North American Benthological Association 29(3):998-1008. Baker ME\n  and RS King. 2013. Of TITAN and straw men: an appeal for greater\n  understanding of community data. Freshwater Science 32(2):489-506.","Creates realistic random trajectories in a 3-D space between two given fix points, so-called conditional empirical random walks (CERWs). The trajectory generation is based on empirical distribution functions extracted from observed trajectories (training data) and thus reflects the geometrical movement characteristics of the mover. A digital elevation model (DEM), representing the Earth's surface, and a background layer of probabilities (e.g. food sources, uplift potential, waterbodies, etc.) can be used to influence the trajectories.\n    Unterfinger M (2018). \"3-D Trajectory Simulation in Movement Ecology: Conditional Empirical Random Walk\". Master's thesis, University of Zurich. &lt;https://www.geo.uzh.ch/dam/jcr:6194e41e-055c-4635-9807-53c5a54a3be7/MasterThesis_Unterfinger_2018.pdf&gt;.\n    Technitis G, Weibel R, Kranstauber B, Safi K (2016). \"An algorithm for empirically informed random trajectory generation between two endpoints\". GIScience 2016: Ninth International Conference on Geographic Information Science, 9, online. &lt;doi:10.5167/uzh-130652&gt;.","Fits continuous-time random walk and correlated random walk state-space models for quality control animal tracking data ('Argos', processed light-level 'geolocation', 'GPS'). Template Model Builder ('TMB') is used for fast estimation. The 'Argos' data can be: (older) least squares-based locations; (newer) Kalman filter-based locations with error ellipse information; or a mixture of both. The models estimate two sets of location states corresponding to: 1) each observation, which are (usually) irregularly timed; and 2) user-specified time intervals (regular or irregular). Latent variable models are provided to estimate move persistence along tracks as an index of behaviour. 'Jonsen I', 'McMahon CR', 'Patterson TA', 'Auger-Mth M', 'Harcourt R', 'Hindell MA', 'Bestley S' (2019) Movement responses to environment: fast inference of variation among southern elephant seals with a mixed effects model. Ecology 100:e02566 &lt;doi:10.1002/ecy.2566&gt;.","Provides functions to upload vectorial data and derive landscape\n    connectivity metrics in habitat or matrix systems. Additionally, includes an \n    approach to assess individual patch contribution to the overall landscape \n    connectivity, enabling the prioritization of habitat patches. The computation\n    of landscape connectivity and patch importance are very useful in Landscape \n    Ecology research. The metrics available are: number of components, number of \n    links, size of the largest component, mean size of components, class coincidence\n    probability, landscape coincidence probability, characteristic path length, \n    expected cluster size, area-weighted flux and integral index of connectivity.\n    Pascual-Hortal, L., and Saura, S. (2006) &lt;doi:10.1007/s10980-006-0013-z&gt;\n    Urban, D., and Keitt, T. (2001) &lt;doi:10.2307/2679983&gt;\n    Laita, A., Kotiaho, J., Monkkonen, M. (2011) &lt;doi:10.1007/s10980-011-9620-4&gt;.","This package uses a probabilistic method to calculate niche\n    regions and pairwise niche overlap using multidimensional niche indicator\n    data (e.g., stable isotopes, environmental variables, etc.). The niche\n    region is defined as the joint probability density function of the\n    multidimensional niche indicators at a user-defined probability alpha\n    (e.g., 95%).  Uncertainty is accounted for in a Bayesian framework, and the\n    method can be extended to three or more indicator dimensions.  It provides\n    directional estimates of niche overlap, accounts for species-specific\n    distributions in multivariate niche space, and produces unique and\n    consistent bivariate projections of the multivariate niche region.  A\n    forthcoming article by Swanson et al. (Ecology, 2014) provides a detailed\n    description of the methodology.  See the package vignette for a worked\n    example using fish stable isotope data.","Additional nonlinear regression functions using self-start (SS) algorithms. One of the functions is the Beta growth function proposed by Yin et al. (2003) &lt;doi:10.1093/aob/mcg029&gt;. There are several other functions with breakpoints (e.g. linear-plateau, plateau-linear, exponential-plateau, plateau-exponential, quadratic-plateau, plateau-quadratic and bilinear), a non-rectangular hyperbola and a bell-shaped curve. Twenty one (21) new self-start (SS) functions in total. This package also supports the publication 'Nonlinear regression Models and applications in agricultural research' by Archontoulis and Miguez (2015) &lt;doi:10.2134/agronj2012.0506&gt;, a book chapter with similar material &lt;doi:10.2134/appliedstatistics.2016.0003.c15&gt; and a publication by Oddi et. al. (2019) in Ecology and Evolution &lt;doi:10.1002/ece3.5543&gt;. The function 'nlsLMList' uses nlsLM for fitting, but it is otherwise almost identical to 'nlme::nlsList'.In addition, this release of the package provides functions for conducting simulation for 'nlme' and 'gnls' objects as well as bootstrapping. These functions are intended to work with the modeling framework of the 'nlme' package. It also provides four vignettes with extended examples.","Distance-sampling is a popular method for estimating density and \n  abundance of organisms in ecology. Rdistance contains routines that  \n  assist with analysis of\n  distance-sampling data collected on point or line transects. \n  Distance models are specified using regression-like formula (similar \n  to lm, glm, etc.). Abundance routines\n  perform automated bootstrapping and automated detection-function \n  selection. Overall (study area) and site-level (transect or point) \n  abundance estimates are available. A large suite of classical, \n  parametric detection functions are \n  included along with some uncommon parametric \n  functions (e.g., Gamma, negative exponential) and non-parametric\n  smoothed distance functions. Custom (user-defined) detection functions\n  are easily implemented (see vignette). \n  The help files and vignettes have been \n  vetted by multiple authors and tested in workshop \n  settings. ","The objective of these functions is to derive a species assemblage\n    that satisfies a functional trait profile. Restoring resilient ecosystems\n    requires a flexible framework for selecting assemblages that are based on the\n    functional traits of species. However, current trait-based models have been\n    limited to algorithms that can only select species by optimising specific trait\n    values, and could not elegantly accommodate the common desire among restoration\n    ecologists to produce functionally diverse assemblages. We have solved this\n    problem by applying a non-linear optimisation algorithm that optimises Rao Q,\n    a closed-form functional trait diversity index that incorporates species\n    abundances, subject to other linear constraints. This framework generalises\n    previous models that only optimised the entropy of the community, and can\n    optimise both functional diversity and entropy simultaneously. This package\n    can also be used to generate experimental assemblages to test the effects\n    of community-level traits on community dynamics and ecosystem function. The\n    method is based on theory discussed in Laughlin (2014, Ecology Letters)\n    &lt;doi.org/10.1111/ele.12288&gt;.","Assists performing tip-dating of phylogenetic trees with BEAST \n    BEAST is a popular software for phylogenetic analysis.\n    The package assists the implementation of various phylogenetic tip-\n\tdating tests using BEAST. It contains two main functions.\n\tThe first one allows preparing date randomization analyses, \n\twhich assess the temporal signal of a data set. \n\tThe second function allows performing leave-one-out analyses,\n\twhich test for the consistency between independent calibration sequences\n\tand allow pinpointing those leading to potential bias.\n\tThe included tutorial provides detailed step-by-step instructions.\n\tAn expanded description of the package can be found in article:\n\tRieux, A. and Khatchikian, C.E. (2017), \n\tTIPDATINGBEAST: an R package to assist the implementation of phylogenetic\n\ttip-dating tests using BEAST. Molecular Ecology Resources, 17: 608-613.\n\t&lt;onlinelibrary.wiley.com/doi/full/10.1111/1755-0998.12603&gt;.","An implementation of about 6 major classes of\n    statistical regression models. The central algorithm is\n    Fisher scoring and iterative reweighted least squares.\n    At the heart of this package are the vector generalized linear\n    and additive model (VGLM/VGAM) classes. VGLMs can be loosely\n    thought of as multivariate GLMs. VGAMs are data-driven\n    VGLMs that use smoothing. The book \"Vector Generalized\n    Linear and Additive Models: With an Implementation in R\"\n    (Yee, 2015) &lt;DOI:10.1007/978-1-4939-2818-7&gt; gives details of\n    the statistical framework and the package. Currently only\n    fixed-effects models are implemented. Many (150+) models and\n    distributions are estimated by maximum likelihood estimation\n    (MLE) or penalized MLE. The other classes are RR-VGLMs\n    (reduced-rank VGLMs), quadratic RR-VGLMs, reduced-rank VGAMs,\n    RCIMs (row-column interaction models)---these classes perform\n    constrained and unconstrained quadratic ordination (CQO/UQO)\n    models in ecology, as well as constrained additive ordination\n    (CAO). Note that these functions are subject to change;\n    see the NEWS and ChangeLog files for latest changes.","Verify accepted taxonomic nomenclature of phytoplankton species, assign \n\tspecies to functional group classifications, and manipulate taxonomic and functional diversity \n\tdata. Possible functional classifications include Morpho-functional group (MFG; \n\tSalmaso et al. 2015 &lt;doi:10.1111/fwb.12520&gt;) and CSR (Reynolds 1988; Functional morphology and \n\tthe adaptive strategies of phytoplankton. In C.D. Sandgren (ed). Growth and reproductive \n\tstrategies of freshwater phytoplankton, 388-433. Cambridge University Press, New York). \n\tVersion 1.2.0 also includes functions to query names using the algaebase online taxonomic \n\tdatabase (&lt;https://www.algaebase.org&gt;; &lt;doi:10.7872/crya.v35.iss2.2014.105&gt;).\n\tThe algaeClassify package is a product of the GEISHA (Global Evaluation of the Impacts of \n\tStorms on freshwater Habitat and Structure of phytoplankton Assemblages), funded by CESAB \n    (Centre for Synthesis and Analysis of Biodiversity) and the USGS John Wesley Powell Center for\n\tSynthesis and Analysis, with data and other support provided by members of GLEON \n\t(Global Lake Ecology Observation Network). This software is preliminary or provisional and\n\tis subject to revision. It is being provided to meet the need for timely best science. \n\tThe software has not received final approval by the U.S. Geological Survey (USGS). \n\tNo warranty, expressed or implied, is made by the USGS or the U.S. Government as to the \n\tfunctionality of the software and related material nor shall the fact of release constitute \n\tany such warranty. The software is provided on the condition that neither the USGS nor the U.S. \n\tGovernment shall be held liable for any damages resulting from the authorized or unauthorized \n\tuse of the software.","Can be used for paternity and maternity assignment and outperforms \n  conventional methods where closely related individuals occur in the pool of \n  possible parents. The method compares the genotypes of offspring with any \n  combination of potentials parents and scores the number of mismatches of these \n  individuals at bi-allelic genetic markers (e.g. Single Nucleotide Polymorphisms). \n  It elaborates on a prior exclusion method based on the Homozygous Opposite Test \n  (HOT; Huisman 2017 &lt;doi:10.1111/1755-0998.12665&gt;) by introducing the \n  additional exclusion criterion HIPHOP (Homozygous Identical Parents, \n  Heterozygous Offspring are Precluded; Cockburn et al., in revision). Potential \n  parents are excluded if they have more mismatches than can \n  be expected due to genotyping error and mutation, and thereby one can identify the \n  true genetic parents and detect situations where one (or both) of the true \n  parents is not sampled. Package 'hiphop' can deal with (a) the case where there \n  is contextual information about parentage of the mother (i.e. a female has been \n  seen to be involved in reproductive tasks such as nest building), but paternity \n  is unknown (e.g. due to promiscuity), (b) where both parents need to be assigned, \n  because there is no contextual information on which female laid eggs and which \n  male fertilized them (e.g. polygynandrous mating system where multiple females \n  and males deposit young in a common nest, or organisms with external fertilisation \n  that breed in aggregations).\n    For details: Cockburn, A., Penalba, J.V.,Jaccoud, D.,Kilian, A., Brouwer, L.,\n       Double, M.C., Margraf, N., Osmond, H.L., van de Pol, M. and Kruuk, L.E.B. \n       (in revision). HIPHOP: improved paternity assignment among close relatives \n       using a simple exclusion method for bi-allelic markers. \n       Molecular Ecology Resources, DOI to be added upon acceptance.","To estimate ecological stochasticity in community assembly. Understanding the community assembly mechanisms controlling biodiversity patterns is a central issue in ecology. Although it is generally accepted that both deterministic and stochastic processes play important roles in community assembly, quantifying their relative importance is challenging. The new index, normalized stochasticity ratio (NST), is to estimate ecological stochasticity, i.e. relative importance of stochastic processes, in community assembly. With functions in this package, NST can be calculated based on different similarity metrics and/or different null model algorithms, as well as some previous indexes, e.g. previous Stochasticity Ratio (ST), Standard Effect Size (SES), modified Raup-Crick metrics (RC). Functions for permutational test and bootstrapping analysis are also included. Previous ST is published by Zhou et al (2014) &lt;doi:10.1073/pnas.1324044111&gt;. NST is modified from ST by considering two alternative situations and normalizing the index to range from 0 to 1 (Ning et al 2019) &lt;doi:10.1073/pnas.1904623116&gt;. A modified version, MST, is a special case of NST, used in some recent or upcoming publications, e.g. Liang et al (2019) &lt;doi:10.1101/638908&gt;. SES is calculated as described in Kraft et al (2011) &lt;doi:10.1126/science.1208584&gt;. RC is calculated as reported by Chase et al (2011) &lt;doi:10.1890/es10-00117.1&gt; and Stegen et al (2013) &lt;doi:10.1038/ismej.2013.93&gt;.","Interpretation of time series data is affected by model choices. Different models can give different or even contradicting estimates of patterns, trends, and mechanisms for the same data--a limitation alleviated by the Bayesian estimator of abrupt change,seasonality, and trend (BEAST) of this package. BEAST seeks to improve time series decomposition by forgoing the \"single-best-model\" concept and embracing all competing models into the inference via a Bayesian model averaging scheme. It is a flexible tool to uncover abrupt changes (i.e., change-points), cyclic variations (e.g., seasonality), and nonlinear trends in time-series observations. BEAST not just tells when changes occur but also quantifies how likely the detected changes are true. It detects not just piecewise linear trends but also arbitrary nonlinear trends. BEAST is applicable to real-valued time series data of all kinds, be it for remote sensing, economics, climate sciences, ecology, and hydrology. Example applications include its use to identify regime shifts in ecological data, map forest disturbance and land degradation from satellite imagery, detect market trends in economic data, pinpoint anomaly and extreme events in climate data, and unravel system dynamics in biological data. Details on BEAST are reported in Zhao et al. (2019) &lt;doi:10.1016/j.rse.2019.04.034&gt;.","Analysis of species count data in ecology often requires normalization to an identical sample size. Rarefying (random subsampling without replacement), which is a popular method for normalization, has been widely criticized for its poor reproducibility and potential distortion of the community structure. In the context of microbiome count data, researchers explicitly advised against the use of rarefying. An alternative to rarefying is scaling with ranked subsampling (SRS). SRS consists of two steps. In the first step, the total counts for all OTUs (operational taxonomic units) or species in each sample are divided by a scaling factor chosen in such a way that the sum of the scaled counts Cscaled equals Cmin. In the second step, the non-integer Cscaled values are converted into integers by an algorithm that we dub ranked subsampling. The Cscaled value for each OTU or species is split into the integer part Cint  (Cint = floor(Cscaled)) and the fractional part Cfrac (Cfrac = Cscaled - Cints). Since the sum of Cint is smaller or equal to Cmin, additional  delta C = Cmin - the sum of Cint counts have to be added to the library to reach the total count of Cmin. This is achieved as follows. OTUs are ranked in the descending order of their Cfrac values. Beginning with the OTU of the highest rank, single count per OTU is added to the normalized library until the total number of added counts reaches delta C and the sum of all counts in the normalized library equals Cmin. When the lowest Cfrag involved in picking delta C counts is shared by several OTUs, the OTUs used for adding a single count to the library are selected in the order of their Cint values. This selection minimizes the effect of normalization on the relative frequencies of OTUs. OTUs with identical Cfrag as well as Cint are sampled randomly without replacement. See Beule &amp; Karlovsky (2020) &lt;doi:10.7717/peerj.9593&gt; for details.","Provides a flexible simulation tool for phylogenetic trees under a general model for speciation and extinction. Trees with a user-specified number of extant tips, or a user-specified stem age are simulated. It is possible to assume any probability distribution for the waiting time until speciation and extinction. Furthermore, the waiting times to speciation / extinction may be scaled in different parts of the tree, meaning we can simulate trees with clade-dependent diversification processes. At a speciation event, one species splits into two. We allow for two different modes at these splits: (i) symmetric, where for every speciation event new waiting times until speciation and extinction are drawn for both daughter lineages; and (ii) asymmetric, where a speciation event results in one species with new waiting times, and another that carries the extinction time and age of its ancestor. The symmetric mode can be seen as an vicariant or allopatric process where divided populations suffer equal evolutionary forces while the asymmetric mode could be seen as a peripatric speciation where a mother lineage continues to exist.  Reference: O. Hagen and T. Stadler (2017). TreeSimGM: Simulating phylogenetic trees under general Bellman Harris models with lineage-specific shifts of speciation and extinction in R. Methods in Ecology and Evolution. &lt;doi:10.1111/2041-210X.12917&gt;.","MetaboAnalystR 3.0 contains the R functions and libraries underlying the popular MetaboAnalyst web server, including metabolomic data analysis, visualization, and functional interpretation. The package is synchronized with the MetaboAnalyst web server. After installing and loading the package, users will be able to reproduce the same results from their local computers using the corresponding R command history downloaded from MetaboAnalyst web site, thereby achieving maximum flexibility and reproducibility.\n\nThe version 3.0 aims to improve the current global metabolomics workflow by implementing a fast parameter optimization algorithm for peak picking, and automated identification of the most suitable method for batch effect correction from 12 well-established approaches. In addition, more support for functional interpretation directly from m/z peaks via mummichog2 (PMID: 23861661), and a new pathway-based method to integrate multi'omics data has been added. To demonstrate this new functionality, we provide the \"MetaboAnalystR 3.0 Workflow: Towards an Optimized Workflow for Global Metabolomics\" vignette, available here as a PDF. In this vignette, we perform end-to-end metabolomics data analysis on the full batch of clinical IBD samples.","See the Article in Methods in Ecology and Evolution &lt;https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12600&gt; for an overview of the package. The four package vignettes provide examples for the entire workflow.","Utilities for spatial-temporal analysis and visualization of animal (e.g. cattle)\ntracking data. The core feature is a 'shiny' web application for customized processing\nof GPS logs, including features for data augmentation (e.g. elevation lookup), data selection,\nexport, plotting, and statistical summaries. A data validation application allows for side-by-side\ncomparison via time series plots and extreme value detection described by\nJ.P. van Brakel &lt;https://stackoverflow.com/questions/22583391/peak-signal-detection-inrealtime-timeseries-data/&gt;.","Functions for species distribution modeling, that is, predicting entire geographic distributions form occurrences at a number of sites and the environment at these sites","Wallace is a modular platform for reproducible modeling of species niches and distributions, written in R. The application guides users through a complete analysis, from the acquisition of data to visualizing model predictions on an interactive map, thus bundling complex workflows into a single, streamlined interface.","At rOpenSci, we have been writing R packages to interact with many sources of species occurrence data, including GBIF, Vertnet, BISON, iNaturalist, the Berkeley ecoengine, and eBird. Other databases are out there as well, which we can pull in. spocc is an R package to query and collect species occurrence data from many sources. The goal is to to create a seamless search experience across data sources, as well as creating unified outputs across data sources.\n\nspocc currently interfaces with nine major biodiversity repositories\n\nGlobal Biodiversity Information Facility (GBIF) (via rgbif) GBIF is a government funded open data repository with several partner organizations with the express goal of providing access to data on Earth's biodiversity. The data are made available by a network of member nodes, coordinating information from various participant organizations and government agencies.\n\nBerkeley Ecoengine (via ecoengine) The ecoengine is an open API built by the Berkeley Initiative for Global Change Biology. The repository provides access to over 3 million specimens from various Berkeley natural history museums. These data span more than a century and provide access to georeferenced specimens, species checklists, photographs, vegetation surveys and resurveys and a variety of measurements from environmental sensors located at reserves across University of California's natural reserve system.\n\niNaturalist iNaturalist provides access to crowd sourced citizen science data on species observations.\n\nVertNet (via rvertnet) Similar to rgbif, ecoengine, and rbison (see below), VertNet provides access to more than 80 million vertebrate records spanning a large number of institutions and museums primarly covering four major disciplines (mammology, herpetology, ornithology, and icthyology).\n\nBiodiversity Information Serving Our Nation (via rbison) Built by the US Geological Survey's core science analytic team, BISON is a portal that provides access to species occurrence data from several participating institutions.\n\neBird (via rebird) ebird is a database developed and maintained by the Cornell Lab of Ornithology and the National Audubon Society. It provides real-time access to checklist data, data on bird abundance and distribution, and communtiy reports from birders.\n\niDigBio (via ridigbio) iDigBio facilitates the digitization of biological and paleobiological specimens and their associated data, and houses specimen data, as well as providing their specimen data via RESTful web services.\n\nOBIS OBIS (Ocean Biogeographic Information System) allows users to search marine species datasets from all of the world's oceans.\n\nAtlas of Living Australia ALA (Atlas of Living Australia) contains information on all the known species in Australia aggregated from a wide range of data providers: museums, herbaria, community groups, government departments, individuals and universities; it contains more than 50 million occurrence records.\n\nThe inspiration for this comes from users requesting a more seamless experience across data sources, and from our work on a similar package for taxonomy data (taxize).\n\nBEWARE: In cases where you request data from multiple providers, especially when including GBIF, there could be duplicate records since many providers' data eventually ends up with GBIF. See ?spocc_duplicates, after installation, for more.","Functions for fitting various models to capture-recapture data\nincluding mixed-effects Cormack-Jolly-Seber(CJS) and multistate models and\nthe multi-variate state model structure for survival\nestimation and POPAN structured Jolly-Seber models for abundance estimation.\nThere are also Hidden Markov model (HMM) implementations of CJS and multistate\nmodels with and without state uncertainty and a simulation capability for HMM\nmodels.","Fits hierarchical models of animal abundance and occurrence to data collected using survey methods such as point counts, site occupancy sampling, distance sampling, removal sampling, and double observer sampling. Parameters governing the state and observation processes can be modeled as functions of covariates.","An interface to the software package MARK that constructs input files for MARK and extracts the output. MARK was developed by Gary White\nand is freely available at &lt;http://www.phidot.org/software/mark/downloads/&gt;\nbut is not open source.","Functions for identifying, fitting, and applying continuous-space, continuoustime stochastic movement models to animal tracking data.\nThe package is described in Calabrese et al (2016) &lt;doi:10.1111/2041-\n210X.12559&gt;, with models and methods based on those introduced in\nFleming &amp; Calabrese et al (2014) &lt;doi:10.1086/675504&gt;,\nFleming et al (2014) &lt;doi:10.1111/2041-210X.12176&gt;,\nFleming et al (2015) &lt;doi:10.1103/PhysRevE.91.032107&gt;,\nFleming et al (2015) &lt;doi:10.1890/14-2010.1&gt;,\nFleming et al (2016) &lt;doi:10.1890/15-1607&gt;,\nPron &amp; Fleming et al (2016) &lt;doi:10.1186/s40462-016-0084-7&gt;,\nFleming &amp; Calabrese (2017) &lt;doi:10.1111/2041-210X.12673&gt;,\nPron et al (2017) &lt;doi:10.1002/ecm.1260&gt;,\nFleming et al (2017) &lt;doi:10.1016/j.ecoinf.2017.04.008&gt;,\nFleming et al (2018) &lt;doi:10.1002/eap.1704&gt;,\nWinner &amp; Noonan et al (2018) &lt;doi:10.1111/2041-210X.13027&gt;,\nFleming et al (2019) &lt;doi:10.1111/2041-210X.13270&gt;,\nand\nNoonan &amp; Fleming et al (2019) &lt;doi:10.1186/s40462-019-0177-1&gt;.","Provides tools for animal movement modelling using hidden Markov\nmodels. These include processing of tracking data, fitting hidden Markov models\nto movement data, visualization of data and fitted model, decoding of the state\nprocess...\n","A collection of tools for the analysis of habitat selection by animals.","landscapemetrics reimplements the most common metrics from\nFRAGSTATS (&lt;https://www.umass.edu/landeco/research/fragstats/fragstats.html&gt;) and new ones from the current literature on landscape metrics.This package supports raster spatial objects and takes\nRasterLayer, RasterStacks, RasterBricks or lists of RasterLayer from the raster package as input arguments. It further provides utility functions\nto visualize patches, select metrics and building blocks to develop new\nmetrics.","Manage and analyze animal movement data. The functionality of 'amt' includes methods to calculate track statistics (e.g. step lengths, speed, or turning angles), prepare data for fitting habitat selection analyses (resource selection functions and step-selection functions &lt;doi:10.1890/04-\n0953&gt; and integrated step-selection functions &lt;doi:10.1111/2041-210X.12528&gt;), and simulation of space-use from fitted step-selection functions &lt;doi:10.1002/ecs2.1771&gt;.","A collection of tools for the estimation of animals home range.","Extended tools for analyzing telemetry data using generalized hidden Markov models. Features of momentuHMM (pronounced ``momentum'') include data pre-processing and visualization, fitting HMMs to location and auxiliary biotelemetry or environmental data, biased and correlated random walk movement models, hierarchical HMMs, multiple imputation for incorporating location measurement error and missing data, user-specified design matrices and constraints for covariate modelling of parameters, random effects, decoding of the state process, visualization of fitted models, model checking and selection, and simulation. See McClintock and Michelot (2018) &lt;doi:10.1111/2041-210X.12995&gt;","Computes revisitation metrics for trajectory data, such as the number of revisitations for each location as well as the time spent for that visit and the time since the previous visit. Also includes functions to plot data.\n","MLWIC2 can be used to automatically classify camera trap images or to train new models for image classification, it contains two pre-trained models: the species_model identifies 58 species and empty images, and the empty_animal model distinguishes between images with animals and those that are empty. MLWIC2 also contains Shiny apps for running the functions. These can be accessed using runShiny. In the steps below, you can see Shiny options for some steps. This indicates that you can run these steps with Shiny apps by running the function provied. Note that when you are using Shiny apps to select directories and files, you can only navigate using the top part half of the screen and you must scroll to the bottom of the window to find the Select button.","Tools for multivariate data analysis. Several methods are provided for the analy\u0002sis (i.e., ordination) of one-table (e.g., principal component analysis, correspondence analysis), two-table (e.g., coinertia analysis, redundancy analysis), three-table (e.g., RLQ analysis) and K-table (e.g., STATIS, multiple coinertia analysis). The philosophy of the package is described in Dray and Dufour (2007) &lt;doi:10.18637/jss.v022.i04&gt;.","This function conducts variation partitioning and hierarchical partitioning to calculate the unique, shared (referred as to 'common') and independent contributions of each predictor (or matrix) to explained variation (R-squared and adjusted R-squared) on canonical analysis (RDA,CCA and db-RDA), applying the algorithm of Lai J.,Zou Y., Zhang J.,Peres-Neto P.(2021).rdacca.hp: an R package for generalizing hierarchical and variation partitioning in multiple regression and canonical analysis. bioRxiv 2021.03.09.434308 &lt;DOI:10.1101/2021.03.09.434308&gt;..","A collection of measures for measuring ecological diversity.\r\n \r\n  Ecological diversity comes in two flavors: alpha diversity measures the\r\n \r\n  diversity within a single site or sample, and beta diversity measures the\r\n \r\n  diversity across two sites or samples. This package overlaps considerably\r\n \r\n  with other R packages such as 'vegan', 'gUniFrac', 'betapart', and 'fossil'.\r\n \r\n  We also include a wide range of functions that are implemented in software\r\n \r\n  outside the R ecosystem, such as 'scipy', 'Mothur', and 'scikit-bio'. The\r\n \r\n  implementations here are designed to be basic and clear to the reader.","The R package 'adagio' will provide methods and algorithms for\r\n \r\n  discrete optimization, e.g. knapsack and subset sum procedures,\r\n \r\n derivative-free Nelder-Mead and Hooke-Jeeves minimization, and\r\n \r\n some (evolutionary) global optimization functions.","A collection of tools for the analysis of habitat selection.","A collection of tools for the analysis of animal movements.","A collection of tools to deal with raster maps.","Functions, data sets and examples for the calculation of various indices of biodiversity including species, functional and phylogenetic diversity. Part of the indices are expressed in terms of equivalent numbers of species. The package also provides ways to partition biodiversity across spatial or temporal scales (alpha, beta, gamma diversities). In addition to the quantification of biodiversity, ordination approaches are available which rely on diversity indices and allow the detailed identification of species, functional or phylogenetic differences between communities.","Computation of A (pedigree), G (genomic-base), and H (A corrected\r\n    by G) relationship matrices for diploid and autopolyploid species. Several methods\r\n    are implemented considering additive and non-additive models.","Functions to implement model selection and multimodel inference based on Akaike's information criterion (AIC) and the second-order AIC (AICc), as well as their quasi-likelihood counterparts (QAIC, QAICc) from various model object classes. The package implements classic model averaging for a given parameter of interest or predicted values, as well as a shrinkage version of model averaging parameter estimates or effect sizes. The package includes diagnostics and goodness-of-fit statistics for certain model types including those of 'unmarkedFit' classes estimating demographic parameters after accounting for imperfect detection probabilities. Some functions also allow the creation of model selection tables for Bayesian models of the 'bugs', 'rjags', and 'jagsUI' classes. Functions also implement model selection using BIC. Objects following model selection and multimodel inference can be formatted to LaTeX using 'xtable' methods included in the package.","2D and 3D animal tracking data can be used to reconstruct tracks through time/space with correction based on known positions. 3D visualization of animal position and attitude.","Animation of observed trajectories using spline-based interpolation (see for example, Buderman, F. E., Hooten, M. B., Ivan, J. S. and Shenk, T. M. (2016), &lt;doi:10.1111/2041-210X.12465&gt; \"A functional model for characterizing long-distance movement behaviour\". Methods Ecol Evol). Intended to be used exploratory data analysis, and perhaps for preparation of presentations.","Simulation and analysis\r\n        of phylogenetic tree topologies using statistical indices. It\r\n        is a companion library of the 'ape' package. It provides\r\n        additional functions for reading, plotting, manipulating\r\n        phylogenetic trees. It also offers convenient web-access to\r\n        public databases, and enables testing null models of\r\n        macroevolution using corrected test statistics.  Trees of class\r\n        \"phylo\" (from 'ape' package) can be converted easily.\r\n        Implements methods described in Bortolussi et al. (2005)\r\n        &lt;doi:10.1093/bioinformatics/bti798&gt; and Maliet et al. (2017) \r\n        &lt;doi:10.1101/224295&gt;.","Fits Modern Analogue Technique and Weighted Averaging transfer \r\n \r\n    function models for prediction of environmental data from species \r\n \r\n  data, and related methods used in palaeoecology.","Manage keys, certificates, secrets, and storage accounts in Microsoft's 'Key Vault' service: &lt;https://azure.microsoft.com/services/key-vault&gt;. Provides facilities to store and retrieve secrets, use keys to encrypt, decrypt, sign and verify data, and manage certificates. Integrates with the 'AzureAuth' package to enable authentication with a certificate, and with the 'openssl' package for importing and exporting cryptographic objects. Part of the 'AzureR' family of packages.","Can be used for simulation of organisms living in\r\n \r\n  communities (Bauer and Zimmermann (2017) &lt;doi:10.1371/journal.pcbi.1005544&gt;). \r\n \r\n  Each organism is represented individually and genome scale\r\n \r\n  metabolic models determine the uptake and release of compounds. Biological\r\n \r\n  processes such as movement, diffusion, chemotaxis and kinetics are available\r\n \r\n  along with data analysis techniques.","Generates version 2 and version 4 request signatures for Amazon Web Services ('AWS') &lt;https://aws.amazon.com/&gt; Application Programming Interfaces ('APIs') and provides a mechanism for retrieving credentials from environment variables, 'AWS' credentials files, and 'EC2' instance metadata. For use on 'EC2' instances, users will need to install the suggested package 'aws.ec2metadata' &lt;https://cran.r-project.org/package=aws.ec2metadata&gt;.","Facilitates the evolutionary analysis and structure conservation study of specified amino acids in proteins.","Includes algorithms to assess alpha and beta\r\n    diversity in all their dimensions (taxon, phylogenetic and functional\r\n    diversity), whether communities are completely sampled or not. It allows\r\n    performing a number of analyses based on either species identities or\r\n    phylogenetic/functional trees or functional kernel n-dimensional hypervolumes\r\n    depicting species relationships.","The model provides an empirical estimate of a movement\r\n \r\n  path using discrete location data obtained at relatively short\r\n \r\n  time intervals.","Provides functions for analyzing and visualizing complex\r\n \r\n  macroevolutionary dynamics on phylogenetic trees. It is a companion\r\n \r\n  package to the command line program BAMM (Bayesian Analysis of\r\n \r\n  Macroevolutionary Mixtures) and is entirely oriented towards the analysis,\r\n \r\n  interpretation, and visualization of evolutionary rates. Functionality\r\n \r\n  includes visualization of rate shifts on phylogenies, estimating\r\n \r\n  evolutionary rates through time, comparing posterior distributions of\r\n \r\n  evolutionary rates across clades, comparing diversification models using\r\n \r\n  Bayes factors, and more.","Analysis tool for evaluating benthic multimetric indices (BENMMIs). \r\n    It generates reproducible reports on the analysis of benthic data, e.g., \r\n    validation and correction of species names, sample pooling, automatic\r\n    conversion of genus to species names, outlier detection, benthic \r\n    indicator calculation, optimization of single and \r\n    multimetric indicators against a pressure gradient, and spatial aggregation\r\n    of benthic indicators. One of its use cases was the development of a common \r\n    benthic indicator for &lt;https://www.ospar.org&gt; (publication accepted by \r\n    Ecological Indicators). See Van Loon et al. (2018) &lt;doi:10.1016/j.ecolind.2017.09.029&gt; for details.","A molecular genetics tool that processes binary data from fragment analysis, such as inter-simple sequence repeats (ISSRs) and amplified fragment length polymorphism (AFLP). It consolidates replicate sample pairs, outputs summary statistics, and produces hierarchical clustering trees and nMDS plots. This package was developed from the M.Sc. thesis entitled \"A genetic analysis of the species and intraspecific lineages of Dactylopius Costa (Hemiptera:Dactylopiidae)\" (van Steenderen, 2019, Rhodes University, Department of Zoology and Entomology, Center for Biological Control (CBC) &lt;https://www.ru.ac.za/centreforbiologicalcontrol/&gt;, Grahamstown, South Africa), &lt;doi:10.13140/RG.2.2.28470.86083&gt;. The GUI version of this package is available on the R Shiny online server at: &lt;https://clarkevansteenderen.shinyapps.io/BINMAT/&gt; , or it is accessible via GitHub by typing: shiny::runGitHub(\"BinMat\", \"CJMvS\") into the console in R. Please see the vignette supplied with the package for a worked example, and detailed explanations of functions.","Functions for error detection and correction in point data quality datasets that are used in species distribution modelling. Includes functions for parsing and converting coordinates into decimal degrees from various formats.","Functions for species distribution modeling, calibration and evaluation, \r\n  ensemble of models, ensemble forecasting and visualization. The package permits to run\r\n  consistently up to 10 single models on a presence/absences (resp presences/pseudo-absences)\r\n  dataset and to combine them in ensemble models and ensemble projections. Some bench of other \r\n  evaluation and visualisation tools are also available within the package.","Contains the JIVE (joint inter and intra-specific model of variance evolution) model and other Bayesian models aimed at understanding trait evolution. The goal of the package is to join phylogenetic comparative models (PCM) that tend to integrate various type of data (individual observations, environmental data, fossil data) into a hierarchical Bayesian framework. It contains various PCMs as well as functions to join those models into a hierarchical Bayesian framework in a flexible and user friendly way. It contains various Markov chain Monte-Carlo (MCMC) algorithms, methods for model comparison and many plotting function for pre- and post-processing data visualization. Finally, this package integrates functions allowing bridges between 'R' and the 'BEAST2' implementations of PCMs. Kostikova A, Silvestro D, Pearman PB, Salamin N (2016) &lt;doi:10.1093/sysbio/syw010&gt;. Gaboriau T, Mendes FK, Joly S, Silvestro D, Salamin N (in prep).","Provides a set of functions to fit general macroevolutionary models for continuous traits evolving in adaptive landscapes of any shape. This package implements the Fokker-Planck-Kolmogorov model (FPK), in which the trait evolves under random diffusion but is also subject to a force that pulls it towards specific values - this force can be of any shape. FPK has a version in which hard reflective bounds exist at the extremes of the trait interval: this second model is called BBMV.","Some elementary matrix algebra tools are implemented to manage\r\n \r\n  block matrices or partitioned matrix, i.e. \"matrix of matrices\"\r\n \r\n  (http://en.wikipedia.org/wiki/Block_matrix). The block matrix is here\r\n \r\n  defined as a new S3 object. In this package, some methods for \"matrix\"\r\n \r\n  object are rewritten for \"blockmatrix\" object. New methods are implemented.\r\n \r\n  This package was created to solve equation systems with block matrices for\r\n \r\n  the analysis of environmental vector time series .\r\n \r\n  Bugs/comments/questions/collaboration of any kind are warmly welcomed.","Fork-safe, raw access to the 'Amazon Web Services' ('AWS') 'SDK' via the 'boto3' 'Python' module, and convenient helper functions to query the 'Simple Storage Service' ('S3') and 'Key Management Service' ('KMS'), partial support for 'IAM', the 'Systems Manager Parameter Store' and 'Secrets Manager'.","Genomic selection and prediction models with the capacity to use multiple traits and environments, through ready-to-use Bayesian models. It consists a group of functions \r\n \r\n  that help to create regression models for some genomic models proposed by Montesinos-Lpez, et al. (2016) &lt;doi:10.1534/g3.116.032359&gt;\r\n \r\n  also in Montesinos-Lpez et al. (2018) &lt;doi:10.1534/g3.118.200728&gt; and Montesinos-Lpez et al. (2018) &lt;doi:10.2134/agronj2018.06.0362&gt;.","Tools to fit Bayesian state-space models to animal tracking data. Models are provided for location \r\n \r\n  filtering, location filtering and behavioural state estimation, and their hierarchical versions. \r\n \r\n  The models are primarily intended for fitting to ARGOS satellite tracking data but options exist to fit \r\n \r\n  to other tracking data types. For Global Positioning System data, consider the 'moveHMM' package. \r\n \r\n  Simplified Markov Chain Monte Carlo convergence diagnostic plotting is provided but users are encouraged \r\n \r\n  to explore tools available in packages such as 'coda' and 'boa'.","Model-based clustering for phylogeographic data comprising mtDNA sequences and geographical locations along with optional environmental characteristics, aiming to identify migration events that led to homogeneous population clusters. The package vignette, I. Manolopoulou, A. Hille, B. C. Emerson (2020) &lt;doi:10.18637/jss.v092.i03&gt;, provides detailed descriptions of the package.","R graphics device using cairographics library that can be used to create high-quality vector (PDF, PostScript and SVG) and bitmap output (PNG,JPEG,TIFF), and high-quality rendering in displays (X11 and Win32). Since it uses the same back-end for all output, copying across formats is WYSIWYG. Files are created without the dependence on X11 or other external programs. This device supports alpha channel (semi-transparent drawing) and resulting images can contain transparent and semi-transparent regions. It is ideal for use in server environments (file output) and as a replacement for other devices that don't have Cairo's capabilities such as alpha support or anti-aliasing. Backends are modular such that any subset of backends is supported.","Compositional analysis of differentially expressed proteins in\r\n \r\n  cancer and cell culture proteomics experiments. The data include lists of up-\r\n \r\n  and down-regulated proteins in different cancer types (breast, colorectal,\r\n \r\n  liver, lung, pancreatic, prostate) and laboratory conditions (hypoxia,\r\n \r\n  hyperosmotic stress, high glucose, 3D cell culture, and proteins secreted in\r\n \r\n  hypoxia), together with amino acid compositions computed for protein sequences\r\n \r\n  obtained from UniProt. Functions are provided to calculate compositional metrics\r\n \r\n  including protein length, carbon oxidation state, and stoichiometric hydration\r\n \r\n  state. In addition, phylostrata (evolutionary ages) of protein-coding genes are\r\n \r\n  compiled using data from Liebeskind et al. (2016) &lt;doi:10.1093/gbe/evw113&gt; or\r\n \r\n  Trigos et al. (2017) &lt;doi:10.1073/pnas.1617743114&gt;. The vignettes contain\r\n \r\n  plots of compositional differences, phylostrata for human proteins, and\r\n \r\n  references for all datasets.","We provide a toolbox to fit a continuous-time fractionally integrated ARMA process (CARFIMA) on univariate and irregularly spaced time series data via both frequentist and Bayesian machinery. A general-order CARFIMA(p, H, q) model for p&gt;q is specified in Tsai and Chan (2005) &lt;doi:10.1111/j.1467-9868.2005.00522.x&gt; and it involves p+q+2 unknown model parameters, i.e., p AR parameters, q MA parameters, Hurst parameter H, and process uncertainty (standard deviation) sigma. Also, the model can account for heteroscedastic measurement errors, if the information about measurement error standard deviations is known. The package produces their maximum likelihood estimates and asymptotic uncertainties using a global optimizer called the differential evolution algorithm. It also produces posterior samples of the model parameters via Metropolis-Hastings within a Gibbs sampler equipped with adaptive Markov chain Monte Carlo. These fitting procedures, however, may produce numerical errors if p&gt;2. The toolbox also contains a function to simulate discrete time series data from CARFIMA(p, H, q) process given the model parameters and observation times.","Efficient phylogenetic analyses on massive phylogenies comprising up to millions of tips. Functions include pruning, rerooting, calculation of most-recent common ancestors, calculating distances from the tree root and calculating pairwise distances. Calculation of phylogenetic signal and mean trait depth (trait conservatism), ancestral state reconstruction and hidden character prediction of discrete characters, simulating and fitting models of trait evolution, fitting and simulating diversification models, dating trees, comparing trees, and reading/writing trees in Newick format. Citation: Louca, Stilianos and Doebeli, Michael (2017) &lt;doi:10.1093/bioinformatics/btx701&gt;.","Classification is based on the recently developed phylogenetic\r\n    approach by Assis and Bachtrog (2013). The method classifies the\r\n    evolutionary mechanisms retaining pairs of duplicate genes (conservation,\r\n    neofunctionalization, subfunctionalization, or specialization) by comparing gene\r\n    expression profiles of duplicate genes in one species to those of their single-\r\n    copy ancestral genes in a sister species.","Provides methodology to analyze cells that move in a two- or three-dimensional\r\n \r\n space. Available measures include displacement, confinement ratio, autocorrelation,\r\n \r\n straightness, turning angle, and fractal dimension. Measures can be applied to entire tracks,\r\n \r\n steps, or subtracks with varying length. While the methodology has been developed for\r\n \r\n cell trajectory analysis, it is applicable to anything that moves including animals,\r\n \r\n people, or vehicles.\r\n \r\n Some of the methodology implemented in this packages was described by: \r\n \r\n Beauchemin, Dixit, and Perelson (2007) &lt;doi:10.4049/jimmunol.178.9.5505&gt;, \r\n \r\n Beltman, Maree, and de Boer (2009) &lt;doi:10.1038/nri2638&gt;,\r\n \r\n Gneiting and Schlather (2004) &lt;doi:10.1137/S0036144501394387&gt;,\r\n \r\n Mokhtari, Mech, Zitzmann, Hasenberg, Gunzer, and Figge (2013) &lt;doi:10.1371/journal.pone.0080808&gt;,\r\n \r\n Moreau, Lemaitre, Terriac, Azar, Piel, Lennon-Dumenil, and Bousso (2012) &lt;doi:10.1016/j.immuni.2012.05.014&gt;,\r\n \r\n Textor, Peixoto, Henrickson, Sinn, von Andrian, and Westermann (2011) &lt;doi:10.1073/pnas.1102288108&gt;,\r\n \r\n  Textor, Sinn, and de Boer (2013) &lt;doi:10.1186/1471-2105-14-S6-S10&gt;,\r\n \r\n  Textor, Henrickson, Mandl, von Andrian, Westermann, de Boer, and Beltman (2014) &lt;doi:10.1371/journal.pcbi.1003752&gt;.","Facilitates searching, download and plotting of Water Framework \r\n \r\n  Directive (WFD) reporting data for all waterbodies within the UK Environment \r\n \r\n  Agency area. The types of data that can be downloaded are: WFD status \r\n \r\n  classification data, Reasons for Not Achieving Good (RNAG) status, \r\n \r\n  objectives set for waterbodies, measures put in place to improve water \r\n \r\n  quality and details of associated protected areas. The site accessed is \r\n \r\n  &lt;https://environment.data.gov.uk/catchment-planning/&gt;. The data are made \r\n \r\n  available under the Open Government Licence v3.0 \r\n \r\n  &lt;https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/&gt;.\r\n \r\n  This package has been peer-reviewed by rOpenSci (v. 0.4.0).","An integrated set of tools for thermodynamic calculations in\r\n  aqueous geochemistry and geobiochemistry. Functions are provided for writing\r\n  balanced reactions to form species from user-selected basis species and for\r\n  calculating the standard molal properties of species and reactions, including\r\n  the standard Gibbs energy and equilibrium constant. Calculations of the\r\n  non-equilibrium chemical affinity and equilibrium chemical activity of species\r\n  can be portrayed on diagrams as a function of temperature, pressure, or\r\n  activity of basis species; in two dimensions, this gives a maximum affinity or\r\n  predominance diagram. The diagrams have formatted chemical formulas and axis\r\n  labels, and water stability limits can be added to Eh-pH, oxygen fugacity-\r\n  temperature, and other diagrams with a redox variable. The package has been\r\n  developed to handle common calculations in aqueous geochemistry, such as\r\n  solubility due to complexation of metal ions, mineral buffers of redox or pH,\r\n  and changing the basis species across a diagram (\"mosaic diagrams\"). CHNOSZ\r\n  also has unique capabilities for comparing the compositional and thermodynamic\r\n  properties of different proteins.","Comprehensive data analysis software, and the name \"cg\" stands for \"compare groups.\" Its genesis and evolution are driven by common needs to compare administrations, conditions, etc. in medicine research and development. The current version provides comparisons of unpaired samples, i.e. a linear model with one factor of at least two levels. It also provides comparisons of two paired samples. Good data graphs, modern statistical methods, and useful displays of results are emphasized.","Circular layout is an efficient way for the visualization of huge \r\n \r\n  amounts of information. Here this package provides an implementation \r\n \r\n  of circular layout generation in R as well as an enhancement of available \r\n \r\n  software. The flexibility of the package is based on the usage of low-level \r\n \r\n  graphics functions such that self-defined high-level graphics can be easily \r\n \r\n  implemented by users for specific purposes. Together with the seamless \r\n \r\n  connection between the powerful computational and visual environment in R, \r\n \r\n  it gives users more convenience and freedom to design figures for \r\n \r\n  better understanding complex patterns behind multiple dimensional data. \r\n \r\n  The package is described in Gu et al. 2014 &lt;doi:10.1093/bioinformatics/btu393&gt;.","Measures morphological diversity from discrete character data and\r\n \r\n  estimates evolutionary tempo on phylogenetic trees. Imports morphological\r\n \r\n  data from #NEXUS (Maddison et al. (1997) &lt;doi:10.1093/sysbio/46.4.590&gt;)\r\n \r\n  format with ReadMorphNexus(), and writes to both #NEXUS and TNT format \r\n \r\n  (Goloboff et al. (2008) &lt;doi:10.1111/j.1096-0031.2008.00217.x&gt;). Main\r\n \r\n  functions are DiscreteCharacterRate(), which implements likelihood ratio\r\n \r\n  tests for discrete character rates introduced across Lloyd et al. (2012)\r\n \r\n  &lt;doi:10.1111/j.1558-5646.2011.01460.x&gt;, Brusatte et al. (2014)\r\n \r\n  &lt;doi:10.1016/j.cub.2014.08.034&gt;, Close et al. (2015)\r\n \r\n  &lt;doi:10.1016/j.cub.2015.06.047&gt;, and Lloyd (2016) &lt;doi:10.1111/bij.12746&gt;,\r\n \r\n  and MorphDistMatrix(), which implements multiple discrete character\r\n \r\n  distance metrics from Gower (1971) &lt;doi:10.2307/2528823&gt;, Wills (1998)\r\n \r\n  &lt;doi:10.1006/bijl.1998.0255&gt;, Lloyd (2016) &lt;doi:10.1111/bij.12746&gt;, and\r\n \r\n  Hopkins and St John (2018) &lt;doi:10.1098/rspb.2018.1784&gt;. Multiple functions\r\n \r\n  implement various morphospace plots: ChronoPhyloMorphospacePlot() implements\r\n \r\n  Sakamoto and Ruta (2012) &lt;doi:10.1371/journal.pone.0039752&gt;,\r\n \r\n  MorphospacePlot() implements Wills et al. (1994)\r\n \r\n  &lt;doi:10.1017/S009483730001263X&gt;, PlotCharacterChanges() implements Wang and\r\n \r\n  Lloyd (2016) &lt;doi:10.1098/rspb.2016.0214&gt;, and StackPlot() implements Foote\r\n \r\n  (1993) &lt;doi:10.1017/S0094837300015864&gt;. Other functions include\r\n \r\n  SafeTaxonomicReduction(), which implements Wilkinson (1995)\r\n \r\n  &lt;doi:10.1093/sysbio/44.4.501&gt;, and DolloSCM() implements the Dollo\r\n \r\n  stochastic character mapping of Tarver et al. (2018)\r\n \r\n  &lt;doi:10.1093/gbe/evy096&gt;.","This package provides the estimation of an index measuring\r\n \r\n  the clinical utility of DNA testing in the context of\r\n \r\n  gene-environment interactions on a disease. The corresponding\r\n \r\n  gene-environment interaction effect on the additive scale can\r\n \r\n  also be obtained. The estimation is based on case-control or\r\n \r\n  cohort data. The method was developed by Nguyen et al. 2013.","Fits predictive and symmetric co-correspondence analysis (CoCA) models to relate one data matrix to another data matrix. More specifically, CoCA maximises the weighted covariance between the weighted averaged species scores of one community and the weighted averaged species scores of another community. CoCA attempts to find patterns that are common to both communities.","species coexistence modeling under asymmetric dispersal\r\n        and fluctuating source-sink dynamics;testing the proportion of\r\n        coexistence scenarios driven by neutral and niche processes","Color and visualize wildlife distributions in\r\n    space-time using raster data. In addition to enabling display of\r\n    sequential change in distributions through the use of small multiples,\r\n    'colorist' provides functions for extracting several features of\r\n    interest from a sequence of distributions and for visualizing those\r\n    features using HCL (hue-chroma-luminance) color palettes. Resulting\r\n    maps allow for \"fair\" visual comparison of intensity values (e.g.,\r\n    occurrence, abundance, or density) across space and time and can be\r\n    used to address questions about where, when, and how consistently a\r\n    species, group, or individual is likely to be found.","Coalescent simulators can rapidly simulate biological sequences\r\n \r\n  evolving according to a given model of evolution.\r\n \r\n  You can use this package to specify such models, to conduct the simulations\r\n \r\n  and to calculate additional statistics from the results.\r\n \r\n  It relies on existing simulators for doing the simulation, and currently\r\n \r\n  supports the programs 'ms', 'msms' and 'scrm'. It also supports finite-sites\r\n \r\n  mutation models by combining the simulators with the program 'seq-gen'.","Manage configuration values across multiple environments (e.g.\r\n \r\n  development, test, production). Read values using a function that determines\r\n \r\n  the current environment and returns the appropriate value.","Fit continuous-time correlated random walk models with time indexed\r\n \r\n  covariates to animal telemetry data. The model is fit using the Kalman-filter on\r\n \r\n  a state space version of the continuous-time stochastic movement process.","Software to facilitates taking movement data in xyt format and pairing it with raster covariates within a continuous time Markov chain (CTMC) framework. As described in Hanks et al. (2015) &lt;DOI:10.1214/14-AOAS803&gt; , this allows flexible modeling of movement in response to covariates (or covariate gradients) with model fitting possible within a Poisson GLM framework.","Collection of generic 'R' scripts which enable you to use existing 'R' routines in 'Cornerstone'.\r\n \r\n \r\n \r\n  The desktop application 'Cornerstone' (&lt;https://www.camline.com/en/products/cornerstone/cornerstone-core.html&gt;)\r\n \r\n  is a data analysis software provided by 'camLine' that empowers engineering teams to find solutions even faster.\r\n \r\n  The engineers incorporate intensified hands-on statistics into their projects.\r\n \r\n  They benefit from an intuitive and uniquely designed graphical Workmap concept: you design\r\n \r\n  experiments (DoE) and explore data, analyze dependencies, and find answers you can act upon,\r\n \r\n  immediately, interactively, and without any programming.\r\n \r\n \r\n \r\n  While 'Cornerstone's' interface to the statistical programming language 'R' has been available\r\n \r\n  since version 6.0, the latest interface with 'R' is even much more efficient.\r\n \r\n  'Cornerstone' release 7.1.1 allows you to integrate user defined 'R' packages directly into the\r\n \r\n  standard 'Cornerstone' GUI.\r\n \r\n  Your engineering team stays in 'Cornerstone's' graphical working environment and can apply 'R'\r\n \r\n  routines, immediately and without the need to deal with programming code.\r\n \r\n  Additionally, your 'R' programming team develops corresponding 'R' packages detached from\r\n \r\n  'Cornerstone' in their favorite 'R' environment.\r\n \r\n \r\n \r\n  Learn how to use 'R' packages in 'Cornerstone' 7.1.1 on 'camLineTV' YouTube channel \r\n \r\n  (&lt;https://www.youtube.com/watch?v=HEQHwq_laXU&gt;) (available in German).","Algorithms implementing populations of agents that interact with one another and sense their environment may exhibit emergent behavior such as self-organization and swarm intelligence. Here, a swarm system called Databionic swarm (DBS) is introduced which was published in Thrun, M.C., Ultsch A.: \"Swarm Intelligence for Self-Organized Clustering\" (2020), Artificial Intelligence, &lt;DOI:10.1016/j.artint.2020.103237&gt;. DBS is able to adapt itself to structures of high-dimensional data such as natural clusters characterized by distance and/or density based structures in the data space. The first module is the parameter-free projection method called Pswarm (Pswarm()), which exploits the concepts of self-organization and emergence, game theory, swarm intelligence and symmetry considerations. The second module is the parameter-free high-dimensional data visualization technique, which generates projected points on the topographic map with hypsometric tints defined by the generalized U-matrix (GeneratePswarmVisualization()). The third module is the clustering method itself with non-critical parameters (DBSclustering()). Clustering can be verified by the visualization and vice versa. The term DBS refers to the method as a whole. It enables even a non-professional in the field of data mining to apply its algorithms for visualization and/or clustering to data sets with completely different structures drawn from diverse research fields. The comparison to common projection methods can be found in the book of Thrun, M.C.: \"Projection Based Clustering through Self-Organization and Swarm Intelligence\" (2018) &lt;DOI:10.1007/978-3-658-20540-9&gt;. A comparison to 26 common clustering algorithms on 15 datasets is presented on the website.","datamap utilizes variable bindings and objects of class\r\n \r\n  \"UserDefinedDatabase\" to provide a simple mapping system to\r\n \r\n  foreign objects. Maps can be used as environments or attached\r\n \r\n  to the search path, and changes to either are persistent.\r\n \r\n  Mapped foreign objects are fetched in real-time and are never\r\n \r\n  cached by the mapping system.","Chaos theory has been hailed as a revolution of thoughts and attracting ever increasing \r\n \r\n  attention of many scientists from diverse disciplines. Chaotic systems are nonlinear deterministic \r\n \r\n  dynamic systems which can behave like an erratic and apparently random motion. A relevant field\r\n \r\n  inside chaos theory and nonlinear time series analysis is the detection of a chaotic behaviour \r\n \r\n  from empirical time series data. One of the main features of chaos is the well known initial value \r\n \r\n  sensitivity property. Methods and techniques related to test the hypothesis of chaos try to quantify \r\n \r\n  the initial value sensitive property estimating the Lyapunov exponents. The DChaos package \r\n \r\n  provides different useful tools and efficient algorithms which test robustly the hypothesis of chaos \r\n \r\n  based on the Lyapunov exponent in order to know if the data generating process behind time series \r\n \r\n  behave chaotically or not.","Provides various tools for analysing density profiles\r\n \r\n  obtained by resistance drilling. It can load individual or\r\n \r\n  multiple files and trim the starting and ending part of each\r\n \r\n  density profile. Tools are also provided to trim profiles\r\n \r\n  manually, to remove the trend from measurements using several\r\n \r\n  methods, to plot the profiles and to detect tree rings\r\n \r\n  automatically. Written with a focus on forestry use of resistance\r\n \r\n  drilling in standing trees.","There are various functions for managing and cleaning data before the application of different approaches. This includes identifying and erasing sudden jumps in dendrometer data not related to environmental change, identifying the time gaps of recordings, and changing the temporal resolution of data to different frequencies. Furthermore, the package calculates daily statistics of dendrometer data, including the daily amplitude of tree growth. Various approaches can be applied to separate radial growth from daily cyclic shrinkage and expansion due to uptake and loss of stem water. In addition, it identifies periods of consecutive days with user-defined climatic conditions in daily meteorological data, then check what trees are doing during that period.","Queues, stacks, and 'deques' are list-like, abstract data types. \r\n \r\n  These are meant to be very cheap to \"grow\", or insert new objects into.\r\n \r\n  A typical use case involves storing data in a list in a streaming fashion,\r\n \r\n  when you do not necessarily know how may elements need to be stored.\r\n \r\n  Unlike R's lists, the new data structures provided here are not\r\n \r\n  necessarily stored contiguously, making insertions and deletions at the\r\n \r\n  front/end of the structure much faster. The underlying implementation\r\n \r\n  is new and uses a head/tail doubly linked list; thus, we do not rely on R's\r\n \r\n  environments or hashing. To avoid unnecessary data copying, most operations\r\n \r\n  on these data structures are performed via side-effects.","Contains functions for the 'DivE' estimator &lt;doi:10.1371/journal.pcbi.1003646&gt;. The 'DivE' estimator is a heuristic approach to estimate the number of classes or the number of species (species richness) in a population.","Functions to describe sampling and diversity dynamics of fossil occurrence datasets (e.g. from the Paleobiology Database). The package includes methods to calculate range- and occurrence-based metrics of taxonomic richness, extinction and origination rates, along with traditional sampling measures. A powerful subsampling tool is also included that implements frequently used sampling standardization methods in a multiple bin-framework. The plotting of time series and the occurrence data can be simplified by the functions incorporated in the package, as well other calculations, such as environmental affinities and extinction selectivity testing. Details can be found in: Kocsis, A.T.; Reddin, C.J.; Alroy, J. and Kiessling, W. (2019) &lt;doi:10.1101/423780&gt;.","Compares the fit of alternative models of continuous trait differentiation between sister species and other paired lineages. Differences in trait means between two lineages arise as they diverge from a common ancestor, and alternative processes of evolutionary divergence are expected to leave unique signatures in the distribution of trait differentiation in datasets comprised of many lineage pairs. Models include approximations of divergent selection, drift, and stabilizing selection. A variety of model extensions facilitate the testing of process-to-pattern hypotheses. Users supply trait data and divergence times for each lineage pair. The fit of alternative models is compared in a likelihood framework.","Provides functions to perform\r\n \r\n  reproducible parallel foreach loops, using independent\r\n \r\n  random streams as generated by L'Ecuyer's combined\r\n \r\n  multiple-recursive generator [L'Ecuyer (1999), &lt;DOI:10.1287/opre.47.1.159&gt;].\r\n \r\n  It enables to easily convert standard %dopar% loops into\r\n \r\n  fully reproducible loops, independently of the number\r\n \r\n  of workers, the task scheduling strategy, or the chosen\r\n \r\n  parallel environment and associated foreach backend.","Provides a generalized method for printing tabular data within \r\n \r\n the R environment in order to make the process of presenting high quality \r\n \r\n tabular output seamless for the user. Output is directed to the R graphics \r\n \r\n device so that tables can be exported to any file format supported by the \r\n \r\n graphics device. Utilizes a formula interface to specify the contents of \r\n \r\n tables often found in manuscripts or business reports. In addition, formula \r\n \r\n interface provides inline formatting of the numeric cells of a table and \r\n \r\n renaming column labels.","Visualization and manipulation of the mineral-chemistry network across deep time on earth.","Fits dose-response models using an evolutionary\r\n \r\n  algorithm to estimate the model parameters. The procedure currently\r\n \r\n  can fit 3-parameter, 4-parameter, and 5-parameter log-logistic models\r\n \r\n  as well as exponential models. Functions are also provided to plot,\r\n \r\n  make predictions, and calculate confidence intervals for the resulting\r\n \r\n  models. For details see \"Nonlinear Dose-response Modeling of\r\n \r\n  High-Throughput Screening Data Using an Evolutionary Algorithm\",\r\n \r\n  Ma, J., Bair, E., Motsinger-Reif, A.; Dose-Response (In press).","Easily compute education inequality measures and the distribution \r\n  of educational attainments for any group of countries, using the data set \r\n  developed in Jorda, V. and Alonso, JM. (2017) &lt;DOI:10.1016/j.worlddev.2016.10.005&gt;. \r\n  The package offers the possibility to compute not only the Gini index, but \r\n  also generalized entropy measures for different values of the sensitivity \r\n  parameter. In particular, the package includes functions to compute the \r\n  mean log deviation, which is more sensitive to the bottom part of the \r\n  distribution; the Theils entropy measure, equally sensitive to all parts \r\n  of the distribution; and finally, the GE measure when the sensitivity \r\n  parameter is set equal to 2, which gives more weight to differences in \r\n  higher education. The decomposition of these measures in the components \r\n  between-country and within-country inequality is also provided. Two \r\n  graphical tools are also provided, to analyse the evolution of the\r\n  distribution of educational attainments: The cumulative distribution \r\n  function and the Lorenz curve.","Implements stochastic simulations of community assembly (ecological\r\n \r\n  diversification) using customizable ecospace frameworks (functional trait\r\n \r\n  spaces). Provides a wrapper to calculate common ecological disparity and\r\n \r\n  functional ecology statistical dynamics as a function of species richness.\r\n \r\n  Functions are written so they will work in a parallel-computing environment.","Provides functions to create simulated time series of environmental\r\n \r\n  exposures (e.g., temperature, air pollution) and health outcomes for use in\r\n \r\n  power analysis and simulation studies in environmental epidemiology. This\r\n \r\n  package also provides functions to evaluate the results of simulation studies\r\n \r\n  based on these simulated time series. This work was supported by a grant\r\n \r\n  from the National Institute of Environmental Health Sciences (R00ES022631) and\r\n \r\n  a fellowship from the Colorado State University Programs for Research and\r\n \r\n  Scholarly Excellence.","Collection of building blocks for the design and analysis\r\n        of evolutionary multiobjective optimization algorithms.","The evolutionary model-based multiresponse approach (EMMA)\r\n \r\n  is a novel methodology to process optimisation and product\r\n \r\n  improvement. The approach is suitable to contexts in which the\r\n \r\n  experimental cost and/or time limit the number of implementable\r\n \r\n  trials.","Provides tools for simulating from discrete-time individual level models for infectious disease data analysis. This epidemic model class contains spatial and contact-network based models with two disease types: Susceptible-Infectious (SI) and Susceptible-Infectious-Removed (SIR).","Provides tools for simulating from continuous-time individual level models of disease transmission, and carrying out infectious disease data analyses with the same models. The epidemic models considered are distance-based and/or contact network-based models within Susceptible-Infectious-Removed (SIR) or Susceptible-Infectious-Notified-Removed (SINR) compartmental frameworks. An overview of the implemented continuous-time individual level models for epidemics is given by Almutiry and Deardon (2019) &lt;doi:10.1515/ijb-2017-0092&gt;.","A tool which allows users to create and evaluate ensembles \r\n \r\n  of species distribution model (SDM) predictions. \r\n \r\n  Functionality is offered through R functions or a GUI (R Shiny app). \r\n \r\n  This tool can assist users in identifying spatial uncertainties and \r\n \r\n  making informed conservation and management decisions. The package is \r\n \r\n  further described in Woodman et al (2019) &lt;doi:10.1111/2041-210X.13283&gt;.","Three semi-parametric methods for detection of outliers in environmental data based on kernel regression and subsequent analysis of smoothing residuals. The first method (Campulova, Michalek, Mikuska and Bokal (2018) &lt;DOI: 10.1002/cem.2997&gt;) analyzes the residuals using changepoint analysis, the second method is based on control charts (Campulova, Veselik and Michalek (2017) &lt;DOI: 10.1016/j.apr.2017.01.004&gt;) and the third method (Holesovsky, Campulova and Michalek (2018) &lt;DOI: 10.1016/j.apr.2017.06.005&gt;) analyzes the residuals using extreme value theory (Holesovsky, Campulova and Michalek (2018) &lt;DOI: 10.1016/j.apr.2017.06.005&gt;).","Functions to predict fish movement parameters plotting leptokurtic fish dispersal kernels (see Radinger and Wolter, 2014: Patterns and predictors of fish dispersal in rivers. Fish and Fisheries. 15:456-473.)","Use spectrophotometry measurements performed on insects as a way to infer pathogens \r\n \r\n  virulence. Insect movements cause fluctuations in fluorescence signal, and functions are \r\n \r\n  provided to estimate when the insect has died as the moment when variance in autofluorescence \r\n \r\n  signal drops to zero. The package provides functions to obtain this estimate together with \r\n \r\n  functions to import spectrophotometry data from a Biotek microplate reader. Details of the method \r\n \r\n  are given in Parthuisot et al. (2018) &lt;doi:10.1101/297929&gt;.","Compute energy fluxes in trophic networks, from resources to their consumers, and can be applied to systems ranging from simple two-species interactions to highly complex food webs. It implements the approach described in Gauzens et al. (2017) &lt;doi:10.1101/229450&gt; to calculate energy fluxes, which are also used to calculate equilibrium stability.","A set of analytical tools useful in analysing ecological and geographical data sets, both ancient and modern. The package includes functions for estimating species richness (Chao 1 and 2, ACE, ICE, Jacknife), shared species/beta diversity, species area curves and geographic distances and areas.","Fits continuous-time random walk and correlated random walk state-space models to filter animal tracking data ('Argos', processed light-level 'geolocation', 'GPS'). Template Model Builder ('TMB') is used for fast estimation. The 'Argos' data can be: (older) least squares-based locations; (newer) Kalman filter-based locations with error ellipse information; or a mixture of both. The models estimate two sets of location states corresponding to: 1) each observation, which are (usually) irregularly timed; and 2) user-specified time intervals (regular or irregular). Latent variable models are provided to estimate move persistence along tracks as an index of behaviour. 'Jonsen I', 'McMahon CR', 'Patterson TA', 'Auger-Methe M', 'Harcourt R', 'Hindell MA', 'Bestley S' (2019) Movement responses to environment: fast inference of variation among southern elephant seals with a mixed effects model. Ecology 100:e02566 &lt;doi:10.1002/ecy.2566&gt;.","This is an evolutionary algorithm for fuzzy systems, a\r\n        genetic algorithm is used to construct a fuzzy system able to\r\n        fit the given training data.  This fuzzy system can then be\r\n        used as a prediction model, it's composed of fuzzy logic rules\r\n        that provide a good lingustic representation.","A machine learning based approach for fungal species identification using barcode sequence data. The multi-class random forest model has been used for prediction purpose, where the gap-pair compositional feature was used to encode the barcode sequence data. The encoded dataset was used as input for prediction purpose. Though this approach has been developed for fungal species identification in particular, can be used for other species identification as well.","Functions to calculate fuzzy versions of species' occurrence patterns based on presence-absence data (including inverse distance interpolation, trend surface analysis and prevalence-independent favourability GLM), and pair-wise fuzzy similarity (based on fuzzy versions of commonly used similarity indices) among those occurrence patterns. Includes also functions for model comparison (overlap and fuzzy similarity, loss or gain), and for data preparation, such as obtaining unique abbreviations of species names, converting species lists (long format) to presence-absence tables (wide format), transposing part of a data frame, assessing the false discovery rate, or analysing and dealing with multicollinearity among variables. Includes also sample datasets for providing practical examples.","Functions for importing, creating, editing and\r\n \r\n  exporting 'FSK' files &lt;https://foodrisklabs.bfr.bund.de/fsk-ml-food-safety-knowledge-markup-language/&gt;\r\n \r\n  using the 'R' programming environment. Furthermore, it enables users\r\n \r\n  to run simulations contained in the 'FSK' files and visualize the results.","GPU'/CPU Benchmarking on Debian-package based systems\r\n \r\n  This package benchmarks performance of a few standard linear algebra\r\n \r\n  operations (such as a matrix product and QR, SVD and LU decompositions)\r\n \r\n  across a number of different 'BLAS' libraries as well as a 'GPU' implementation.\r\n \r\n  To do so, it takes advantage of the ability to 'plug and play' different\r\n \r\n  'BLAS' implementations easily on a Debian and/or Ubuntu system. The current\r\n \r\n  version supports\r\n \r\n  - 'Reference BLAS' ('refblas') which are un-accelerated as a baseline\r\n \r\n  - Atlas which are tuned but typically configure single-threaded\r\n \r\n  - Atlas39 which are tuned and configured for multi-threaded mode\r\n \r\n  - 'Goto Blas' which are accelerated and multi-threaded\r\n \r\n  - 'Intel MKL' which is a commercial accelerated and multithreaded version.\r\n \r\n  As for 'GPU' computing, we use the CRAN package\r\n \r\n  - 'gputools'\r\n \r\n  For 'Goto Blas', the 'gotoblas2-helper' script from the ISM in Tokyo can be\r\n \r\n  used. For 'Intel MKL' we use the Revolution R packages from Ubuntu 9.10.","Power and sample size calculations for genetic association studies allowing \r\n \r\n  for misspecification of the model of genetic susceptibility.\r\n \r\n  Power and/or sample size can be calculated for logistic (case/control study design) \r\n \r\n  and linear (continuous phenotype) regression models, using additive, dominant, \r\n \r\n  recessive or degree of freedom coding of the genetic covariate while assuming \r\n \r\n  a true dominant, recessive or additive genetic effect. In addition, power and \r\n \r\n  sample size calculations can be performed for gene by environment interactions. \r\n \r\n  These methods are extensions of Gauderman (2002) \r\n \r\n  &lt;doi:10.1093/aje/155.5.478&gt; and Gauderman (2002) &lt;doi:10.1002/sim.973&gt;\r\n \r\n  and are described in: \r\n \r\n  Moore CM, Jacobson S, Fingerlin TE. Power and Sample Size Calculations \r\n \r\n  for Genetic Association Studies in the Presence of Genetic Model Misspecification. \r\n \r\n  American Society of Human Genetics. \r\n \r\n  October 2018, San Diego.","Provides S4 classes and data import, preprocessing, graphing, \r\n \r\n  manipulation and export methods for geo-Spectral datasets (datasets with space/time/spectral \r\n \r\n  dimensions). These type of data are frequently collected within earth observation projects \r\n \r\n  (remote sensing, spectroscopy, bio-optical oceanography, mining, agricultural, atmospheric, \r\n \r\n  environmental or similar branch of science).","Additional annotations, stats, geoms and scales for plotting \r\n \r\n  \"light\" spectra with 'ggplot2', together with specializations of ggplot()\r\n \r\n  and autoplot() methods for spectral data and waveband definitions\r\n \r\n  stored in objects of classes defined in package 'photobiology'. Part of the \r\n \r\n  'r4photobiology' suite, Aphalo P. J. (2015) &lt;doi:10.19232/uv4pb.2015.1.14&gt;.","Functions are provided for quantifying evolution and selection on complex traits. \r\n \r\n  The package implements effective handling and analysis algorithms scaled for \r\n \r\n  genome-wide data and calculates a composite statistic, denoted Ghat, which is used \r\n \r\n  to test for selection on a trait. The package provides a number of simple examples \r\n \r\n  for handling and analysing the genome data and visualising the output and results. \r\n \r\n  Beissinger et al., (2018) &lt;doi:10.1534/genetics.118.300857&gt;.","A collection of functions useful in (vegetation) community analyses and ordinations. Includes automatic species selection for ordination diagrams, species response curves and rank-abundance curves as well as calculation and sorting of synoptic tables.","Identifies global (\"unknown\" or \"free\") objects in R expressions\r\n \r\n  by code inspection using various strategies, e.g. conservative or liberal.\r\n \r\n  The objective of this package is to make it as simple as possible to\r\n \r\n  identify global objects for the purpose of exporting them in distributed\r\n \r\n  compute environments.","The 'gRbase' package provides graphical modelling features\r\n \r\n  used by e.g. the packages 'gRain', 'gRim' and 'gRc'.\r\n \r\n  'gRbase' contains data sets relevant for use in connection with\r\n \r\n  graphical models (in particular all data sets used in the book Graphical\r\n \r\n  Models with R (2012)). \r\n \r\n  'gRbase' implements graph algorithms including\r\n \r\n  (i) maximum cardinality search (for marked and unmarked graphs).\r\n \r\n  (ii) moralize.\r\n \r\n  (iii) triangulate.\r\n \r\n  (iv) junction tree.\r\n \r\n  'gRbase' facilitates array operations,\r\n \r\n  'gRbase' implements functions for testing for conditional independence.\r\n \r\n  'gRbase' illustrates how hierarchical log-linear models may be\r\n \r\n  implemented and describes concept of graphical meta\r\n \r\n  data. These features, however, are not maintained anymore and\r\n \r\n  remains in 'gRbase' only because there exists a paper describing\r\n \r\n  these facilities: A Common Platform for Graphical Models in R:\r\n \r\n  The 'gRbase' Package, Journal of Statistical Software, Vol 14,\r\n \r\n  No 17, 2005.\r\n \r\n  NOTICE 'gRbase' requires that the packages graph,\r\n \r\n  'Rgraphviz' and 'RBGL' are installed from 'bioconductor'; for\r\n \r\n  installation instructions please refer to the web page given below.","A native R implementation of grammatical evolution (GE). GE facilitates the discovery of programs that can achieve a desired goal. This is done by performing an evolutionary optimisation over a population of R expressions generated via a user-defined context-free grammar (CFG) and cost function.","Performs genetic association analyses of case-parent triad (trio) data with multiple markers. It can also incorporate complete or incomplete control triads, for instance independent control children. Estimation is based on haplotypes, for instance SNP haplotypes, even though phase is not known from the genetic data. 'Haplin' estimates relative risk (RR + conf.int.) and p-value associated with each haplotype. It uses maximum likelihood estimation to make optimal use of data from triads with missing genotypic data, for instance if some SNPs has not been typed for some individuals. 'Haplin' also allows estimation of effects of maternal haplotypes and parent-of-origin effects, particularly appropriate in perinatal epidemiology. 'Haplin' allows special models, like X-inactivation, to be fitted on the X-chromosome. A GxE analysis allows testing interactions between environment and all estimated genetic effects. The models were originally described in Gjessing, HK and Lie, RT (2006) &lt;doi:10.1111/j.1529-8817.2005.00218.x&gt;.","Access to large host-parasite data is often hampered by the \r\n  availability of data and difficulty in obtaining it in a programmatic way\r\n  to encourage analyses. 'helminthR' provides a programmatic interface to the \r\n  London Natural History Museum's host-parasite database, one of the largest \r\n  host-parasite databases existing currently &lt;http://www.nhm.ac.uk/research-curation/scientific-resources/taxonomy-systematics/host-parasites/&gt;. The package allows the user\r\n  to query by host species, parasite species, and geographic location.","User-friendly and fast set of functions for estimating parameters of hierarchical Bayesian species distribution models (Latimer et al. 2006 &lt;doi:10.1890/04-0609&gt;). Such models allow interpreting the observations (occurrence and abundance of a species) as a result of several hierarchical processes including ecological processes (habitat suitability, spatial dependence and anthropogenic disturbance) and observation processes (species detectability). Hierarchical species distribution models are essential for accurately characterizing the environmental response of species, predicting their probability of occurrence, and assessing uncertainty in the model results.","The heatex package calculates heat storage in the body and\r\n \r\n  the components of heat exchange (conductive, convective,\r\n \r\n  radiative, and evaporative) between the body and the\r\n \r\n  environment during physical activity based on the principles of\r\n \r\n  partitional calorimetry. The program enables heat exchange\r\n \r\n  calculations for a range of environmental conditions when\r\n \r\n  wearing various clothing ensembles.","State-of-the-art version of the Particle Swarm Optimisation (PSO) algorithm (SPSO-2011 and SPSO-2007 capable). hydroPSO can be used as a replacement of the 'optim' R function for (global) optimization of non-smooth and non-linear functions. However, the main focus of hydroPSO is the calibration of environmental and other real-world models that need to be executed from the system console. hydroPSO is model-independent, allowing the user to easily interface any computer simulation model with the calibration engine (PSO). hydroPSO communicates with the model through the model's own input and output files, without requiring access to the model's source code. Several PSO variants and controlling options are included to fine-tune the performance of the calibration engine to different calibration problems. An advanced sensitivity analysis function together with user-friendly plotting summaries facilitate the interpretation and assessment of the calibration results. hydroPSO is parallel-capable, to alleviate the computational burden of complex models with \"long\" execution time. Bugs reports/comments/questions are very welcomed (in English, Spanish or Italian). See Zambrano-Bigiarini and Rojas (2013) &lt;doi:10.1016/j.envsoft.2013.01.004&gt; for more details.","Estimates the shape and volume of high-dimensional datasets and performs set operations: intersection / overlap, union, unique components, inclusion test, and hole detection. Uses stochastic geometry approach to high-dimensional kernel density estimation, support vector machine delineation, and convex hull generation. Applications include modeling trait and niche hypervolumes and species distribution modeling.","S3 functions for management, analysis, interpolation and plotting of time series used in hydrology and related environmental sciences. In particular, this package is highly oriented to hydrological modelling tasks. The focus of this package has been put in providing a collection of tools useful for the daily work of hydrologists (although an effort was made to optimise each function as much as possible, functionality has had priority over speed). Bugs / comments / questions / collaboration of any kind are very welcomed, and in particular, datasets that can be included in this package for academic purposes.","Imperialist Competitive Algorithm (ICA)\r\n         &lt;http://en.wikipedia.org/wiki/Imperialist_competitive_algorithm&gt;\r\n         is a computational method that is used to solve optimization\r\n\t problems of different types and it is the mathematical model\r\n\t and the computer simulation of human social evolution.  \r\n         The package provides a minimum value for the cost function\r\n\t and the best value for the optimization variables by\r\n\t Imperialist Competitive Algorithm. \r\n         Users can easily define their own objective function\r\n\t depending on the problem at hand. \r\n         This version has been successfully applied to solve\r\n\t optimization problems, for continuous functions.","The Integro-Difference Equation model is a linear, dynamical model used to model\r\n   phenomena that evolve in space and in time; see, for example, Cressie and Wikle (2011,\r\n   ISBN:978-0-471-69274-4) or Dewar et al. (2009) &lt;doi:10.1109/TSP.2008.2005091&gt;. At the\r\n   heart of the model is the kernel, which dictates how the process evolves from one time\r\n   point to the next. Both process and parameter reduction are used to facilitate computation,\r\n   and spatially-varying kernels are allowed. Data used to estimate the parameters are assumed\r\n   to be readings of the process corrupted by Gaussian measurement error. Parameters are fitted\r\n   by maximum likelihood, and estimation is carried out using an evolution algorithm.","A Bay Area high level phylogenetic analysis package mostly\r\n \r\n  using the birth-death process. Analysis of species tree\r\n \r\n  branching times and simulation of species trees under a number\r\n \r\n  of different time variable birth-death processes. Analysis of\r\n \r\n  gene tree species tree reconciliations and simulations of gene\r\n \r\n  trees in species trees.","Different high-level graphics functions for displaying large datasets, displaying circular data in a very flexible way, finding local maxima, brewing color ramps, drawing nice arrows, zooming 2D-plots, creating figures with differently colored margin and plot region. In addition, the package contains auxiliary functions for data manipulation like omitting observations with irregular values or selecting data by logical vectors, which include NAs. Other functions are especially useful in spectroscopy and analyses of environmental data: robust baseline fitting, finding peaks in spectra, converting humidity measures.","This is an alternative mechanism for importing\r\n \r\n  objects from packages. The syntax allows for importing multiple objects\r\n \r\n  from a package with a single command in an expressive way. The import\r\n \r\n  package bridges some of the gap between using library (or require) and\r\n \r\n  direct (single-object) imports. Furthermore the imported objects are not\r\n \r\n  placed in the current environment. It is also possible to import\r\n \r\n  objects from stand-alone .R files. For more information, refer to\r\n \r\n  the package vignette.","Provides simple functions to compute and plot two\r\n    types (sample-size- and coverage-based) rarefaction and extrapolation of species\r\n    diversity (Hill numbers) for individual-based (abundance) data or sampling-unit-\r\n    based (incidence) data. (Hsieh, Ma and Chao 2014) &lt;doi: 10.1111/2041-210X.12613&gt;.","Functions for predicting and mapping potential and realized distributions of invasive species within the invaded range.","We provide an R tool for teaching in Social Sciences. It allows the computation of index numbers. It is a measure of the evolution of a fixed magnitude for only a product of for several products. It is very useful in Social Sciences. Among others, we obtain simple index numbers (in chain or in serie), index numbers for not only a product or weighted index numbers as the Laspeyres index (Laspeyres, 1864), the Paasche index (Paasche, 1874) or the Fisher index (Lapedes, 1978).","This resource provides tools to create, compare, and post-process \r\n \r\n  spatial isotope assignment models of animal origin. It generates \r\n \r\n  probability-of-origin maps for individuals based on user-provided tissue and \r\n \r\n  environment isotope values (e.g., as generated by IsoMAP, Bowen [2010] \r\n \r\n  &lt;doi:10.1111/2041-210X.12147&gt;) using the framework established in Bowen \r\n \r\n  (2014) &lt;doi:10.1111/2041-210X.12147&gt;. The package 'isocat' can then \r\n \r\n  quantitatively compare and cluster these maps to group individuals by \r\n \r\n  similar origin. It also includes techniques for applying four approaches \r\n \r\n  (cumulative sum, odds ratio, quantile only, and quantile simulation) with \r\n \r\n  which users can summarize geographic origins and probable distance traveled \r\n \r\n  by individuals.","Fits univariate and joint N-mixture models for data on two unmarked site-associated species. Includes functions to estimate latent abundances through empirical Bayes methods.","These functions calculate the taxonomic measures presented in Miranda-Esquivel (2016). \r\n   The package introduces Jack-knife resampling in evolutionary distinctiveness prioritization analysis, \r\n   as a way to evaluate the support of the ranking in area prioritization, and the persistence of a given area \r\n   in a conservation analysis.\r\n   The algorithm is described in: Miranda-Esquivel, D (2016) &lt;DOI:10.1007/978-3-319-22461-9_11&gt;.","Fits joint species distribution models ('jSDM')\r\n \r\n  in a hierarchical Bayesian framework (Warton et al. 2015\r\n \r\n  &lt;doi:10.1016/j.tree.2015.09.007&gt;). The Gibbs sampler is written\r\n \r\n  in C++. It uses 'Rcpp', 'Armadillo' and 'GSL' to maximize computation\r\n \r\n  efficiency.","Makes it possible to create 'Java' objects and to execute 'Java' methods from the 'R' environment. \r\n \r\n  The 'Java' Virtual Machine is handled by a gateway server. Commands are sent to the server \r\n \r\n  through a socket connection from the 'R' environment. Calls to 'Java' methods allow for \r\n \r\n  vectors so that a particular method is iteratively run on each element of the vector. A \r\n \r\n  score algorithm also makes the calls to 'Java' methods less restrictive. The gateway server\r\n \r\n  relies on the runnable 'Java' library 'j4r.jar'. This library is licensed under the LGPL-3. Its\r\n \r\n  sources are included in this package.","Simulating species migration and range dynamics under stable or changing environmental conditions based on a simple, raster-based, stochastic migration model. Providing accessibility for considering species migration in niche-based species distribution models.","It uses species accumulation curves and diverse estimators to assess, at the same time, the levels of survey coverage in multiple geographic cells of a size defined by the user or polygons. It also enables the geographical depiction of observed species richness, survey effort and completeness values including a background with administrative areas.","A variety of ordination and community analyses\r\n \r\n  useful in analysis of data sets in community ecology. \r\n \r\n  Includes many of the common ordination methods, with \r\n \r\n  graphical routines to facilitate their interpretation, \r\n \r\n  as well as several novel analyses.","Provides a complete environment for Bayesian inference using a variety of different samplers (see ?LaplacesDemon for an overview). The README describes the history of the package development process.","Provides functionality to calculate cost surfaces based on slope (e.g. Herzog, 2010; Llobera and Sluckin, 2007 &lt;doi:10.1016/j.jtbi.2007.07.020&gt;; Pars Roche, 2002; Tobler, 1993), traversing slope (Bell and Lock, 2000), and landscape features (Llobera, 2000) to be used when modelling pathways and movement potential within a landscape (e.g. Llobera, 2015; Verhagen, 2013; White and Barber, 2012 &lt;doi:10.1016/j.jas.2012.04.017&gt;).","Variable selection for latent class analysis for model-based clustering of multivariate categorical data. The package implements a general framework for selecting the subset of variables with relevant clustering information and discard those that are redundant and/or not informative. The variable selection method is based on the approach of Fop et al. (2017) &lt;doi:10.1214/17-AOAS1061&gt; and Dean and Raftery (2010) &lt;doi:10.1007/s10463-009-0258-9&gt;. Different algorithms are available to perform the selection: stepwise, swap-stepwise and evolutionary stochastic search. Concomitant covariates used to predict the class membership probabilities can also be included in the latent class analysis model. The selection procedure can be run in parallel on multiple cores machines.","Performs Bayesian linear regression and forecasting in astronomy. The method accounts for heteroscedastic errors in both the independent and the dependent variables, intrinsic scatters (in both variables) and scatter correlation, time evolution of slopes, normalization, scatters, Malmquist and Eddington bias, upper limits and break of linearity. The posterior distribution of the regression parameters is sampled with a Gibbs method exploiting the JAGS library.","Creates plots of peptides from shotgun proteomics analysis of secretome and lysate samples. These plots contain associated protein features and scores for potential secretion and truncation.","A scripting and command-line front-end\r\n \r\n  is provided by 'r' (aka 'littler') as a lightweight binary wrapper around\r\n \r\n  the GNU R language and environment for statistical computing and graphics.\r\n \r\n  While R can be used in batch mode, the r binary adds full support for\r\n \r\n  both 'shebang'-style scripting (i.e. using a hash-mark-exclamation-path\r\n \r\n  expression as the first line in scripts) as well as command-line use in\r\n \r\n  standard Unix pipelines. In other words, r provides the R language without\r\n \r\n  the environment.","Classes and methods for spatially explicit land use change\r\n \r\n  modelling in R.","Rapid satellite data streams in operational applications\r\n \r\n  have clear benefits for monitoring land cover, \r\n \r\n  especially when information can be delivered as fast as changing\r\n \r\n  surface conditions. Over the past decade,\r\n \r\n  remote sensing has become a key tool for monitoring and predicting\r\n \r\n  environmental variables by using satellite data. \r\n \r\n  This package presents the main applications in remote sensing for \r\n \r\n  land surface monitoring and land cover mapping (soil, vegetation, water...).\r\n \r\n  Tomlinson, C.J., Chapman, L., Thornes, E., Baker, C (2011) &lt;doi:10.1002/met.287&gt;.","Facilitates the incorporation of biological processes in biogeographical analyses. It offers conveniences in fitting, comparing and extrapolating models of biological processes such as physiology and phenology. These spatial extrapolations can be informative by themselves, but also complement traditional correlative species distribution models, by mixing environmental and process-based predictors.","A set of tools for likelihood-based estimation, model selection and testing of two- and three-range shift and migration models for animal movement data as described in Gurarie et al. (2017) &lt;doi: 10.1111/1365-2656.12674&gt;. Provided movement data (X, Y and Time), including irregularly sampled data, functions estimate the time, duration and location of one or two range shifts, as well as the ranging area and auto-correlation structure of the movment. Tests assess, for example, whether the shift was \"significant\", and whether a two-shift migration was a true return migration.","Download data from the Ada and Archibald MacLeish Field \r\n \r\n  Station in Whately, MA. The Ada \r\n \r\n  and Archibald MacLeish Field Station is a 260-acre patchwork of \r\n \r\n  forest and farmland located in West Whately, MA that provides opportunities \r\n \r\n  for faculty and students to pursue environmental research, outdoor \r\n \r\n  education, and low-impact recreation \r\n \r\n  (see &lt;http://www.smith.edu/ceeds/macleish.php&gt; for more information). \r\n \r\n  This package contains \r\n \r\n  weather data over several years, and spatial data on various man-made and \r\n \r\n  natural structures.","Can detect relatively weak spatial genetic patterns by using Moran's Eigenvector Maps (MEM) to extract only the spatial component of genetic variation. Has applications in landscape genetics where the movement and dispersal of organisms are studied using neutral genetic variation.","A graphical user interface (GUI) for performing Multidimensional Scaling applications and interactively analysing the results all within the GUI environment. The MDS-GUI provides means of performing Classical Scaling, Least Squares Scaling, Metric SMACOF, Non-Metric SMACOF, Kruskal's Analysis and Sammon Mapping with animated optimisation.","Tools to generate random landscape graphs, evaluate species\r\n    occurrence in dynamic landscapes, simulate future landscape occupation and\r\n    evaluate range expansion when new empty patches are available (e.g. as a\r\n    result of climate change).","Resolves complex metagenomic mixtures by analysing\r\n    deep sequencing data, using a mixture model based approach.\r\n    The use of parallel Monte Carlo Markov chains for the exploration\r\n    of the species space enables the identification of the set\r\n    of species more likely to contribute to the mixture.","A glycolipid mass spectrometry technology has the potential to accurately identify individual bacterial species from polymicrobial samples. To develop bacterial identification algorithms (e.g. machine learning) using this glycolipid technology, it is necessary to generate a large number of various in-silico polymicrobial mass spectra that are similar to real mass spectra. 'MGMS2' (Membrane Glycolipid Mass Spectrum Simulator) generates such in-silico mass spectra, considering errors in m/z (mass-to-charge ratio) and variances of intensity values, occasions of missing signature ions, and noise peaks. It estimates summary statistics of monomicrobial mass spectra for each strain or species and simulates polymicrobial glycolipid mass spectra using the summary statistics of monomicrobial mass spectra. References: Ryu, S.Y., Wendt, G.A., Chandler, C.E., Ernst, R.K. and Goodlett, D.R. (2019) &lt;doi:10.1021/acs.analchem.9b03340&gt; \"Model-based Spectral Library Approach for Bacterial Identification via Membrane Glycolipids.\" Gibb, S. and Strimmer, K. (2012) &lt;doi:10.1093/bioinformatics/bts447&gt; \"MALDIquant: a versatile R package for the analysis of mass spectrometry data.\"","Ten tools for analysis of major histocompatibility complex (MHC) data in non-\r\n    model species. The functions are tailored for amplicon data sets that have been \r\n    filtered using the 'dada2' method (for more information visit \r\n    &lt;https://benjjneb.github.io/dada2&gt;), but even other data sets can be analyzed, \r\n    if the data tables are formatted according to the description in each function.\r\n    The ReplMatch() function matches replicates in data sets in order to evaluate \r\n    genotyping success.\r\n    The GetReplTable() and GetReplStats() functions perform such an evaluation.\r\n    The HpltFind() function infers putative haplotypes from families in the data \r\n    set. \r\n    The GetHpltTable() and GetHpltStats() functions evaluate the accuracy of \r\n    the haplotype inference.\r\n    The PapaDiv() function compares parent pairs in the data set and calculate their \r\n    joint MHC diversity, taking into account sequence variants that occur in both \r\n    parents.\r\n    The CalcPdist() function calculates the p-distances from pairwise comparisons of all \r\n    sequences in a data set, and mean p-distances of all pairwise comparisons within each \r\n    sample in a data set. The function includes the options to specify which codons to \r\n    compare and to calculate amino acid p-distances.\r\n    The CreateFas() function creates a fasta file with all the sequences in the data \r\n    set.\r\n    The CreateSamplesFas() function creates a fasta file for each sample in the data \r\n    set.","Provides functions to compute and visualize movement-based kernel density estimates (MKDEs) for animal utilization distributions in 2 or 3 spatial dimensions.","A framework package aimed to provide standardized computational environment for specialist work via object classes to represent the data coded by samples, taxa and segments (i.e. subpopulations, repeated measures). It supports easy processing of the data along with cross tabulation and relational data tables for samples and taxa. An object of class `mefa' is a project specific compendium of the data and can be easily used in further analyses. Methods are provided for extraction, aggregation, conversion, plotting, summary and reporting of `mefa' objects. Reports can be generated in plain text or LaTeX format. Vignette contains worked examples.","Analyses species distribution models and evaluates their performance. It includes functions for performing variation partitioning, calculating several measures of model discrimination and calibration, optimizing prediction thresholds based on a number of criteria, performing multivariate environmental similarity surface (MESS) analysis, and displaying various analytical plots.","Model for simulating language evolution in terms of cultural evolution (Smith &amp; Kirby (2008) &lt;DOI:10.1098/rstb.2008.0145&gt;; Deacon 1997). The focus is on the emergence of argument-marking systems (Dowty (1991) &lt;DOI:10.1353/lan.1991.0021&gt;, Van Valin 1999, Dryer 2002, Lestrade 2015a), i.e. noun marking (Aristar (1997) &lt;DOI:10.1075/sl.21.2.04ari&gt;, Lestrade (2010) &lt;DOI:10.7282/T3ZG6R4S&gt;), person indexing (Ariel 1999, Dahl (2000) &lt;DOI:10.1075/fol.7.1.03dah&gt;, Bhat 2004), and word order (Dryer 2013), but extensions are foreseen. Agents start out with a protolanguage (a language without grammar; Bickerton (1981) &lt;DOI:10.17169/langsci.b91.109&gt;, Jackendoff 2002, Arbib (2015) &lt;DOI:10.1002/9781118346136.ch27&gt;) and interact through language games (Steels 1997). Over time, grammatical constructions emerge that may or may not become obligatory (for which the tolerance principle is assumed; Yang 2016). Throughout the simulation, uniformitarianism of principles is assumed (Hopper (1987) &lt;DOI:10.3765/bls.v13i0.1834&gt;, Givon (1995) &lt;DOI:10.1075/z.74&gt;, Croft (2000), Saffran (2001) &lt;DOI:10.1111/1467-8721.01243&gt;, Heine &amp; Kuteva 2007), in which maximal psychological validity is aimed at (Grice (1975) &lt;DOI:10.1057/9780230005853_5&gt;, Levelt 1989, Gaerdenfors 2000) and language representation is usage based (Tomasello 2003, Bybee 2010). In Lestrade (2015b) &lt;DOI:10.15496/publikation-8640&gt;, Lestrade (2015c) &lt;DOI:10.1075/avt.32.08les&gt;, and Lestrade (2016) &lt;DOI:10.17617/2.2248195&gt;), which reported on the results of preliminary versions, this package was announced as WDWTW (for who does what to whom), but for reasons of pronunciation and generalization the title was changed.","An implementation of metaheuristic algorithms for continuous optimization. Currently, the package contains the implementations of 21 algorithms, as follows: particle swarm optimization (Kennedy and Eberhart, 1995), ant lion optimizer (Mirjalili, 2015 &lt;doi:10.1016/j.advengsoft.2015.01.010&gt;), grey wolf optimizer (Mirjalili et al., 2014 &lt;doi:10.1016/j.advengsoft.2013.12.007&gt;), dragonfly algorithm (Mirjalili, 2015 &lt;doi:10.1007/s00521-015-1920-1&gt;), firefly algorithm (Yang, 2009 &lt;doi:10.1007/978-3-642-04944-6_14&gt;), genetic algorithm (Holland, 1992, ISBN:978-0262581110), grasshopper optimisation algorithm (Saremi et al., 2017 &lt;doi:10.1016/j.advengsoft.2017.01.004&gt;), harmony search algorithm (Mahdavi et al., 2007 &lt;doi:10.1016/j.amc.2006.11.033&gt;), moth flame optimizer (Mirjalili, 2015 &lt;doi:10.1016/j.knosys.2015.07.006&gt;, sine cosine algorithm (Mirjalili, 2016 &lt;doi:10.1016/j.knosys.2015.12.022&gt;), whale optimization algorithm (Mirjalili and Lewis, 2016 &lt;doi:10.1016/j.advengsoft.2016.01.008&gt;), clonal selection algorithm (Castro, 2002 &lt;doi:10.1109/TEVC.2002.1011539&gt;), differential evolution (Das &amp; Suganthan, 2011), shuffled frog leaping (Eusuff, Landsey &amp; Pasha, 2006), cat swarm optimization (Chu et al., 2006), artificial bee colony algorithm (Karaboga &amp; Akay, 2009), krill-herd algorithm (Gandomi &amp; Alavi, 2012), cuckoo search (Yang &amp; Deb, 2009), bat algorithm (Yang, 2012), gravitational based search (Rashedi et al., 2009) and black hole optimization (Hatamlou, 2013).","Extract from two-dimensional x-y coordinates of an arm-reaching trajectory, several dependent measures such as area under the curve, latency to start the movement, x-flips, etc.; which characterize the action-dynamics of the response. Mainly developed to analyze data coming from mouse-tracking experiments.","Mouse-tracking, the analysis of mouse movements in computerized\r\n \r\n  experiments, is a method that is becoming increasingly popular in the\r\n \r\n  cognitive sciences. The mousetrap package offers functions for importing,\r\n \r\n  preprocessing, analyzing, aggregating, and visualizing mouse-tracking data.","Provides the facility to calculate non-isotropic accumulated cost surface and least-cost paths using a number of human-movement-related cost functions that can be selected by the user. It just requires a Digital Terrain Model, a start location and (optionally) destination locations. See Alberti (2019) &lt;doi:10.1016/j.softx.2019.100331&gt;.","Tools to visualize movement data (e.g. from GPS tracking) and temporal changes of environmental data (e.g. from remote sensing) by creating video animations.","Estimating wind speed from trajectories of individually tracked birds using a maximum likelihood approach.","Computationally efficient functions to provide direct likelihood-based\r\n \r\n  inference for partially-observed multivariate birth-death processes. Such processes\r\n \r\n  range from a simple Yule model to the complex susceptible-infectious-removed model\r\n \r\n  in disease dynamics. Efficient likelihood evaluation facilitates maximum likelihood\r\n \r\n  estimation and Bayesian inference.","A set of tools for fitting Markov-modulated linear regression, where responses Y(t) are time-additive, and model operates in the external environment, which is described as a continuous time Markov chain with finite state space. \r\n \r\n  Model is proposed by Alexander Andronov (2012) &lt;arXiv:1901.09600v1&gt; and algorithm of parameters estimation is based on eigenvalues and eigenvectors decomposition.\r\n \r\n  Markov-switching regression models have the same idea of varying the regression parameters randomly in accordance with external environment. The difference is that for Markov-modulated linear regression model the external environment is described as a continuous-time homogeneous irreducible Markov chain with known parameters while switching models consider Markov chain as unobserved and estimation procedure involves estimation of transition matrix.\r\n \r\n  These models have significant differences in terms of the analytical approach.\r\n \r\n  Also, package provides a set of data simulation tools for Markov-modulated linear regression (for academical/research purposes).\r\n \r\n  Research project No. 1.1.1.2/VIAA/1/16/075.","Functions for fitting models of trait evolution\r\n \r\n  on phylogenies for continuous traits. The majority of functions\r\n \r\n  described in Thomas and Freckleton (2012) &lt;doi:10.1111/j.2041-210X.2011.00132.x&gt; and include functions that allow for tests of variation in the rates of trait evolution.","Code to support a systems biology research program from inception through publication. The methods focus on dimension reduction approaches to detect patterns in complex, multivariate experimental data and places an emphasis on informative visualizations. The goal for this project is to create a package that will evolve over time, thereby remaining relevant and reflective of current methods and techniques. As a result, we encourage suggested additions to the package, both methodological and graphical.","Fits multivariate (Brownian Motion, Early Burst, ACDC, Ornstein-Uhlenbeck and Shifts) models of continuous traits evolution on trees and time series. 'mvMORPH' also proposes high-dimensional multivariate comparative tools (linear models using Generalized Least Squares and multivariate tests) based on penalized likelihood. See\r\n \r\n  Clavel et al. (2015) &lt;DOI:10.1111/2041-210X.12420&gt;, Clavel et al. (2019) &lt;DOI:10.1093/sysbio/syy045&gt;, and Clavel &amp; Morlon (2020) &lt;DOI:10.1093/sysbio/syaa010&gt;.","Build and run spatially explicit\r\n \r\n  agent-based models using only the R platform. 'NetLogoR' follows the same\r\n \r\n  framework as the 'NetLogo' software\r\n \r\n  (Wilensky, 1999 &lt;http://ccl.northwestern.edu/netlogo/&gt;) and is a translation\r\n \r\n  in R of the structure and functions of 'NetLogo'.\r\n \r\n  'NetLogoR' provides new R classes to define model agents and functions to\r\n \r\n  implement spatially explicit agent-based models in the R environment.\r\n \r\n  This package allows benefiting of the fast and easy coding phase from the\r\n \r\n  highly developed 'NetLogo' framework, coupled with the versatility, power\r\n \r\n  and massive resources of the R software.\r\n \r\n  Examples of three models (Ants &lt;http://ccl.northwestern.edu/netlogo/models/Ants&gt;,\r\n \r\n  Butterfly (Railsback and Grimm, 2012) and Wolf-Sheep-Predation\r\n \r\n  &lt;http://ccl.northwestern.edu/netlogo/models/WolfSheepPredation&gt;) written using\r\n \r\n  'NetLogoR' are available. The 'NetLogo' code of the original version of these\r\n \r\n  models is provided alongside.\r\n \r\n  A programming guide inspired from the 'NetLogo' Programming Guide\r\n \r\n  (&lt;https://ccl.northwestern.edu/netlogo/docs/programming.html&gt;) and a dictionary\r\n \r\n  of 'NetLogo' primitives (&lt;https://ccl.northwestern.edu/netlogo/docs/dictionary.html&gt;)\r\n \r\n  equivalences are also available.\r\n \r\n  NOTE: To increment 'time', these functions can use a for loop or can be\r\n \r\n  integrated with a discrete event simulator, such as 'SpaDES'\r\n \r\n  (&lt;https://cran.r-project.org/package=SpaDES&gt;).\r\n \r\n  The suggested package 'fastshp' can be installed with\r\n \r\n  'install.packages(\"fastshp\", repos = \"https://rforge.net\", type = \"source\")'.","Simulates the extinction of species in the food web and it analyzes\r\n             its cascading effects, described in Dunne et al. (2002) &lt;doi:10.1073/pnas.192407699&gt;.","This is a non-parametric method for joint adaptive mean-variance regularization and variance stabilization of high-dimensional data. It is suited for handling difficult problems posed by high-dimensional multivariate datasets (p &gt;&gt; n paradigm). Among those are that the variance is often a function of the mean, variable-specific estimators of variances are not reliable, and tests statistics have low powers due to a lack of degrees of freedom. Key features include:\r\n \r\n  (i) Normalization and/or variance stabilization of the data,\r\n \r\n  (ii) Computation of mean-variance-regularized t-statistics (F-statistics to follow),\r\n \r\n  (iii) Generation of diverse diagnostic plots,\r\n \r\n  (iv) Computationally efficient implementation using C/C++ interfacing and an option for parallel computing to enjoy a faster and easier experience in the R environment.","Investigate the evolution of biological processes by capturing evolutionary signatures in transcriptomes (Drost et al. (2017) &lt;doi:10.1093/bioinformatics/btx835&gt;). The aim of this tool is to provide a transcriptome analysis environment for answering questions regarding the evolution of biological processes (Drost et al. (2016) &lt;doi:10.1101/051565&gt;).","Provides residuals and overdispersion metrics to assess the fit of N-mixture models obtained using the package 'unmarked'. \r\n \r\n  Details on the methods are given in Knape et al. (2017) &lt;doi:10.1101/194340&gt;.","Setup, run and analyze 'NetLogo' (&lt;https://ccl.northwestern.edu/netlogo/&gt;) model simulations in 'R'.\r\n \r\n  'nlrx' experiments use a similar structure as 'NetLogos' Behavior Space experiments. \r\n \r\n  However, 'nlrx' offers more flexibility and additional tools for running and analyzing complex simulation designs and sensitivity analyses.\r\n \r\n  The user defines all information that is needed in an intuitive framework, using class objects.\r\n \r\n  Experiments are submitted from 'R' to 'NetLogo' via 'XML' files that are dynamically written, based on specifications defined by the user.\r\n \r\n  By nesting model calls in future environments, large simulation design with many runs can be executed in parallel.\r\n \r\n  This also enables simulating 'NetLogo' experiments on remote high performance computing machines.\r\n \r\n  In order to use this package, 'Java' and 'NetLogo' (&gt;= 5.3.1) need to be available on the executing system.","Wrapper package for the nmfgpu library, which implements several\r\n \r\n  Non-negative Matrix Factorization (NMF) algorithms for CUDA platforms.\r\n \r\n  By using the acceleration of GPGPU computing, the NMF can be used for\r\n \r\n  real-world problems inside the R environment. All CUDA devices starting with\r\n \r\n  Kepler architecture are supported by the library.","Calculates spatial pattern analysis using a T-square sample procedure.\r\n \r\n  This method is based on two measures \"x\" and \"y\".\r\n \r\n  \"x\" - Distance from the random point to the nearest individual.\r\n \r\n  \"y\" - Distance from individual to its nearest neighbor.\r\n \r\n  This is a methodology commonly used in phytosociology or marine benthos ecology to analyze the species' distribution (random, uniform or clumped patterns).\r\n \r\n  Ludwig &amp; Reynolds (1988, ISBN:0471832359).","Likelihood based optimal partitioning and indicator\r\n  species analysis. Finding the best binary partition for each species\r\n  based on model selection, with the possibility to take into account\r\n  modifying/confounding variables as described\r\n  in Kemencei et al. (2014) &lt;doi:10.1556/ComEc.15.2014.2.6&gt;.\r\n  The package implements binary and multi-level response models,\r\n  various measures of uncertainty, Lorenz-curve based thresholding,\r\n  with native support for parallel computations.","The multispecies and individual-based model (IBM) 'OSMOSE'\r\n  (Shin and Curry (2001) &lt;doi:10.1016/S0990-7440(01)01106-8&gt; and Shin and Curry \r\n  (2004) &lt;doi:10.1139/f03-154&gt;) focuses on fish species. This model assumes \r\n  opportunistic predation based on spatial co-occurrence and size adequacy \r\n  between a predator and its prey (size-based opportunistic predation). It\r\n  represents fish individuals grouped into schools, which are characterized by \r\n  their size, weight, age, taxonomy and geographical location (2D model), and \r\n  which undergo major processes of fish life cycle (growth, explicit predation, \r\n  natural and starvation mortalities, reproduction and migration) and fishing \r\n  exploitation. The model needs basic biological parameters that are often \r\n  available for a wide range of species, and which can be found in 'FishBase' \r\n  for instance (see &lt;http://www.fishbase.org/search.php&gt;), and fish spatial \r\n  distribution data. This package provides tools to build and run simulations \r\n  using the 'OSMOSE' model.","A Bayesian approach to estimate the number of occurred-but-not-yet-reported cases from incomplete, time-stamped reporting data for disease outbreaks. 'NobBS' learns the reporting delay distribution and the time evolution of the epidemic curve to produce smoothed nowcasts in both stable and time-varying case reporting settings, as described in McGough et al. (2019) &lt;doi:10.1101/663823&gt;.","Creating maps for statistical analysis such as proportional circles, chroropleth, typology and flows. Some functions use 'shiny' or 'leaflet' technologies for dynamism and interactivity.\r\n \r\n The great features are :\r\n \r\n - Create maps in a web environment where the parameters are modifiable on the fly ('shiny' and 'leaflet' technology).\r\n \r\n - Create interactive maps through zoom and pop-up ('leaflet' technology).\r\n \r\n - Create frozen maps with the possibility to add labels.","Provides functions to fit kernel density functions to\r\n data on temporal activity patterns of animals; estimate coefficients\r\n of overlapping of densities for two species; and calculate bootstrap\r\n estimates of confidence intervals.","Offers a gene-based meta-analysis test with filtering to detect gene-environment interactions (GxE) with association data, proposed by Wang et al. (2018) &lt;doi:10.1002/gepi.22115&gt;. It first conducts a meta-filtering test to filter out unpromising SNPs by combining all samples in the consortia data. It then runs a test of omnibus-filtering-based GxE meta-analysis (ofGEM) that combines the strengths of the fixed- and random-effects meta-analysis with meta-filtering. It can also analyze data from multiple ethnic groups.","Implements a set of online fault detectors for time-series, called: PEWMA see M. Carter\r\n \r\n  et al. (2012) &lt;doi:10.1109/SSP.2012.6319708&gt;, SD-EWMA and TSSD-EWMA see H. Raza et al. \r\n \r\n  (2015) &lt;doi:10.1016/j.patcog.2014.07.028&gt;, KNN-CAD see E. Burnaev et al. (2016)\r\n \r\n  &lt;arXiv:1608.04585&gt;, KNN-LDCD see V. Ishimtsev et al. (2017) &lt;arXiv:1706.03412&gt; and \r\n \r\n  CAD-OSE see M. Smirnov (2018) &lt;https://github.com/smirmik/CAD&gt;. The first three \r\n \r\n  algorithms belong to prediction-based techniques and the last three belong to \r\n \r\n  window-based techniques. In addition, the SD-EWMA and PEWMA algorithms are algorithms \r\n \r\n  designed to work in stationary environments, while the other four \r\n \r\n  are algorithms designed to work in non-stationary environments.","Implements the Panel Data Approach Method for program evaluation as developed in Hsiao, Ching and Ki Wan (2012). pampe estimates the effect of an intervention by comparing the evolution of the outcome for a unit affected by an intervention or treatment to the evolution of the unit had it not been affected by the intervention.","Tools to sort, edit and prune pedigrees and to extract the inbreeding coefficients\r\n    and the relationship matrix (includes code for pedigrees from self-pollinated species). \r\n    The use of pedigree data is central to genetics research within the animal and plant breeding communities to predict \r\n    breeding values. The relationship matrix between the individuals can be derived from pedigree structure following\r\n    the algorithms described for example in Vazquez et al., 2010 &lt;doi:10.2527/jas.2009-1952&gt;.","Performs statistical inference on the models of adaptive trait evolution under approximate Bayesian computation. This can simulate traits from four models, compute trait data summary statistics. Parameters are estimated under Approximate Bayesian Computation, model selection as well as posterior parameter mean will be reported. Users need to enter a comparative dataset and a phylogenetic tree.","Utilities for computing measures to assess model quality,\r\n \r\n  which are not directly provided by R's 'base' or 'stats' packages. These \r\n \r\n  include e.g. measures like r-squared, intraclass correlation coefficient\r\n \r\n  (Nakagawa, Johnson &amp; Schielzeth (2017) &lt;doi:10.1098/rsif.2017.0213&gt;), \r\n \r\n  root mean squared error or functions to check models for overdispersion, \r\n \r\n  singularity or zero-inflation and more. Functions apply to a large variety of\r\n \r\n  regression models, including generalized linear models, mixed effects models\r\n \r\n  and Bayesian models.","Fill missing symmetrical data with mirroring, calculate Procrustes alignments with or without scaling, and compute standard or vector correlation and covariance matrices (congruence coefficients) of 3D landmarks. Tolerates missing data for all analyses.","A PEP, or Portable Encapsulated Project, is a dataset that \r\n \r\n  subscribes to the PEP structure for organizing metadata. It is written using\r\n \r\n  a simple YAML + CSV format, it is your one-stop solution to metadata \r\n \r\n  management across data analysis environments. This package reads this \r\n \r\n  standardized project configuration structure into R.","Set of tools for reading, writing and transforming spatial and seasonal data in ecology, model selection and specific statistical tests. It includes functions to discretize polylines into regular point intervals, link observations to those points, compute geographical coordinates at regular intervals between waypoints, read subsets of big rasters, compute zonal statistics or table of categories within polygons or circular buffers from raster. The package also provides miscellaneous functions for model selection, spatial statistics, geometries, writing data.frame with Chinese characters, and some other functions for field ecologists.","An alternative to carrying out phenotypic adaptability and stability analyses, taking into account nonparametric statistics. Can be used as a robust approach, less sensitive to departures from common genotypic, environmental, and GxE effects data assumptions (e.g., normal distribution of errors).","Functions for reading, and in some cases writing, foreign files \r\n \r\n  containing spectral data from spectrometers and their associated software, \r\n \r\n  output from daylight simulation models in common use, and some spectral \r\n \r\n  data repositories. As well as functions for exchange of spectral data with \r\n \r\n  other R packages. Part of the 'r4photobiology' suite, \r\n \r\n  Aphalo P. J. (2015) &lt;doi:10.19232/uv4pb.2015.1.14&gt;.","Fits models of gene expression evolution to expression data from\r\n    coregulated groups of genes, assuming inverse gamma distributed rate\r\n    variation.","Spectral response data for broadband ultraviolet and visible\r\n \r\n  radiation sensors. Different data sources were used: author-supplied data\r\n \r\n  from scientific research papers, sensor-manufacturer supplied data, and\r\n \r\n  published sensor specifications. Part of the 'r4photobiology' suite\r\n \r\n  Aphalo P. J. (2015) &lt;doi:10.19232/uv4pb.2015.1.14&gt;.","A wide range of functions for phylogenetic analysis. Functionality is concentrated in phylogenetic comparative biology, but also includes numerous methods for visualizing, manipulating, reading or writing, and even inferring phylogenetic trees and data. Included among the functions in phylogenetic comparative biology are various for ancestral state reconstruction, model-fitting, simulation of phylogenies and data, and multivariate analysis. There are a broad range of plotting methods for phylogenies and comparative data which include, but are not restricted to, methods for mapping trait evolution on trees, for projecting trees into phenotypic space or a geographic map, and for visualizing correlated speciation between trees. Finally, there are a number of functions for reading, writing, analyzing, inferring, simulating, and manipulating phylogenetic trees and comparative data not covered by other packages. For instance, there are functions for randomly or non-randomly attaching species or clades to a phylogeny, for estimating supertrees or consensus phylogenies from a set, for simulating trees and phylogenetic data under a range of models, and for a wide variety of other manipulations and analyses that phylogenetic biologists might find useful in their research.","Provide function that reads binary genotype produced by 'PLINK' &lt;https://www.cog-genomics.org/plink/1.9/input#bed&gt; into a R matrix, or scan the genotype one variant at a time like apply(), it also provides functions that reads and writes genotype relatedness/kinship matrices created by 'PLINK' &lt;https://www.cog-genomics.org/plink/1.9/distance#make_rel&gt; or 'GCTA' &lt;https://cnsgenomics.com/software/gcta/#MakingaGRM&gt;. Currently it does not support writing back into 'PLINK' binary, it is best used for bringing data produced by 'PLINK' and 'GCTA' into R environment.","Population dynamic models underpin a range of analyses and applications in ecology and epidemiology. The various approaches for analysing population dynamics models (MPMs, IPMs, ODEs, POMPs, PVA) each require the model to be defined in a different way. This makes it difficult to combine different modelling approaches and data types to solve a given problem. 'pop' aims to provide a flexible and easy to use common interface for constructing population dynamic models and enabling to them to be fitted and analysed in lots of different ways.","Download and generate summaries for the rodent,\r\n \r\n  plant, ant, and weather data from the Portal Project. Portal is a\r\n \r\n  long-term (and ongoing) experimental monitoring site in the Chihuahua\r\n \r\n  desert. The raw data files can be found at\r\n \r\n  &lt;https://github.com/weecology/portaldata&gt;.","Performs wood cell anatomical data analyses on spatially explicit xylem (tracheids) datasets \r\n \r\n  derived from thin sections of woody tissue. The package includes functions for visualisation, \r\n \r\n  detection and alignment of continuous tracheid radial file (defined as rows) and individual tracheid position \r\n \r\n  within an annual ring of coniferous species. This package is designed to be used with elaborate cell output, \r\n \r\n  e.g. as provided with ROXAS (von Arx &amp; Carrer, 2014 &lt;doi:10.1016/j.dendro.2013.12.001&gt;). The package has been validated for Picea abies, \r\n \r\n  Larix Siberica, Pinus cembra and Pinus sylvestris.","Biodiversity is in crisis. The overarching aim of conservation\r\n    is to preserve biodiversity patterns and processes. To this end, protected\r\n    areas are established to buffer species and preserve biodiversity processes.\r\n    But resources are limited and so protected areas must be cost-effective.\r\n    This package contains tools to generate plans for protected areas\r\n    (prioritizations), using spatially explicit targets for biodiversity\r\n    patterns and processes. To obtain solutions in a feasible amount  of time,\r\n    this package uses the commercial 'Gurobi' software package (obtained from\r\n    &lt;http://www.gurobi.com/&gt;). For more information on using\r\n    this package, see Hanson et al. (2018) &lt;doi:10.1111/2041-210X.12862&gt;.","Includes functions for the calculation of spatially and non-spatially explicit rarefaction curves using different indices of taxonomic, functional and phylogenetic diversity. The user can also rarefy any biodiversity metric as provided by a self-written function (or an already existent one) that gives as output a vector with the values of a certain index of biodiversity calculated per plot (Ricotta, C., Acosta, A., Bacaro, G., Carboni, M., Chiarucci, A., Rocchini, D., Pavoine, S. (2019) &lt;doi:10.1016/j.ecolind.2019.105606&gt;; Bacaro, G., Altobelli, A., Cameletti, M., Ciccarelli, D., Martellos, S., Palmer, M. W.,  Chiarucci, A. (2016) &lt;doi:10.1016/j.ecolind.2016.04.026&gt;; Bacaro, G., Rocchini, D., Ghisla, A., Marcantonio, M., Neteler, M., &amp; Chiarucci, A. (2012) &lt;doi:10.1016/j.ecocom.2012.05.007&gt;).","Processing pipeline for 'SamBada' from pre- to post-processing. \r\n \r\n 'SamBada' is a landscape genomic software designed to run univariate or multivariate logistic regression between the presence of a genotype and one or several environmental variables. See Stucki (2017) &lt;doi:10.1111/1755-0998.12629&gt; and &lt;https://github.com/Sylvie/sambada&gt;. \r\n \r\n The package provides functions that can be classified into four categories: \r\n \r\n 1) Install 'SamBada' \r\n \r\n 2) Preprocessing (prepare genomic file into standards compatible with 'SamBada' and apply quality-control; retrieve environmental conditions at sampling location; prepare environmental file including removal of correlated variables and computation of population structure) \r\n \r\n 3) Processing (run 'SamBada' on multiple cores using 'Supervision')\r\n \r\n 4) Post-processing (calculate p-values and q-values, produce interactive Manhattan plots and query 'Ensembl' database, produce maps).","Implements the Range Ancestral State Estimation for phylogeography described in Quintero, I., Keil, P., Jetz, W., &amp; Crawford, F. W. (2015) &lt;DOI:10.1093/sysbio/syv057&gt;. It also includes Bayesian inference of ancestral states under a Brownian Motion model of character evolution and Maximum Likelihood estimation of rase for n-dimensional data. Visualizing functions in 3D are implemented using the rgl package.","Interface to &lt;http://proyectoavis.com&gt; database. \r\n    It provides means to download data filtered by species, order,\r\n    family, and several other criteria. Provides also basic functionality to\r\n    plot exploratory maps of the datasets.","A set of tools to help with species conservation assessments (Red List threat assessments). Includes tool for Extent of occurrence, Area of Occupancy, Minimum Enclosing Rectangle, a geographic Projection Wizard and Species batch processing.","Useful methods and data fields to extend the\r\n \r\n  bare bones 'R6' class provided by the 'R6' package - ls-method, \r\n \r\n  hashes, warning- and message-method, general get-method and a \r\n \r\n  debug-method that assigns self and private to the global environment.","Compute spatially explicit land-use metrics for stream survey sites in GRASS GIS and R as an open-source implementation of IDW-PLUS (Inverse Distance Weighted Percent Land Use for Streams). The package includes functions for preprocessing digital elevation and streams data, and one function to compute all the spatially explicit land use metrics described in Peterson et al. (2011) &lt;doi:10.1111/j.1365-2427.2010.02507.x&gt; and previously implemented by Peterson and Pearse (2017) &lt;doi:10.1111/1752-1688.12558&gt; in ArcGIS-Python as IDW-PLUS.","It has tools for normalization of rank abundance\r\n \r\n  distributions (RAD) to a desired number of ranks using MaxRank\r\n \r\n  Normalization method.\r\n \r\n  RADs are commonly used in biology/ecology and mathematically equivalent\r\n \r\n  to complementary cumulative distributions (CCDFs) which are used in\r\n \r\n  physics, linguistics and sociology and more generally in data science.","Tests gene-environment interaction for rare genetic variants using Sequence Kernel Association Test (SKAT) type gene-based tests. Includes two tests for the interaction term only, and one joint test for genetic main effects and gene-environment interaction.","An R interface to a modified version of the Free Evolutionary Algorithm Kit FrEAK. FrEAK is a toolkit written in Java to design and analyze evolutionary algorithms. Both the R interface and an extended version of FrEAK are contained in the RFreak package. For more information on FrEAK see http://sourceforge.net/projects/freak427/.","Allow access to both public and private end points to Coinbase Pro (erstwhile GDAX) \r\n \r\n  cryptocurrency exchange. For authenticated flow, users must have valid api, secret and passphrase to be able to connect.","An efficient C++ based implementation of the 'DEoptim'\r\n \r\n  function which performs global optimization by differential evolution. \r\n \r\n  Its creation was motivated by trying to see if the old approximation \"easier,\r\n \r\n  shorter, faster: pick any two\" could in fact be extended to achieving all\r\n \r\n  three goals while moving the code from plain old C to modern C++. The\r\n \r\n  initial version did in fact do so, but a good part of the gain was due to \r\n \r\n  an implicit code review which eliminated a few inefficiencies which have\r\n \r\n  since been eliminated in 'DEoptim'.","Distance-sampling is a popular method for estimating density and \r\n \r\n  abundance of organisms in ecology. Rdistance contains routines that \r\n \r\n  assist with analysis of\r\n \r\n  distance-sampling data collected on point or line transects. \r\n \r\n  Distance models are specified using regression-like formula (similar \r\n \r\n  to lm, glm, etc.). Abundance routines\r\n \r\n  perform automated bootstrapping and automated detection-function \r\n \r\n  selection. Overall (study area) and site-level (transect or point) \r\n \r\n  abundance estimates are available. A large suite of classical, \r\n \r\n  parametric detection functions are \r\n \r\n  included along with some uncommon parametric \r\n \r\n  functions (e.g., Gamma, negative exponential) and non-parametric\r\n \r\n  smoothed distance functions. Custom (user-defined) detection functions\r\n \r\n  are easily implemented (see vignette). \r\n \r\n  The help files and vignettes have been \r\n \r\n  vetted by multiple authors and tested in workshop \r\n \r\n  settings.","Contains three functions that access\r\n \r\n  environmental data from any 'ERDDAP' data web service. The rxtracto() function extracts\r\n \r\n  data along a trajectory for a given \"radius\" around the point. The\r\n \r\n  rxtracto_3D() function extracts data in a box. The rxtractogon() function\r\n \r\n  extracts data in a polygon. All of those three function use the 'rerddap' package\r\n \r\n  to extract the data, and should work with any 'ERDDAP' server.\r\n \r\n  There are also two functions, plotBBox() and plotTrack() that use the 'plotdap'\r\n \r\n  package to simplify the creation of maps of the data.","Debugging with restore points instead of break points. A restore\r\n \r\n  point stores all local variables when called inside a function. The stored\r\n \r\n  values can later be retrieved and evaluated in a modified R console that\r\n \r\n  replicates the function's environment. To debug step by step, one can simply\r\n \r\n  copy &amp; paste the function body from the R script. Particularly convenient\r\n \r\n  in combination with \"RStudio\". See the \"Github\" page inst/vignettes for a\r\n \r\n  tutorial.","A method 'generate()' is implemented in this package for the random\r\n \r\n  generation of vector time series according to models obtained by 'RMAWGEN',\r\n \r\n  'vars' or other packages. This package was created to generalize the\r\n \r\n  algorithms of the 'RMAWGEN' package for the analysis and generation of any\r\n \r\n  environmental vector time series.","Interpreted interface between 'GRASS' 7 geographical \r\n \r\n  information system and R, based on starting R from within the 'GRASS' 'GIS'\r\n \r\n  environment, or running free-standing R in a temporary 'GRASS' location;\r\n \r\n  the package provides facilities for using all 'GRASS' commands from the \r\n \r\n  R command line. This package may not be used for 'GRASS' 6, for which\r\n \r\n  'spgrass6' should be used.","Provides interface to the Bioinfo-C (internal name: BIOS) library and utilities. 'ribiosUtils' is a swiss-knife for computational biology in drug discovery, providing functions and utilities with minimal external dependency and maximal efficiency.","An interface between R and the metasim simulation engine.\r\n \r\n  The simulation environment is documented in: \"Strand, A.(2002) &lt;DOI:10.1046/j.1471-8286.2002.00208.x&gt; Metasim 1.0: an individual-based environment for simulating population genetics of \r\n \r\n  complex population dynamics. Mol. Ecol. Notes. \r\n \r\n  Please see the vignettes CreatingLandscapes and Simulating to get some ideas on how to use the packages. \r\n \r\n  See the rmetasim vignette to get an overview and to see important changes to the \r\n \r\n  code in the most recent version.","An R package for the OpenSecrets.org web services API.","An interface to the 'Open Tree of Life' API to retrieve\r\n    phylogenetic trees, information about studies used to assemble the\r\n    synthetic tree, and utilities to match taxonomic names to 'Open Tree\r\n    identifiers'. The 'Open Tree of Life' aims at assembling a\r\n    comprehensive phylogenetic tree for all named species.","An interface (wrapper) to MPI. It also \r\n \r\n  provides interactive R manager and worker environment.","Integrates R and the 'PostgreSQL/PostGIS' database \r\n \r\n  system to build and manage animal trajectory (movement) data sets. \r\n \r\n  The package relies on 'ltraj' objects from the R package 'adehabitatLT',\r\n \r\n  building the analogous 'pgtraj' data structure in 'PostGIS'. Functions\r\n \r\n  allow users to seamlessly transfer between 'ltraj' and 'pgtraj', as\r\n \r\n  well as build new 'pgtraj' directly from location data stored in the \r\n \r\n  database.","Provides the libraries of the EUtils operations for the\r\n \r\n  RNCBI package.","Gene-environment (GE) interactions have important implications to elucidate the \r\n \r\n  etiology of complex diseases beyond the main genetic and environmental effects. \r\n \r\n  Outliers and data contamination in disease phenotypes of GE studies have been commonly \r\n \r\n  encountered, leading to the development of a broad spectrum of robust penalization methods. \r\n \r\n  Nevertheless, within the Bayesian framework, the issue has not been taken care of in existing \r\n \r\n  studies. We develop a robust Bayesian variable selection method for GE interaction \r\n \r\n  studies. The proposed Bayesian method can effectively accommodate heavy-tailed errors and \r\n \r\n  outliers in the response variable while conducting variable selection by accounting for \r\n \r\n  structural sparsity. In particular, the spike-and-slab priors have been imposed on both \r\n \r\n  individual and group levels to identify important main and interaction effects. An efficient \r\n \r\n  Gibbs sampler has been developed to facilitate fast computation. The Markov chain Monte Carlo \r\n \r\n  algorithms of the proposed and alternative methods are efficiently implemented in C++.","This package implements an interface to Oracle Data Mining\r\n \r\n  (ODM). It provides an ideal environment for rapid development\r\n \r\n  of demos and proof of concept data mining studies. It\r\n \r\n  facilitates the prototyping of vertical applications and makes\r\n \r\n  ODM and the RDBMS environment easily accessible to\r\n \r\n  statisticians and data analysts familiar with R but not fluent\r\n \r\n  in SQL or familiar with the database environment. It also\r\n \r\n  facilitates the benchmarking and testing of ODM functionality\r\n \r\n  including the production of summary statistics, performance\r\n \r\n  metrics and graphics. It enables the scripting and control of\r\n \r\n  production data mining methodologies from a high-level\r\n \r\n  environment. Oracle Data Mining (ODM) is an option of Oracle\r\n \r\n  Relational Database Management System (RDBMS) Enterprise\r\n \r\n  Edition (EE). It contains several data mining and data analysis\r\n \r\n  algorithms for classification, prediction, regression,\r\n \r\n  clustering, associations, feature selection, anomaly detection,\r\n \r\n  feature extraction, and specialized analytics. It provides\r\n \r\n  means for the creation, management and operational deployment\r\n \r\n  of data mining models inside the database environment. For more\r\n \r\n  information consult the entry for \"Oracle Data Mining\" in\r\n \r\n  Wikipedia (en.wikipedia.org).","Tools for the guided selection of satellite data and environmental predictors, the combination of remote sensing and animal movement data and the mapping of resource suitability. Based on the paper by Remelgado et al. (2015) &lt;doi:10.1111/2041-210X.13199&gt;.","Implements the P-model (Stocker et al., 2019 &lt;doi:10.5194/gmd-2019-200&gt;), predicting acclimated parameters of the enzyme kinetics of C3 photosynthesis, assimilation, and dark respiration rates as a function of the environment (temperature, CO2, vapour pressure deficit, light, atmospheric pressure).","Functions for phylogenetic analysis (Castiglione et al, 2018 &lt;doi:10.1111/2041-210X.12954&gt;). The functions perform the estimation of phenotypic evolutionary rates, identification of phenotypic evolutionary rate shifts, quantification of direction and size of evolutionary change in multivariate traits, the computation of ontogenetic shape vectors and test for morphological convergence.","Provides an interface with the 'Wildbook' mark-recapture ecological database framework. It \r\n    helps users to pull data from the 'Wildbook' framework and format data for further analysis\r\n    with mark-recapture applications like 'Program MARK' (which can be accessed via the 'RMark' package in 'R').\r\n    Further information on the 'Wildbook' framework is available at: &lt;http://www.wildbook.org/doku.php&gt;.","An implementation of the framework described in \"Toward a unified\r\n \r\n  framework for connectivity that disentangles movement and mortality in space\r\n \r\n  and time\" by Fletcher et al. (2019) &lt;doi:10.1111/ele.13333&gt;. \r\n \r\n  Incorporates both resistance and absorption with spatial absorbing Markov \r\n \r\n  chains (SAMC) to provide several short-term and long-term predictions for \r\n \r\n  metrics related to connectivity in landscapes.","Provides a generic implementation of the 'RStudio' connection contract to \r\n \r\n  make it easier for database connections, and other type of connections, opened \r\n \r\n  via R packages integrate with the connections pane inside the 'RStudio' interactive\r\n \r\n  development environment (IDE).","Compares the evolution of an original network X to an augmented network Y by counting the number of Shortcuts, Detours, Dead Ends (SDDE), equal paths and disconnected nodes.","Implementation of evolutionary fuzzy systems for the data mining task called\r\n    \"subgroup discovery\". In particular, the algorithms presented in this package are:\r\n    M. J. del Jesus, P. Gonzalez, F. Herrera, M. Mesonero (2007) &lt;doi:10.1109/TFUZZ.2006.890662&gt;\r\n    M. J. del Jesus, P. Gonzalez, F. Herrera (2007) &lt;doi:10.1109/MCDM.2007.369416&gt;\r\n    C. J. Carmona, P. Gonzalez, M. J. del Jesus, F. Herrera (2010) &lt;doi:10.1109/TFUZZ.2010.2060200&gt;\r\n    C. J. Carmona, V. Ruiz-Rodado, M. J. del Jesus, A. Weber, M. Grootveld, P. Gonzlez, D. Elizondo (2015) &lt;doi:10.1016/j.ins.2014.11.030&gt;\r\n    It also provide a Shiny App to ease the analysis. The algorithms work with data sets provided in\r\n    KEEL, ARFF and CSV format and also with data.frame objects.","Functions provided by this pedagogic package allow to compute models with two popular machine learning approaches, BRT (Boosted Regression Trees) and MaxEnt (Maximum Entropy) applied on sets of marine biological and environmental data. They include the possibility of managing the main parameters for the construction of the models. Classic tools to evaluate model performance are provided (Area Under the Curve, omission rate and confusion matrix, map standard deviation) and are completed with tools to perform null models. The biological dataset includes original occurrences of two species of the class Echinoidea (sea urchins) present on the Kerguelen Plateau and that show contrasted ecological niches. The environmental dataset includes the corresponding statistics for 15 abiotic and biotic descriptors summarized for the Kerguelen Plateau and for different periods in a raster format. The package can be used for practicals to teach and learn the basics of species distribution modelling. Maps of potential distribution can be produced based on the example data included in the package, which brings prior observations of the influence of spatial and temporal heterogeneities on modelling performances. The user can also provide his own datasets to use the modelling functions.","Terrestrial and marine predictors for species distribution modelling\r\n    from multiple sources, including WorldClim &lt;http://www.worldclim.org/&gt;,,\r\n    ENVIREM &lt;http://envirem.github.io/&gt;, Bio-ORACLE &lt;http://bio-oracle.org/&gt;\r\n    and MARSPEC &lt;http://www.marspec.org/&gt;.","User-friendly framework that enables the training and the\r\n \r\n  evaluation of species distribution models (SDMs). The package implements\r\n \r\n  functions for data driven variable selection and model tuning and includes\r\n \r\n  numerous utilities to display the results. All the functions used to select\r\n \r\n  variables or to tune model hyperparameters have an interactive real-time\r\n \r\n  chart displayed in the 'RStudio' viewer pane during their execution.","A software package help user to create virtual species for species distribution modelling. It includes\r\n \r\n  several methods to help user to create virtual species distribution map.\r\n \r\n  Those maps can be used for Species Distribution Modelling (SDM) study. SDM use\r\n \r\n  environmental data for sites of occurrence of a species to predict all the sites\r\n \r\n  where the environmental conditions are suitable for the species to persist, and\r\n \r\n  may be expected to occur.","Functions for post processing the outcomes of ecological niche models. It further include methods to build stability maps based on species distribution models under climate change projections and to reduce environmental data.","Functions to estimate the density and size of a spatially distributed animal population sampled with an array of passive detectors, such as traps, or by searching polygons or transects. Models incorporating distance-dependent detection are fitted by maximizing the likelihood. Tools are included for data manipulation and model selection.","Tools for designing spatially explicit capture-recapture studies of animal populations. This is primarily a simulation manager for package 'secr'. Extensions in version 2.5.0 include costing and evaluation of detector spacing.","Allow sharing sensitive information, for example passwords,\r\n \r\n  'API' keys, etc., in R packages, using public key cryptography.","Encrypt text using a simple shifting substitution cipher with setcode(), providing two numeric keys used to define the encryption algorithm. The resulting text can be decoded using decode() function and the two numeric keys specified during encryption.","Tools for spatially explicit capture-recapture analysis of animal populations in linear habitats, extending package 'secr'.","In silico experimental evolution offers a cost-and-time effective means to test evolutionary hypotheses. Existing evolutionary simulation tools focus on simulations in a limited experimental framework, and tend to report on only the results presumed of interest by the tools designer. The R-package for Simulated Haploid Asexual Population Evolution ('rSHAPE') addresses these concerns by implementing a robust simulation framework that outputs complete population demographic and genomic information for in silico evolving communities. Allowing more than 60 parameters to be specified, 'rSHAPE' simulates evolution across discrete time-steps for an evolving community of haploid asexual populations with binary state genomes. These settings are for the current state of 'rSHAPE' and future steps will be to increase the breadth of evolutionary conditions permitted. At present, most effort was placed into permitting varied growth models to be simulated (such as constant size, exponential growth, and logistic growth) as well as various fitness landscape models to reflect the evolutionary landscape (e.g.: Additive, House of Cards - Stuart Kauffman and Simon Levin (1987) &lt;doi:10.1016/S0022-5193(87)80029-2&gt;, NK - Stuart A. Kauffman and Edward D. Weinberger (1989) &lt;doi:10.1016/S0022-5193(89)80019-0&gt;, Rough Mount Fuji - Neidhart, Johannes and Szendro, Ivan G and Krug, Joachim (2014) &lt;doi:10.1534/genetics.114.167668&gt;). This package includes numerous functions though users will only need defineSHAPE(), runSHAPE(), shapeExperiment() and summariseExperiment(). All other functions are called by these main functions and are likely only to be on interest for someone wishing to develop 'rSHAPE'. Simulation results will be stored in files which are exported to the directory referenced by the shape_workDir option (defaults to tempdir() but do change this by passing a folderpath argument for workDir when calling defineSHAPE() if you plan to make use of your results beyond your current session). 'rSHAPE' will generate numerous replicate simulations for your defined range of experimental parameters. The experiment will be built under the experimental working directory (i.e.: referenced by the option shape_workDir set using defineSHAPE() ) where individual replicate simulation results will be stored as well as processed results which I have made in an effort to facilitate analyses by automating collection and processing of the potentially thousands of files which will be created. On that note, 'rSHAPE' implements a robust and flexible framework with highly detailed output at the cost of computational efficiency and potentially requiring significant disk space (generally gigabytes but up to tera-bytes for very large simulation efforts). So, while 'rSHAPE' offers a single framework in which we can simulate evolution and directly compare the impacts of a wide range of parameters, it is not as quick to run as other in silico simulation tools which focus on a single scenario with limited output. There you have it, 'rSHAPE' offers you a less restrictive in silico evolutionary playground than other tools and I hope you enjoy testing your hypotheses.","An implementation of sensitivity analysis for phylogenetic comparative\r\n methods. The package is an umbrella of statistical and graphical methods that \r\n estimate and report different types of uncertainty in PCM:\r\n (i) Species Sampling uncertainty (sample size; influential species and clades).\r\n (ii) Phylogenetic uncertainty (different topologies and/or branch lengths).\r\n (iii) Data uncertainty (intraspecific variation and measurement error).","Structural equation modeling is a powerful statistical approach for the testing of networks of direct and indirect theoretical causal relationships in complex data sets with inter-correlated dependent and independent variables. Here we implement a simple method for spatially explicit structural equation modeling based on the analysis of variance co-variance matrices calculated across a range of lag distances. This method provides readily interpreted plots of the change in path coefficients across scale.","Here we describe a simple geometric rigid rotation approach that removes the effect of random translation and rotation, enabling the morphological analysis of 3D articulated structures. Our method is based on Cartesian coordinates in 3D space so it can be applied to any morphometric problem that also uses 3D coordinates. See Vidal-Garca, M., Bandara, L., Keogh, J.S. (2018) &lt;doi:10.1002/ece3.4018&gt;.","\r\n  Analyse species-habitat associations in R. Therefore, information about the \r\n  location of the species is needed and about the environmental conditions. To test \r\n  for significance habitat associations, one of the two components is randomized. \r\n  Methods are mainly based on Plotkin et al. (2000) &lt;doi:10.1006/jtbi.2000.2158&gt; and \r\n  Harms et al. (2001) &lt;doi:10.1111/j.1365-2745.2001.00615.x&gt;.","A platform for experimenting with spatially explicit individual-based vegetation models.","Supports safe and reproducible solutions development in R.\r\n \r\n  It will help you with environment separation per project,\r\n \r\n  dependency management, local packages creation and preparing\r\n \r\n  deployment packs for your solutions.","Animal movement models including moving-resting process\r\n \r\n  with embedded Brownian motion according to\r\n \r\n  Yan et al. (2014) &lt;doi:10.1007/s10144-013-0428-8&gt;,\r\n \r\n  Pozdnyakov et al. (2017) &lt;doi:10.1007/s11009-017-9547-6&gt;,\r\n \r\n  Brownian motion with measurement error according to\r\n \r\n  Pozdnyakov et al. (2014) &lt;doi:10.1890/13-0532.1&gt;,\r\n \r\n  and moving-resting-handling process with embedded Brownian motion,\r\n \r\n  Pozdnyakov et al. (2018) &lt;arXiv:1806.00849&gt;.","An interface to the web service methods provided by the United States Department of Agriculture (USDA). The Agricultural Research Service (ARS) provides a large set of databases. The current version of the package holds interfaces to the Systematic Mycology and Microbiology Laboratory (SMML), which consists of four databases: Fungus-Host Distributions, Specimens, Literature and the Nomenclature database. It provides functions for querying these databases. The main function is \\code{associations}, which allows searching for fungus-host combinations.","Metapackage for implementing a variety of event-based models, with\r\n \r\n  a focus on spatially explicit models. These include raster-based,\r\n \r\n  event-based, and agent-based models. The core simulation components\r\n \r\n  (provided by 'SpaDES.core') are built upon a discrete event simulation (DES;\r\n \r\n  see Matloff (2011) ch 7.8.3 &lt;https://nostarch.com/artofr.htm&gt;)\r\n \r\n  framework that facilitates modularity, and easily enables the user to\r\n \r\n  include additional functionality by running user-built simulation modules\r\n \r\n  (see also 'SpaDES.tools'). Included are numerous tools to visualize rasters\r\n \r\n  and other maps (via 'quickPlot'), and caching methods for reproducible\r\n \r\n  simulations (via 'reproducible'). Additional functionality is provided by\r\n \r\n  the 'SpaDES.addins' and 'SpaDES.shiny' packages.","A generic tool for manifold analysis. It allows to infer a relevant transition or evolutionary path which can highlights the features involved in a specific process. 'spathial' can be useful in all the scenarios where the temporal (or pseudo-temporal) evolution is the main problem (e.g. tumor progression). The algorithm for finding the principal path is described in: Ferrarotti et al., (2019) &lt;doi:10.1109/TNNLS.2018.2884792&gt;.\"","SPECIES is an R package for estimation of species richness\r\n        or diversity.","Sampling surface simulation is useful in the comparison of different areal sampling methods\r\n \r\n  in forestry, ecology and natural resources. The sampSurf package allows the simulation \r\n \r\n of numerous sampling methods for standing trees and downed woody debris in a spatial context.\r\n \r\n It also provides an S4 class and method structure that facilitates the addition of new sampling\r\n \r\n methods.","Analysis of species limits and DNA barcoding data. Included are functions for generating important summary statistics from DNA barcode data, assessing specimen identification efficacy, testing and optimizing divergence threshold limits, assessment of diagnostic nucleotides, and calculation of the probability of reciprocal monophyly. Additionally, a sliding window function offers opportunities to analyse information across a gene, often used for marker design in degraded DNA studies. Further information on the package has been published in Brown et al (2012) &lt;doi:10.1111/j.1755-0998.2011.03108.x&gt;.","A set of functions that can be used to spatially thin\r\n    species occurrence data. The resulting thinned data can be used in ecological\r\n    modeling, such as ecological niche modeling.","Allows to map species richness and endemism based on stacked\r\n \r\n  species distribution models (SSDM). Individuals SDMs can be created using a\r\n \r\n  single or multiple algorithms (ensemble SDMs). For each species, an SDM can\r\n \r\n  yield a habitat suitability map, a binary map, a between-algorithm variance\r\n \r\n  map, and can assess variable importance, algorithm accuracy, and between-\r\n \r\n  algorithm correlation. Methods to stack individual SDMs include summing\r\n \r\n  individual probabilities and thresholding then summing. Thresholding can be\r\n \r\n  based on a specific evaluation metric or by drawing repeatedly from a Bernoulli\r\n \r\n  distribution. The SSDM package also provides a user-friendly interface.","Species sensitivity distributions are \r\n  cumulative probability distributions which are fitted to \r\n  toxicity concentrations for different species as described by\r\n  Posthuma et al.(2001) &lt;isbn:9781566705783&gt;.\r\n  The ssdtools package uses Maximum Likelihood to fit distributions \r\n  such as the log-normal, gamma, burr Type-III, log-logistic, \r\n  log-Gumbel, Gompertz and Weibull.\r\n  The user can provide custom distributions.\r\n  Multiple distributions can be averaged using Information Criteria.\r\n  Confidence intervals on hazard concentrations and proportions are produced by \r\n  parametric bootstrapping.","A coalescent simulator that allows the rapid simulation of\r\n \r\n  biological sequences under neutral models of evolution. Different to other\r\n \r\n  coalescent based simulations, it has an optional approximation parameter that\r\n \r\n  allows for high accuracy while maintaining a linear run time cost for long\r\n \r\n  sequences. It is optimized for simulating massive data sets as produced by Next-\r\n \r\n  Generation Sequencing technologies for up to several thousand sequences.","Functions for post processing the outcomes of ecological niche models. It further include methods to build stability maps based on species distribution models under climate change projections and to reduce environmental data.","Publication bias, the fact that studies identified for inclusion in a meta analysis do not represent all studies on the topic of interest, is commonly recognized as a threat to the validity of the results of a meta analysis. One way to explicitly model publication bias is via selection models or weighted probability distributions. In this package we provide implementations of several parametric and nonparametric weight functions. The novelty in Rufibach (2011) is the proposal of a non-increasing variant of the nonparametric weight function of Dear &amp; Begg (1992). The new approach potentially offers more insight in the selection process than other methods, but is more flexible than parametric approaches. To maximize the log-likelihood function proposed by Dear &amp; Begg (1992) under a monotonicity constraint we use a differential evolution algorithm proposed by Ardia et al (2010a, b) and implemented in Mullen et al (2009). In addition, we offer a method to compute a confidence interval for the overall effect size theta, adjusted for selection bias as well as a function that computes the simulation-based p-value to assess the null hypothesis of no selection as described in Rufibach (2011, Section 6).","Routines for identifying metabolic subpathways mediated by microRNAs (miRNAs) through topologically locating miRNAs and genes within reconstructed Kyoto Encyclopedia of Genes and Genomes (KEGG) metabolic pathway graphs embedded by miRNAs. (1) This package can obtain the reconstructed KEGG metabolic pathway graphs with genes and miRNAs as nodes, through converting KEGG metabolic pathways to graphs with genes as nodes and compounds as edges, and then integrating miRNA-target interactions verified by low-throughput experiments from four databases (TarBase, miRecords, mirTarBase and miR2Disease) into converted pathway graphs. (2) This package can locate metabolic subpathways mediated by miRNAs by topologically analyzing the \"lenient distance\" of miRNAs and genes within reconstructed KEGG metabolic pathway graphs.(3) This package can identify significantly enriched miRNA-mediated metabolic subpathways based on located subpathways by hypergenomic test. (4) This package can support six species for metabolic subpathway identification, such as caenorhabditis elegans, drosophila melanogaster, danio rerio, homo sapiens, mus musculus and rattus norvegicus, and user only need to update interested organism-specific environment variables.","Analyse species-habitat associations in R. Therefore, information about the \r\n \r\n  location of the species is needed and about the environmental conditions. To test \r\n \r\n  for significance habitat associations, one of the two components is randomized. \r\n \r\n  Methods are mainly based on Plotkin et al. (2000) &lt;doi:10.1006/jtbi.2000.2158&gt; and \r\n \r\n  Harms et al. (2001) &lt;doi:10.1111/j.1365-2745.2001.00615.x&gt;.","An implementation of a phylogenetic comparative method. It can fit univariate among-species Ornstein-Uhlenbeck models of phenotypic trait evolution, where the trait evolves towards a primary optimum. The optimum can be modelled as a single parameter, as multiple discrete regimes on the phylogenetic tree, and/or with continuous covariates. See also Hansen (1997) &lt;doi:10.2307/2411186&gt;, Butler &amp; King (2004) &lt;doi:10.1086/426002&gt;, Hansen et al. (2008) &lt;doi:10.1111/j.1558-5646.2008.00412.x&gt;.","Software that leverages the capabilities of Circos by manipulating data, preparing configuration files, and running the Perl-native Circos directly from the R environment with minimal user intervention. Circos is a novel software that addresses the challenges in visualizing genetic data by creating circular ideograms composed of tracks of heatmaps, scatter plots, line plots, histograms, links between common markers, glyphs, text, and etc. Please see &lt;http://www.circos.ca&gt;.","Handling taxonomic lists through objects of class 'taxlist'.\r\n    This package provides functions to import species lists from 'Turboveg'\r\n    (&lt;https://www.synbiosys.alterra.nl/turboveg&gt;) and the possibility to create\r\n    backups from resulting R-objects.\r\n    Also quick displays are implemented as summary-methods.","An integrated set of extensions to the 'ergm' package to analyze and simulate network evolution based on exponential-family random graph models (ERGM). 'tergm' is a part of the 'statnet' suite of packages for network analysis.","Routines for creating, manipulating, and performing \r\n \r\n  Bayesian inference about Gaussian processes in \r\n \r\n  one and two dimensions using the Fourier basis approximation: \r\n \r\n  simulation and plotting of processes, calculation of \r\n \r\n  coefficient variances, calculation of process density, \r\n \r\n  coefficient proposals (for use in MCMC). It uses R environments to\r\n \r\n  store GP objects as references/pointers.","Functionalities to perform Stability Analysis of Genotype by Environment Interaction (GEI) to identify superior and stable genotypes under diverse environments. It performs Eberhart &amp; Russel's ANOVA (1966) (&lt;doi:10.2135/cropsci1966.0011183X000600010011x&gt;), Finlay and Wilkinson (1963) Joint Linear Regression (&lt;doi:10.1071/AR9630742&gt;), Wricke (1962, 1964) Ecovalence, Shukla's stability variance parameter (1972) (&lt;doi:10.1038/hdy.1972.87&gt;) and Kang's (1991) (&lt;doi:10.2134/agronj1991.00021962008300010037x&gt;) simultaneous selection for high yielding and stable parameter.","Plant ecologists often need to collect \"traits\" data\r\n    about plant species which are often scattered among various\r\n    databases: TR8 contains a set of tools which take care of\r\n    automatically retrieving some of those functional traits data\r\n    for plant species from publicly available databases (Biolflor,\r\n    The Ecological Flora of the British Isles, LEDA traitbase, Ellenberg\r\n    values for Italian Flora, Mycorrhizal intensity databases, Catminat, BROT,\r\n    PLANTS, Jepson Flora Project).\r\n    The TR8 name, inspired by \"car plates\" jokes, was chosen since\r\n    it both reminds of the main object of the package and is\r\n    extremely short to type.","Data frame class for storing collective movement data (e.g. fish\r\n \r\n  schools, ungulate herds, baboon troops) collected from GPS trackers or\r\n \r\n  computer vision tracking software.","A collection of commonly used tools for animal movement and other tracking \r\n \r\n  data. Variously distance, angle, bearing, distance-to, bearing-to and speed are \r\n \r\n  provided for geographic data that can be used directly or within 'tidyverse' \r\n \r\n  syntax. Distances and bearings are calculated using modern geodesic methods as \r\n \r\n  provided by Charles F. F. Karney (2013) &lt;doi:10.1007/s00190-012-0578-z&gt; \r\n \r\n  via the 'geodist' and 'geosphere' packages.","In order to smoothly animate the transformation of polygons and \r\n \r\n  paths, many aspects needs to be taken into account, such as differing number \r\n \r\n  of control points, changing center of rotation, etc. The 'transformr' \r\n \r\n  package provides an extensive framework for manipulating the shapes of \r\n \r\n  polygons and paths and can be seen as the spatial brother to the 'tweenr' \r\n \r\n  package.","Treatment and visualization of membrane (selective) transport \r\n    data. Transport profiles involving up to three species are produced as\r\n    publication-ready plots and several membrane performance parameters \r\n    (e.g. separation factors as defined in Koros et al. (1996) \r\n    &lt;doi:10.1351/pac199668071479&gt; and non-linear regression parameters\r\n    for the equations described in Rodriguez de San Miguel et al. (2014)\r\n    &lt;doi:10.1016/j.jhazmat.2014.03.052&gt;) can be obtained. Many widely used \r\n    experimental setups (e.g. membrane physical aging) can be easily studied\r\n    through the package's graphical representations.","Interface to the API for 'TreeBASE' &lt;http://treebase.org&gt;\r\n    from 'R.' 'TreeBASE' is a repository of user-submitted phylogenetic\r\n    trees (of species, population, or genes) and the data used to create\r\n    them.","Adds support for R startup configuration via '.Renviron.d' and '.Rprofile.d' directories in addition to '.Renviron' and '.Rprofile' files. This makes it possible to keep private / secret environment variables separate from other environment variables. It also makes it easier to share specific startup settings by simply copying a file to a directory.","(i) For a given species phylogeny on present day data which is calibrated to calendar-time, a method for estimating maximum likelihood speciation and extinction processes is provided. The method allows for non-constant rates. Rates may change (1) as a function of time, i.e. rate shifts at specified times or mass extinction events (likelihood implemented as LikShifts, optimization as bd.shifts.optim and visualized as bd.shifts.plot) or (2) as a function of the number of species, i.e. density-dependence (likelihood implemented as LikDD and optimization as bd.densdep.optim) or (3) extinction rate may be a function of species age (likelihood implemented as LikAge and optimization as bd.age.optim.matlab). Note that the methods take into account the whole phylogeny, in particular it accounts for the \"pull of the present\" effect. (1-3) can take into account incomplete species sampling, as long as each species has the same probability of being sampled. For a given phylogeny on higher taxa (i.e. all but one species per taxa are missing), where the number of species is known within each higher taxa, speciation and extinction rates can be estimated under model (1) (implemented within LikShifts and bd.shifts.optim with groups !=0). (ii) For a given phylogeny with sequentially sampled tips, e.g. a virus phylogeny, rates can be estimated under a model where rates vary across time using bdsky.stt.optim based on likelihood LikShiftsSTT (extending LikShifts and bd.shifts.optim). Furthermore, rates may vary as a function of host types using LikTypesSTT (multitype branching process extending functions in R package diversitree). This function can furthermore calculate the likelihood under an epidemiological model where infected individuals are first exposed and then infectious.","Simulation methods for phylogenetic trees where (i) all tips are sampled at one time point or (ii) tips are sampled sequentially through time. (i) For sampling at one time point, simulations are performed under a constant rate birth-death process, conditioned on having a fixed number of final tips (sim.bd.taxa()), or a fixed age (sim.bd.age()), or a fixed age and number of tips (sim.bd.taxa.age()). When conditioning on the number of final tips, the method allows for shifts in rates and mass extinction events during the birth-death process (sim.rateshift.taxa()). The function sim.bd.age() (and sim.rateshift.taxa() without extinction) allow the speciation rate to change in a density-dependent way. The LTT plots of the simulations can be displayed using LTT.plot(), LTT.plot.gen() and LTT.average.root(). TreeSim further samples trees with n final tips from a set of trees generated by the common sampling algorithm stopping when a fixed number m&gt;&gt;n of tips is first reached (sim.gsa.taxa()). This latter method is appropriate for m-tip trees generated under a big class of models (details in the sim.gsa.taxa() man page). For incomplete phylogeny, the missing speciation events can be added through simulations (corsim()). (ii) sim.rateshifts.taxa() is generalized to sim.bdsky.stt() for serially sampled trees, where the trees are conditioned on either the number of sampled tips or the age. Furthermore, for a multitype-branching process with sequential sampling, trees on a fixed number of tips can be simulated using sim.bdtypes.stt.taxa(). This function further allows to simulate under epidemiological models with an exposed class. The function sim.genespeciestree() simulates coalescent gene trees within birth-death species trees, and sim.genetree() simulates coalescent gene trees.","Innovative Trend Analysis is a graphical method to examine the trends in time series data. Sequential Mann-Kendall test uses the intersection of prograde and retrograde series to indicate the possible change point in time series data. Distribution free cumulative sum charts indicate location and significance of the change point in time series.\r\n \r\n  Zekai, S. (2011). &lt;doi:10.1061/(ASCE)HE.1943-5584.0000556&gt;.\r\n \r\n  Grayson, R. B. et al. (1996). Hydrological Recipes: Estimation Techniques in Australian Hydrology. Cooperative Research Centre for Catchment Hydrology, Australia, p. 125. \r\n \r\n  Sneyers, S. (1990). On the statistical analysis of series of observations. Technical note no 5 143, WMO No 725 415. Secretariat of the World Meteorological Organization, Geneva, 192 pp.","Classical methods for combining summary data from genome-wide\r\n \r\n  association studies (GWAS) only use marginal genetic effects and power can\r\n \r\n  be compromised in the presence of heterogeneity. 'subgxe' is a R package\r\n \r\n  that implements p-value assisted subset testing for association (pASTA),\r\n \r\n  a method developed by Yu et al. (2019) &lt;doi:10.1159/000496867&gt;. pASTA\r\n \r\n  generalizes association analysis based on subsets by incorporating\r\n \r\n  gene-environment interactions into the testing procedure.","Functions for accessing and manipulating spatial data for animal\r\n \r\n  tracking, with straightforward coercion from and to other formats. Filter\r\n \r\n  for speed and create time spent maps from animal track data. There are\r\n \r\n  coercion methods to convert between 'trip' and 'ltraj' from 'adehabitatLT', \r\n \r\n  and between 'trip' and 'psp' and 'ppp' from 'spatstat'. Trip objects\r\n \r\n  can be created from raw or grouped data frames, and from types in the 'sp', \r\n \r\n  'sf', 'amt', 'trackeR', 'mousetrap', and other packages.","This is a framework that aims to provide methods and tools for assessing the impact of different sources of uncertainties (e.g.positional uncertainty) on performance of species distribution models (SDMs).)","Provides an interface to a 'HashiCorp' vault server over\r\n \r\n  its http API (typically these are self-hosted; see\r\n \r\n  &lt;https://www.vaultproject.io&gt;). This allows for secure storage and\r\n \r\n  retrieval of secrets over a network, such as tokens, passwords and\r\n \r\n  certificates. Authentication with vault is supported through\r\n \r\n  several backends including user name/password and authentication via\r\n \r\n  'GitHub'.","Static and dynamic 3D plots to be used with ordination\r\n \r\n  results and in diversity analysis, especially with the vegan package.","This package computes diversity for community data sets\r\n        using the methods outlined by Jost (2006, 2007). While there\r\n        are differing opinions on the ideal way to calculate diversity\r\n        (e.g. Magurran 2004), this method offers the advantage of\r\n        providing diversity numbers equivalents, independent alpha and\r\n        beta diversities, and the ability to incorporate 'order' (q) as\r\n        a continuous measure of the importance of rare species in the\r\n        metrics. The functions provided in this package largely\r\n        correspond with the equations offered by Jost in the cited\r\n        papers. The package computes alpha diversities, beta\r\n        diversities, gamma diversities, and similarity indices.\r\n        Confidence intervals for diversity measures are calculated\r\n        using a bootstrap method described by Chao et al. (2008).  For\r\n        datasets with many samples (sites, plots), sim.table creates\r\n        tables of all pairwise comparisons possible, and for grouped\r\n        samples sim.groups calculates pairwise combinations of within-\r\n        and between-group comparisons.","Using frequency matrices, very low frequency variants (VLFs) are assessed for amino acid and nucleotide sequences.  The VLFs are then compared to see if they occur in only one member of a species, singleton VLFs, or if they occur in multiple members of a species, shared VLFs.  The amino acid and nucleotide VLFs are then compared to see if they are concordant with one another.  Amino acid VLFs are also assessed to determine if they lead to a change in amino acid residue type, and potential changes to protein structures.","Designed to facilitate the assimilation, analysis and synthesis of animal location and movement data collected by the VEMCO suite of acoustic transmitters and receivers. As well as database and geographic information capabilities the principal feature of VTrack is the qualification and identification of ecologically relevant events from the acoustic detection and sensor data. This procedure condenses the acoustic detection database by orders of magnitude, greatly enhancing the synthesis of acoustic detection data.","Dynamic interaction refers to spatial-temporal associations\r\n    in the movements of two (or more) animals. This package provides tools for\r\n    calculating a suite of indices used for quantifying dynamic interaction with\r\n    wildlife telemetry data. For more information on each of the methods employed\r\n    see the references within. The package (as of version 0.3) also has new tools for\r\n    automating contact analysis in large tracking datasets. The package draws \r\n    heavily on the classes and methods developed in the 'adehabitat' packages.","Miscellaneous functions for SciViews or general use: manage a\r\n \r\n  temporary environment attached to the search path for temporary variables you\r\n \r\n  do not want to save() or load(), test if Aqua, Mac, Win, ... Show progress\r\n \r\n  bar, etc.","Data analysis of proteomics experiments by mass spectrometry is supported by this collection of functions mostly dedicated to the analysis of (bottom-up) quantitative (XIC) data. \r\n    Fasta-formatted proteomes (eg from Uniprot) can be read with automatic parsing and multiple annotation types (like species origin, abbreviated gene names, etc) extracted. \r\n    Quantitative proteomics measurements frequently contain multiple NA values, due to physical absence of given peptides in some samples, limitations in sensitivity or other reasons. \r\n    The functions provided here help to inspect graphically the data to investigate the nature of NA-values via their respective replicate measurements and to help/confirm the choice of NA-replacement by low random values. \r\n    Dedicated filtering and statistical testing using the framework of package 'limma' &lt;doi:10.18129/B9.bioc.limma&gt; can be run, enhanced by multiple rounds of NA-replacements to provide robustness towards rare stochastic events. \r\n    Multi-species samples, as frequently used in benchmark-tests (eg Navarro et al 2016 &lt;doi:10.1038/nbt.3685&gt;, Ramus et al 2016 &lt;doi:10.1016/j.jprot.2015.11.011&gt;), can be run with special options separating the data into sub-groups during normalization and testing. \r\n    Subsequently, ROC curves can be constructed to compare multiple analysis approaches.","Functions to compute compositional turnover using zeta-diversity,\r\n    the number of species shared by multiple assemblages. The package includes\r\n    functions to compute zeta-diversity for a specific number of\r\n    assemblages and to compute zeta-diversity for a range of numbers of\r\n    assemblages. It also includes functions to explain how zeta-diversity\r\n    varies with distance and with differences in environmental variables\r\n    between assemblages, using generalised linear models, linear models\r\n    with negative constraints, generalised additive models,shape\r\n    constrained additive models, and I-splines.","Implementation of estimator ACME, described in Wolpert (2015), ACME: A  \t\tPartially Periodic Estimator of Avian &amp; Chiropteran Mortality at Wind\n        Turbines (submitted). Unlike most other models, this estimator\n        supports decreasing-hazard Weibull model for persistence;\n        decreasing search proficiency as carcasses age; variable\n        bleed-through at successive searches; and interval mortality\n        estimates. The package provides, based on search data, functions\n        for estimating the mortality inflation factor in Frequentist and\n        Bayesian settings.","Toolset for the exploration of genetic and genomic\n    data. Adegenet provides formal (S4) classes for storing and handling\n    various genetic data, including genetic markers with varying ploidy\n    and hierarchical population structure ('genind' class), alleles counts\n    by populations ('genpop'), and genome-wide SNP data ('genlight'). It\n    also implements original multivariate methods (DAPC, sPCA), graphics,\n    statistical tests, simulation tools, distance and similarity measures,\n    and several spatial methods. A range of both empirical and simulated\n    datasets is also provided to illustrate various methods.","Multivariate tools to analyze comparative data, i.e. a phylogeny\n    and some traits measured for each taxa.","Extract and process bird sightings records from\n    eBird (&lt;http://ebird.org&gt;), an online tool for recording bird\n    observations.  Public access to the full eBird database is via the\n    eBird Basic Dataset (EBD; see &lt;http://ebird.org/ebird/data/download&gt;\n    for access), a downloadable text file. This package is an interface to\n    AWK for extracting data from the EBD based on taxonomic, spatial, or\n    temporal filters, to produce a manageable file size that can be\n    imported into R.","The North American Breeding Bird Survey (BBS) is a long-running\n  program that seeks to monitor the status and trends of the breeding birds in\n  North America. Since its start in 1966, the BBS has accumulated over 50 years \n  of data for over 500 species of North American Birds. Given the temporal and \n  spatial structure of the data, hierarchical Bayesian models are used to assess \n  the status and trends of these 500+ species of birds. 'bbsBayes' allows you to perform \n  hierarchical Bayesian analysis of BBS data. You can run a full\n  model analysis for one or more species that you choose, or you can take\n  more control and specify how the data should be stratified, prepared\n  for 'JAGS', or modelled. The functions provided here allow you to replicate\n  analyses performed by the United State Geological Survey (USGS, see Link \n  and Sauer (2011) &lt;doi:10.1525/auk.2010.09220&gt;) and Canadian Wildlife Service\n  (CWS, see Smith and Edwards (2020) &lt;doi:10.1101/2020.03.26.010215&gt;).","Extract, visualize and summarize aerial movements of birds and\n    insects from weather radar data. See &lt;doi:10.1111/ecog.04028&gt;\n    for a software paper describing package and methodologies.","R functions to read EURING data and analyse re-encounter data of birds marked by metal rings. For a tutorial, go to &lt;http://www.tandfonline.com/doi/full/10.1080/03078698.2014.933053&gt;.","A highly scientific and utterly addictive \n  bird point count simulator \n  to test statistical assumptions, aid survey design,\n  and have fun while doing it.\n  The simulations follow time-removal and distance sampling models \n  based on Matsuoka et al. (2012) &lt;doi:10.1525/auk.2012.11190&gt;,\n  Solymos et al. (2013) &lt;doi:10.1111/2041-210X.12106&gt;,\n  and Solymos et al. (2018) &lt;doi:10.1650/CONDOR-18-32.1&gt;,\n  and sound attenuation experiments by \n  Yip et al. (2017) &lt;doi:10.1650/CONDOR-16-93.1&gt;.","The number of bird or bat fatalities from collisions with buildings, towers or wind energy turbines can be estimated based on carcass searches and experimentally assessed carcass persistence times and searcher efficiency. Functions for estimating the probability that a bird or bat that died is found by a searcher are provided. Further functions calculate the posterior distribution of the number of fatalities based on the number of carcasses found and the estimated detection probability.","Contains functions to detect and visualise periods of climate\n    sensitivity (climate windows) for a given biological response. \n    Please see van de Pol et al. (2016) &lt;doi:10.1111/2041-210X.12590&gt; \n    and Bailey and van de Pol (2016) &lt;doi:10.1371/journal.pone.0167980&gt; for details.","A simple way of fitting detection functions to distance sampling\n    data for both line and point transects. Adjustment term selection, left and\n    right truncation as well as monotonicity constraints and binning are\n    supported. Abundance and density estimates can also be calculated (via a\n    Horvitz-Thompson-like estimator) if survey area information is provided. See\n    Miller et al. (2019) &lt;doi:10.18637/jss.v089.i01&gt; for more information on\n    methods and &lt;https://examples.distancesampling.org/&gt; for example analyses.","Utilities to represent, visualize, filter, analyse, and summarize\n\t     time-depth recorder (TDR) data.  Miscellaneous functions for\n\t     handling location data are also provided.","Density surface modelling of line transect data. A Generalized\n    Additive Model-based approach is used to calculate spatially-explicit estimates\n    of animal abundance from distance sampling (also presence/absence and strip\n    transect) data. Several utility functions are provided for model checking,\n    plotting and variance estimation.","Tools to download, map, plot and analyze eBird\n    Status and Trends data\n    (&lt;https://ebird.org/science/status-and-trends&gt;). eBird\n    (&lt;https://ebird.org/home&gt;) is a global database of bird observations\n    collected by citizen scientists. eBird Status and Trends uses these\n    data to analyze continental bird abundances, range boundaries,\n    habitats, and trends.","Obtain estimated marginal means (EMMs) for many linear, generalized \n  linear, and mixed models. Compute contrasts or linear functions of EMMs,\n  trends, and comparisons of slopes. Plots and other displays.\n  Least-squares means are discussed, and the term \"estimated marginal means\"\n  is suggested, in Searle, Speed, and Milliken (1980) Population marginal means \n  in the linear model: An alternative to least squares means, The American \n  Statistician 34(4), 216-221 &lt;doi:10.1080/00031305.1980.10483031&gt;.","Functions for range estimation in birds based on Pennycuick (2008)\n    and Pennycuick (1975), 'Flight' program which compliments Pennycuick (2008)\n    requires manual entry of birds which can be tedious when there are thousands\n    of birds to estimate. Implemented are two ODE methods discussed in Pennycuick (1975)\n    and time-marching computation method \"constant muscle mass\" as in Pennycuick (1998).\n    See Pennycuick (1975, ISBN:978-0-12-249405-5), Pennycuick (1998) &lt;doi:10.1006/jtbi.1997.0572&gt;,\n    and Pennycuick (2008, ISBN:9780080557816).","Functions for range estimation in birds based on Pennycuick (2008)\n    and Pennycuick (1975), 'Flight' program which compliments Pennycuick (2008)\n    requires manual entry of birds which can be tedious when there are hundreds\n    of birds to estimate. Implemented are two ODE methods discussed in Pennycuick (1975)\n    and time-marching computation methods as in Pennycuick (1998) and Pennycuick (2008).\n    See Pennycuick (1975, ISBN:978-0-12-249405-5), Pennycuick (1998) &lt;doi:10.1006/jtbi.1997.0572&gt;,\n    and Pennycuick (2008, ISBN:9780080557816).","Provides basic functions for global\n    positioning based on light intensity measurements over time.\n    Positioning process includes the determination of sun events, a\n    discrimination of residency and movement periods, the\n    calibration of period-specific data and, finally, the\n    calculation of positions.","Read, manipulate, and digitize landmark data, generate shape\n    variables via Procrustes analysis for points, curves and surfaces, perform\n    shape analyses, and provide graphical depictions of shapes and patterns of\n    shape variation.","Rstudio add-in that delivers a graphical interface for editing 'ggplot2' theme elements.","Fit linear and generalized linear mixed-effects models.\n    The models and their components are represented using S4 classes and\n    methods.  The core computational algorithms are implemented using the\n    'Eigen' C++ library for numerical linear algebra and 'RcppEigen' \"glue\".","Fits Multivariate Generalised Linear Mixed Models (and related models) using Markov chain Monte Carlo techniques. ","Generalized additive (mixed) models, some of their extensions and \n             other generalized ridge regression with multiple smoothing \n             parameter estimation by (Restricted) Marginal Likelihood, \n             Generalized Cross Validation and similar, or using iterated \n             nested Laplace approximation for fully Bayesian inference. See \n             Wood (2017) &lt;doi:10.1201/9781315370279&gt; for an overview. \n             Includes a gam() function, a wide variety of smoothers, 'JAGS' \n             support and distributions beyond the exponential family. ","Creates and runs Bayesian mixing models to analyze\n    biological tracer data (i.e. stable isotopes, fatty acids), which estimate the\n    proportions of source (prey) contributions to a mixture (consumer). 'MixSIAR'\n    is not one model, but a framework that allows a user to create a mixing model\n    based on their data structure and research questions, via options for fixed/\n    random effects, source data types, priors, and error terms. 'MixSIAR' incorporates\n    several years of advances since 'MixSIR' and 'SIAR'.","Download and processing functionalities for the Moderate Resolution\n    Imaging Spectroradiometer (MODIS). The package provides automated access to the\n    global online data archives LP DAAC (&lt;https://lpdaac.usgs.gov/&gt;), LAADS \n    (&lt;https://ladsweb.modaps.eosdis.nasa.gov/&gt;) and NSIDC (&lt;https://nsidc.org/&gt;) \n    as well as processing capabilities such as file conversion, mosaicking, \n    subsetting and time series filtering.","Programmatic interface to the Oak Ridge National Laboratories\n    'MODIS Land Products Subsets' web services \n    (&lt;https://modis.ornl.gov/data/modis_webservice.html&gt;). Allows for easy\n    downloads of 'MODIS' time series directly to your R workspace or\n    your computer.","Acoustic template detection and monitoring database interface. Create, modify, save, and use templates for detection of animal vocalizations. View, verify, and extract results. Upload a MySQL schema to a existing instance, manage survey metadata, write and read templates and detections locally or to the database. ","Functions to estimate start and duration of moult from moult \n             data, based on models developed in Underhill \n             and Zucchini (1988, 1990). ","Animal abundance estimation via conventional, multiple covariate\n    and mark-recapture distance sampling (CDS/MCDS/MRDS). Detection function\n    fitting is performed via maximum likelihood. Also included are diagnostics\n    and plotting for fitted detection functions. Abundance estimation is via a\n    Horvitz-Thompson-like estimator.","Tools for performing model selection and model averaging. Automated\n             model selection through subsetting the maximum model, with optional\n             constraints for model inclusion. Model parameter and prediction\n             averaging based on model weights derived from information criteria\n             (AICc and alike) or custom model weighting schemes.","Provides utility functions, distributions, and methods for improving Markov chain Monte Carlo (MCMC) sampling efficiency for ecological Spatial Capture-Recapture (SCR) and Open Population Spatial Capture-Recapture (OPSCR) models.  The methods provided implement the techniques presented in \"Estimation of Large-Scale Spatial Capture-Recapture Models\" (Turek, et al (2020) &lt;doi:10.1101/2020.05.07.081182&gt;).","Functions for non-spatial and spatial open-population capture-recapture analysis.","Download and import of 'OpenStreetMap' ('OSM') data as 'sf'\n    or 'sp' objects.  'OSM' data are extracted from the 'Overpass' web\n    server (&lt;https://overpass-api.de/&gt;) and processed with very fast 'C++'\n    routines for return to 'R'.","The 'ggplot2' package provides a strong API for sequentially \n    building up a plot, but does not concern itself with composition of multiple\n    plots. 'patchwork' is a package that expands the API to allow for \n    arbitrarily complex composition of plots by, among others, providing \n    mathematical operators for combining multiple plots. Other packages that try \n    to address this need (but with a different approach) are 'gridExtra' and \n    'cowplot'.","This package provides a set of functions useful when\n        evaluating the results of presence-absence models. Package\n        includes functions for calculating threshold dependent measures\n        such as confusion matrices, pcc, sensitivity, specificity, and\n        Kappa, and produces plots of each measure as the threshold is\n        varied. It will calculate optimal threshold choice according to\n        a choice of optimization criteria. It also includes functions\n        to plot the threshold independent ROC curves along with the\n        associated AUC (area under the curve).","Performs goodness-of-fit tests for capture-recapture models. Also contains several functions to process capture-recapture data.","Reading, writing, manipulating, analyzing and modeling of spatial data. The package implements basic and high-level functions for raster data and for vector data operations such as intersections. See the manual and tutorials on &lt;https://rspatial.org/&gt; to get started.","Estimation of abundance and other demographic parameters for closed \n             populations, open populations and the robust design in capture-recapture  \n             experiments using loglinear models.   ","A programmatic client for the eBird database \n    (&lt;https://ebird.org/home&gt;), including functions for searching for bird \n    observations by geographic location (latitude, longitude), eBird \n    hotspots, location identifiers, by notable sightings, by region, and by \n    taxonomic name.","Resource Selection (Probability) Functions\n  for use-availability wildlife data\n  based on weighted distributions as described in\n  Lele and Keim (2006) &lt;doi:10.1890/0012-9658(2006)87%5B3021:WDAEOR%5D2.0.CO;2&gt;,\n  Lele (2009) &lt;doi:10.2193/2007-535&gt;,\n  and Solymos &amp; Lele (2016) &lt;doi:10.1111/2041-210X.12432&gt;.","A programmatic interface to the Web Service methods\n    provided by the Global Biodiversity Information Facility ('GBIF';\n    &lt;https://www.gbif.org/developer/summary&gt;). 'GBIF' is a database\n    of species occurrence records from sources all over the globe.\n    'rgbif' includes functions for searching for taxonomic names,\n    retrieving information on data providers, getting species occurrence\n    records, getting counts of occurrence records, and using the 'GBIF'\n    tile map service to make 'rasters' summarizing huge amounts of data.","A programmatic interface to the API provided by the 'iNaturalist' website &lt;https://www.inaturalist.org/&gt; to download species occurrence data submitted by citizen scientists.","Interface to the JAGS MCMC library.","Three methods to calculate R2 for models with correlated errors, \n    including Phylogenetic GLS, Phylogenetic Logistic Regression, Linear Mixed \n    Models (LMMs), and Generalized Linear Mixed Models (GLMMs). See details in \n    Ives 2018 &lt;doi:10.1093/sysbio/syy060&gt;.","'IUCN' Red List (&lt;http://apiv3.iucnredlist.org/api/v3/docs&gt;) client.\n    The 'IUCN' Red List is a global list of threatened and endangered species.\n    Functions cover all of the Red List 'API' routes. An 'API' key is required.","User-facing R functions are provided to parse, compile, test,\n    estimate, and analyze Stan models by accessing the header-only Stan library\n    provided by the 'StanHeaders' package. The Stan project develops a probabilistic\n    programming language that implements full Bayesian statistical inference\n    via Markov Chain Monte Carlo, rough Bayesian inference via 'variational'\n    approximation, and (optionally penalized) maximum likelihood estimation via\n    optimization. In all three cases, automatic differentiation is used to quickly\n    and accurately evaluate gradients without burdening the user with the need to\n    derive the partial derivatives.","Estimates previously compiled regression models using the 'rstan'\n    package, which provides the R interface to the Stan C++ library for Bayesian\n    estimation. Users specify models via the customary R syntax with a formula and\n    data.frame plus some additional arguments for priors.","Functions for analysing, manipulating, displaying, editing and synthesizing time waves (particularly sound).  This package processes time analysis (oscillograms and envelopes), spectral content, resonance quality factor, entropy, cross correlation and autocorrelation, zero-crossing, dominant frequency, analytic signal, frequency coherence, 2D and 3D spectrograms and many other analyses.","Support for simple features, a standardized way to\n    encode spatial vector data. Binds to 'GDAL' for reading and writing\n    data, to 'GEOS' for geometrical operations, and to 'PROJ' for\n    projection conversions and datum transformations. Optionally uses the 's2'\n    package for spherical geometry operations on geographic coordinates.","Simulates the cultural evolution of quantitative traits of bird song. 'SongEvo' is an individual- (agent-) based model. 'SongEvo' is spatially-explicit and can be parameterized with, and tested against, measured song data. Functions are available for model implementation, sensitivity analyses, parameter optimization, model validation, and hypothesis testing. ","Classes and methods for spatial\n  data; the classes document where the spatial location information\n  resides, for 2D or 3D data. Utility functions are provided, e.g. for\n  plotting data as maps, spatial selection, as well as methods for\n  retrieving coordinates, for subsetting, print, summary, etc.","Interacts with a suite of web 'APIs' for taxonomic tasks,\n    such as getting database specific taxonomic identifiers, verifying\n    species names, getting taxonomic hierarchies, fetching downstream and\n    upstream taxonomic names, getting taxonomic synonyms, converting\n    scientific to common names and vice versa, and more.","The 'tidyverse' is a set of packages that work in harmony\n    because they share common data representations and 'API' design. This\n    package is designed to make it easy to install and load multiple\n    'tidyverse' packages in a single step. Learn more about the\n    'tidyverse' at &lt;https://www.tidyverse.org&gt;.","Thematic maps are geographical maps in which spatial data distributions are visualized. This package offers a flexible, layer-based, and easy to use approach to create thematic maps, such as choropleths and bubble maps.","Analyze music and speech, extract features like MFCCs, handle wave files and their representation in various ways, read mp3, read midi, perform steps of a transcription, ...\n        Also contains functions ported from the 'rastamat' 'Matlab' package.","Provides a convenient interface for constructing plots to visualize the fit of regression models arising from a wide variety of models in R ('lm', 'glm', 'coxph', 'rlm', 'gam', 'locfit', 'lmer', 'randomForest', etc.)","Provides means for downloading historical weather data from \n    the Environment and Climate Change Canada website \n    (&lt;https://climate.weather.gc.ca/historical_data/search_historic_data_e.html&gt;). \n    Data can be downloaded from multiple stations and over large date ranges \n    and automatically processed into a single dataset. Tools are also provided \n    to identify stations either by name or proximity to a location."],["packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","packagefinder","volePhD","volePhD","volePhD","StatsForBios","StatsForBios","StatsForBios","samdeanscience","samdeanscience","samdeanscience","taoofcoffee\n","taoofcoffee\n","taoofcoffee","taoofcoffee","BenMMarshall","BenMMarshall","BenMMarshall","BenMMarshall","Thorbjrn Sievert","Pedro Peres-Neto (@com_ecology)","Jiangshan Lai","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","Russell Gray","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo","beausoleilmo"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Name<\/th>\n      <th>Short Description<\/th>\n      <th>Long Description<\/th>\n      <th>Source<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
